<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.6.33">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

    <meta name="author" content="Simon Oiry¹">
    <meta name="author" content="Bede Ffinian Rowe Davies¹">
    <meta name="author" content="Pierre Gernez¹">
    <meta name="author" content="Laurent Barillé¹">
    <meta name="dcterms.date" content="2024-12-17">
    <meta name="keywords" content="Remote Sensing, Invasive species, Coastal Ecosystems, Biodiversity">

    <title>Spatial and temporal distribution of the invasive Gracilaria vermiculophylla through remote sensing at its first described European site</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
      /* CSS for citations */
      div.csl-bib-body { }
      div.csl-entry {
        clear: both;
        margin-bottom: 0em;
      }
      .hanging-indent div.csl-entry {
        margin-left:2em;
        text-indent:-2em;
      }
      div.csl-left-margin {
        min-width:2em;
        float:left;
      }
      div.csl-right-inline {
        margin-left:2em;
        padding-left:1em;
      }
      div.csl-indent {
        margin-left: 2em;
      }    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-07ba0ad10f5680c660e360ac31d2f3b6.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-545cdd561de3535e9b364c647f747893.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
     <link href="site_libs/tabwid-1.1.3/tabwid.css" rel="stylesheet">

<script src="site_libs/tabwid-1.1.3/tabwid.js"></script>
   <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script> 
      <meta name="citation_title" content="Spatial and temporal distribution of the invasive *Gracilaria vermiculophylla* through remote sensing at its first described European site">
<meta name="citation_abstract" content="To be Written">
<meta name="citation_keywords" content="Remote Sensing,Invasive species,Coastal Ecosystems,Biodiversity">
<meta name="citation_author" content="Simon Oiry¹">
<meta name="citation_author" content="Bede Ffinian Rowe Davies¹">
<meta name="citation_author" content="Pierre Gernez¹">
<meta name="citation_author" content="Laurent Barillé¹">
<meta name="citation_publication_date" content="2024-12-17">
<meta name="citation_cover_date" content="2024-12-17">
<meta name="citation_year" content="2024">
<meta name="citation_online_date" content="2024-12-17">
<meta name="citation_language" content="en">
<meta name="citation_journal_title" content="Remote Sensing of Environment">
<meta name="citation_reference" content="citation_title=Life history and molecular sequences of gracilaria vermiculophylla (gracilariales, rhodophyta), a new introduction to european waters;,citation_author=Jan Rueness;,citation_publication_date=2005;,citation_cover_date=2005;,citation_year=2005;,citation_issue=1;,citation_volume=44;,citation_journal_title=Phycologia;,citation_publisher=Taylor &amp;amp;amp; Francis;">
<meta name="citation_reference" content="citation_title=Smoothing and differentiation of data by simplified least squares procedures.;,citation_author=Abraham Savitzky;,citation_author=Marcel JE Golay;,citation_publication_date=1964;,citation_cover_date=1964;,citation_year=1964;,citation_issue=8;,citation_volume=36;,citation_journal_title=Analytical chemistry;,citation_publisher=ACS Publications;">
<meta name="citation_reference" content="citation_title=Integrating airborne remote sensing and field campaigns for ecology and earth system science;,citation_author=K Dana Chadwick;,citation_author=Philip G Brodrick;,citation_author=Kathleen Grant;,citation_author=Tristan Goulden;,citation_author=Amanda Henderson;,citation_author=Nicola Falco;,citation_author=Haruko Wainwright;,citation_author=Kenneth H Williams;,citation_author=Markus Bill;,citation_author=Ian Breckheimer;,citation_author=others;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_issue=11;,citation_volume=11;,citation_journal_title=Methods in Ecology and Evolution;,citation_publisher=Wiley Online Library;">
<meta name="citation_reference" content="citation_title=Spread dynamics of invasive species;,citation_author=Matı́as Arim;,citation_author=Sebastián R Abades;,citation_author=Paula E Neill;,citation_author=Mauricio Lima;,citation_author=Pablo A Marquet;,citation_publication_date=2006;,citation_cover_date=2006;,citation_year=2006;,citation_issue=2;,citation_volume=103;,citation_journal_title=Proceedings of the National Academy of Sciences;,citation_publisher=National Acad Sciences;">
<meta name="citation_reference" content="citation_title=Different flowering phenology of alien invasive species in spain: Evidence for the use of an empty temporal niche?;,citation_author=O Godoy;,citation_author=P Castro-Dı́ez;,citation_author=F Valladares;,citation_author=M Costa-Tenorio;,citation_publication_date=2009;,citation_cover_date=2009;,citation_year=2009;,citation_issue=6;,citation_volume=11;,citation_journal_title=Plant Biology;,citation_publisher=Wiley Online Library;">
<meta name="citation_reference" content="citation_title=A global-scale screening of non-native aquatic organisms to identify potentially invasive species under current and future climate conditions;,citation_author=Lorenzo Vilizzi;,citation_author=Gordon H Copp;,citation_author=Jeffrey E Hill;,citation_author=Boris Adamovich;,citation_author=Luke Aislabie;,citation_author=Daniel Akin;,citation_author=Abbas J Al-Faisal;,citation_author=David Almeida;,citation_author=MN Amal Azmai;,citation_author=Rigers Bakiu;,citation_author=others;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_volume=788;,citation_journal_title=Science of the Total Environment;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Remonter le temps;,citation_author=undefined IGN;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_publisher=https://remonterletemps.ign.fr; Institut national de l’information géographique et forestière (IGN);">
<meta name="citation_reference" content="citation_title=Mapping species distributions with MAXENT using a geographically biased sample of presence data: A performance assessment of methods for correcting sampling bias;,citation_author=Yoan Fourcade;,citation_author=Jan O Engler;,citation_author=Dennis Rödder;,citation_author=Jean Secondi;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_issue=5;,citation_volume=9;,citation_journal_title=PloS one;,citation_publisher=Public Library of Science San Francisco, USA;">
<meta name="citation_reference" content="citation_title=Low altitude spatial assessment and monitoring of intertidal seagrass meadows beyond the visible spectrum using a remotely piloted aircraft system;,citation_author=Subhash Chand;,citation_author=Barbara Bollard;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_volume=255;,citation_journal_title=Estuarine, Coastal and Shelf Science;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Predicted changes in the biogeographical range of gracilaria vermiculophylla under present and future climate scenarios;,citation_abstract=Global change effects have favoured the introduction of new species in marine ecosystems in recent years. Gracilaria vermiculophylla, a red seaweed native from the north-eastern Pacific, has successfully colonised large regions in the Northern Hemisphere. In this research, we implemented species distribution models (SDMs) to (i) examine which were the most important environmental factors defining the presence of G. vermiculophylla at a global scale, and (ii) determine the potential current and future distribution of G. vermiculophylla based on two climate scenarios (representative concentration pathways (RCP 2.6 and RCP 8.5)). Our results suggest that temperature and salinity were the most important variables explaining the distribution of the target species. Additionally, the SDMs for present climate settings showed a potential wider distribution than is recorded to date. In addition, a subtle habitat expansion of 2.9° into higher latitudes was reported under the RCP 2.6 scenario by the end of this century. The high-carbon-emission scenario (RCP 8.5) delivered a potential large habitat expansion (6.0°), even reaching arctic latitudes, and a remarkable habitat loss of 11° in its southern distribution range. SMDs also forecasted suitable areas for this species in the Southern Hemisphere, pointing toward a potential global expansion in the coming decades.;,citation_author=Clara Mendoza-Segura;,citation_author=Emilio Fernández;,citation_author=Pedro Beca-Carretero;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.mdpi.com/2077-1312/11/2/367;,citation_issue=2;,citation_doi=10.3390/jmse11020367;,citation_issn=2077-1312;,citation_volume=11;,citation_journal_title=Journal of Marine Science and Engineering;">
<meta name="citation_reference" content="citation_title=Mapping estuarine habitats using airborne hyperspectral imagery, with special focus on seagrass meadows;,citation_author=Mireia Valle;,citation_author=Vicenc Pala;,citation_author=Virgine Lafon;,citation_author=Aurelie Dehouck;,citation_author=Joxe Mikel Garmendia;,citation_author=Angel Borja;,citation_author=Guillem Chust;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_volume=164;,citation_journal_title=Estuarine, Coastal and Shelf Science;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Seasonal and spatial variability in patchiness of microphytobenthos on intertidal flats from sentinel-2 satellite imagery;,citation_author=Tisja D Daggers;,citation_author=Peter MJ Herman;,citation_author=Daphne Van der Wal;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_volume=7;,citation_journal_title=Frontiers in Marine Science;,citation_publisher=Frontiers Media SA;">
<meta name="citation_reference" content="citation_title=Long-term analysis of zostera noltei: A retrospective approach for understanding seagrasses’ dynamics;,citation_author=Felipe Calleja;,citation_author=Cristina Galván;,citation_author=Ana Silió-Calzada;,citation_author=José A Juanes;,citation_author=Bárbara Ondiviela;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_volume=130;,citation_journal_title=Marine environmental research;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Increase in seagrass distribution at bourgneuf bay (france) detected by spatial remote sensing;,citation_author=Laurent Barillé;,citation_author=Marc Robin;,citation_author=Nicolas Harin;,citation_author=Annaëlle Bargain;,citation_author=Patrick Launeau;,citation_publication_date=2010;,citation_cover_date=2010;,citation_year=2010;,citation_issue=3;,citation_volume=92;,citation_journal_title=Aquatic Botany;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Processus adaptatifs des végétaux marins face au changement climatique à différentes échelles de temps et d’espace: Dynamique de populations, métabolomique, écophysiologie et potentiels de valorisation;,citation_author=Gwladys Surget;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_dissertation_institution=Université de Bretagne occidentale-Brest;">
<meta name="citation_reference" content="citation_title=Spreading and autoecology of the invasive species gracilaria vermiculophylla (gracilariales, rhodophyta) in the lagoons of the north-western adriatic sea (mediterranean sea, italy);,citation_author=Adriano Sfriso;,citation_author=MARION ADELHEID Wolf;,citation_author=S Maistro;,citation_author=Katia Sciuto;,citation_author=Isabella Moro;,citation_publication_date=2012;,citation_cover_date=2012;,citation_year=2012;,citation_volume=114;,citation_journal_title=Estuarine, Coastal and Shelf Science;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Review of gracilaria vermiculophylla (ohmi) papenfuss and other species in japan and asia;,citation_author=R Terada;,citation_author=H Yamamoto;,citation_publication_date=2002;,citation_cover_date=2002;,citation_year=2002;,citation_volume=8;,citation_journal_title=Taxonomy of economic seaweeds, with special reference to Pacific species;">
<meta name="citation_reference" content="citation_title=Monitoring the marine invasive alien species rugulopteryx okamurae using unmanned aerial vehicles and satellites;,citation_author=Mar Roca;,citation_author=Martha Bonnet Dunbar;,citation_author=Alejandro Román;,citation_author=Isabel Caballero;,citation_author=Maria Laura Zoffoli;,citation_author=Pierre Gernez;,citation_author=Gabriel Navarro;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_volume=9;,citation_journal_title=Frontiers in Marine Science;,citation_publisher=Frontiers Media SA;">
<meta name="citation_reference" content="citation_title=Quantifying intertidal macroalgae stocks in the NW iberian peninsula using unmanned aerial vehicle (UAV) multispectral imagery;,citation_author=Miguel J Peidro-Devesa;,citation_author=Andrea Martı́nez-Movilla;,citation_author=Juan Luis Rodrı́guez-Somoza;,citation_author=Joaquı́n Martı́nez Sánchez;,citation_author=Marta Román;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_journal_title=Regional Studies in Marine Science;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Seaweed habitats on the shore: Characterization through hyperspectral UAV imagery and field sampling;,citation_abstract=Intertidal macroalgal habitats are major components of temperate coastal ecosystems. Their distribution was studied using field sampling and hyperspectral remote mapping on a rocky shore of Porspoder (western Brittany, France). Covers of both dominating macroalgae and the sessile fauna were characterized in situ at low tide in 24 sampling spots, according to four bathymetric levels. A zone of ca. 17,000 m2 was characterized using a drone equipped with a hyperspectral camera. Macroalgae were identified by image processing using two classification methods to assess the representativeness of spectral classes. Finally, a comparison of the remote imaging data to the field sampling data was conducted. Seven seaweed classes were distinguished by hyperspectral pictures, including five different species of Fucales. The maximum likelihood (MLC) and spectral angle mapper (SAM) were both trained using image-derived spectra. MLC was more accurate to classify the main dominating species (Overall Accuracy (OA) 95.1;,citation_author=Wendy Diruit;,citation_author=Anthony Le Bris;,citation_author=Touria Bajjouk;,citation_author=Sophie Richier;,citation_author=Mathieu Helias;,citation_author=Thomas Burel;,citation_author=Marc Lennon;,citation_author=Alexandre Guyot;,citation_author=Erwan Ar Gall;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://www.mdpi.com/2072-4292/14/13/3124;,citation_issue=13;,citation_doi=10.3390/rs14133124;,citation_issn=2072-4292;,citation_volume=14;,citation_journal_title=Remote Sensing;">
<meta name="citation_reference" content="citation_title=Spatial assessment of intertidal seagrass meadows using optical imaging systems and a lightweight drone;,citation_author=James P Duffy;,citation_author=Laura Pratt;,citation_author=Karen Anderson;,citation_author=Peter E Land;,citation_author=Jamie D Shutler;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_volume=200;,citation_journal_title=Estuarine, Coastal and Shelf Science;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Introduced marine macroalgae and habitat modifiers: Their ecological role and significant attributes;,citation_author=Cecilia D Nyberg;,citation_publication_date=2007;,citation_cover_date=2007;,citation_year=2007;">
<meta name="citation_reference" content="citation_title=Terra: Spatial data analysis;,citation_author=Robert J. Hijmans;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=terra;">
<meta name="citation_reference" content="citation_title=Shiny: Web application framework for r;,citation_author=Winston Chang;,citation_author=Joe Cheng;,citation_author=JJ Allaire;,citation_author=Carson Sievert;,citation_author=Barret Schloerke;,citation_author=Yihui Xie;,citation_author=Jeff Allen;,citation_author=Jonathan McPherson;,citation_author=Alan Dipert;,citation_author=Barbara Borges;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=shiny;">
<meta name="citation_reference" content="citation_title=Shiny app for validation dataset building;,citation_author=Oiry Simon;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://oirysimon.shinyapps.io/shiny_validate/;,citation_publisher=https://oirysimon.shinyapps.io/shiny_validate/; ShinyApps.io;">
<meta name="citation_reference" content="citation_title=An overview of marine non-indigenous species found in three contrasting biogeographic metropolitan french regions: Insights on distribution, origins and pathways of introduction;,citation_abstract=Biological invasions are one of the main global threats to biodiversity in terrestrial, freshwater and marine ecosystems worldwide, requiring effective inventorying and monitoring programs. Here, we present an updated list of non-indigenous species in French marine and transitional waters. Focused on eukaryote pluricellular species found throughout the three metropolitan French marine regions (Western Mediterranean Sea, Bay of Biscay and the Northern Seas), a total of 342 non-indigenous, including 42 cryptogenic, species are listed as having been introduced since the 13th century. The majority of the species originated from the temperate Northern Pacific. They mainly arrived through both ballast and hull fouling and also are associated with shellfish farming activities. Most of them have been introduced since the 1970s, a time when maritime and aquaculture trade intensified. Despite important human-aided opportunities for species transfer between the three marine regions (for instance, via recreational boating or aquaculture transfers), only a third of these NIS are common to all regions, as expected due to their environmental specificities.;,citation_author=Cécile Massé;,citation_author=Frédérique Viard;,citation_author=Suzie Humbert;,citation_author=Elvire Antajan;,citation_author=Isabelle Auby;,citation_author=Guy Bachelet;,citation_author=Guillaume Bernard;,citation_author=Vincent M. P. Bouchet;,citation_author=Thomas Burel;,citation_author=Jean-Claude Dauvin;,citation_author=Alice Delegrange;,citation_author=Sandrine Derrien-Courtel;,citation_author=Gabin Droual;,citation_author=Benoit Gouillieux;,citation_author=Philippe Goulletquer;,citation_author=Laurent Guérin;,citation_author=Anne-Laure Janson;,citation_author=Jérôme Jourde;,citation_author=Céline Labrune;,citation_author=Nicolas Lavesque;,citation_author=Jean-Charles Leclerc;,citation_author=Michel Le Duff;,citation_author=Vincent Le Garrec;,citation_author=Pierre Noël;,citation_author=Antoine Nowaczyk;,citation_author=Christine Pergent-Martini;,citation_author=Jean-Philippe Pezy;,citation_author=Aurore Raoux;,citation_author=Virginie Raybaud;,citation_author=Sandrine Ruitton;,citation_author=Pierre-Guy Sauriau;,citation_author=Nicolas Spilmont;,citation_author=Delphine Thibault;,citation_author=Dorothée Vincent;,citation_author=Amelia Curd;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.mdpi.com/1424-2818/15/2/161;,citation_issue=2;,citation_doi=10.3390/d15020161;,citation_issn=1424-2818;,citation_volume=15;,citation_journal_title=Diversity;">
<meta name="citation_reference" content="citation_title=Introduction into france of the japanese oyster (crassostrea gigas);,citation_author=Henri Grizel;,citation_author=Maurice Heral;,citation_publication_date=1991;,citation_cover_date=1991;,citation_year=1991;,citation_issue=3;,citation_volume=47;,citation_journal_title=ICES Journal of Marine Science;,citation_publisher=Oxford University Press;">
<meta name="citation_reference" content="citation_title=Using 1st derivative reflectance signatures within a remote sensing framework to identify macroalgae in marine environments;,citation_abstract=Macroalgae blooms (MABs) are a global natural hazard that are likely to increase in occurrence with climate change and increased agricultural runoff. MABs can cause major issues for indigenous species, fish farms, nuclear power stations, and tourism activities. This project focuses on the impacts of MABs on the operations of a British nuclear power station. However, the outputs and findings are also of relevance to other coastal operators with similar problems. Through the provision of an early-warning detection system for MABs, it should be possible to minimize the damaging effects and possibly avoid them altogether. Current methods based on satellite imagery cannot be used to detect low-density mobile vegetation at various water depths. This work is the first step towards providing a system that can warn a coastal operator 6–8 h prior to a marine ingress event. A fundamental component of such a warning system is the spectral reflectance properties of the problematic macroalgae species. This is necessary to optimize the detection capability for the problematic macroalgae in the marine environment. We measured the reflectance signatures of eight species of macroalgae that we sampled in the vicinity of the power station. Only wavelengths below 900 nm (700 nm for similarity percentage (SIMPER)) were analyzed, building on current methodologies. We then derived 1st derivative spectra of these eight sampled species. A multifaceted univariate and multivariate approach was used to visualize the spectral reflectance, and an analysis of similarities (ANOSIM) provided a species-level discrimination rate of 85;,citation_author=Ben Mcilwaine;,citation_author=Monica Rivas Casado;,citation_author=Paul Leinster;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=https://www.mdpi.com/2072-4292/11/6/704;,citation_issue=6;,citation_doi=10.3390/rs11060704;,citation_issn=2072-4292;,citation_volume=11;,citation_journal_title=Remote Sensing;">
<meta name="citation_reference" content="citation_title=How far can we classify macroalgae remotely? An example using a new spectral library of species from the south west atlantic (argentine patagonia);,citation_author=O Magalı́ Olmedo-Masat;,citation_author=M Paula Raffo;,citation_author=Daniel Rodrı́guez-Pérez;,citation_author=Marianela Arijón;,citation_author=Noela Sánchez-Carnero;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_issue=23;,citation_volume=12;,citation_journal_title=Remote Sensing;,citation_publisher=MDPI;">
<meta name="citation_reference" content="citation_title=New hyperspectral procedure to discriminate intertidal macroalgae;,citation_abstract=The recent development and miniaturization of hyperspectral sensors embedded in drones has allowed the acquisition of hyperspectral images with high spectral and spatial resolution. The characteristics of both the embedded sensors and drones (viewing angle, flying altitude, resolution) create opportunities to consider the use of hyperspectral imagery to map and monitor macroalgae communities. In general, the overflight of the areas to be mapped is conconmittently associated accompanied with measurements carried out in the field to acquire the spectra of previously identified objects. An alternative to these simultaneous acquisitions is to use a hyperspectral library made up of pure spectra of the different species in place, that would spare field acquisition of spectra during each flight. However, the use of such a technique requires developed appropriate procedure for testing the level of species classification that can be achieved, as well as the reproducibility of the classification over time. This study presents a novel classification approach based on the use of reflectance spectra of macroalgae acquired in controlled conditions. This overall approach developed is based on both the use of the spectral angle mapper (SAM) algorithm applied on first derivative hyperspectral data. The efficiency of this approach has been tested on a hyperspectral library composed of 16 macroalgae species, and its temporal reproducibility has been tested on a monthly survey of the spectral response of different macro-algae species. In addition, the classification results obtained with this new approach were also compared to the results obtained through the use of the most recent and robust procedure published. The classification obtained shows that the developed approach allows to perfectly discriminate the different phyla, whatever the period. At the species level, the classification approach is less effective when the individuals studied belong to phylogenetically close species (i.e., Fucus spiralis and Fucus serratus).;,citation_author=Florian Douay;,citation_author=Charles Verpoorter;,citation_author=Gwendoline Duong;,citation_author=Nicolas Spilmont;,citation_author=François Gevaert;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://www.mdpi.com/2072-4292/14/2/346;,citation_issue=2;,citation_doi=10.3390/rs14020346;,citation_issn=2072-4292;,citation_volume=14;,citation_journal_title=Remote Sensing;">
<meta name="citation_reference" content="citation_title=Multi-and hyperspectral classification of soft-bottom intertidal vegetation using a spectral library for coastal biodiversity remote sensing;,citation_author=Bede Ffinian Rowe Davies;,citation_author=Pierre Gernez;,citation_author=Andréa Geraud;,citation_author=Simon Oiry;,citation_author=Philippe Rosa;,citation_author=Maria Laura Zoffoli;,citation_author=Laurent Barillé;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_volume=290;,citation_journal_title=Remote Sensing of Environment;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Linking microphytobenthos distribution and mudflat geomorphology under varying sedimentary regimes using unoccupied aerial vehicle (UAV)-acquired multispectral reflectance and photogrammetry;,citation_author=Tristan J Douglas;,citation_author=Nicholas C Coops;,citation_author=Mark C Drever;,citation_author=Brian PV Hunt;,citation_author=Tara G Martin;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_journal_title=Science of The Total Environment;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Evolution of the french guiana coast from late pleistocene to holocene based on chenier and beach sand dating;,citation_author=Guillaume Brunier;,citation_author=Toru Tamura;,citation_author=Edward J Anthony;,citation_author=Philippe Dussouillez;,citation_author=Antoine Gardel;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=4;,citation_volume=22;,citation_journal_title=Regional Environmental Change;,citation_publisher=Springer;">
<meta name="citation_reference" content="citation_title=Discriminating seagrasses from green macroalgae in european intertidal areas using high-resolution multispectral drone imagery;,citation_abstract=Coastal areas support seagrass meadows, which offer crucial ecosystem services, including erosion control and carbon sequestration. However, these areas are increasingly impacted by human activities, leading to habitat fragmentation and seagrass decline. In situ surveys, traditionally performed to monitor these ecosystems, face limitations on temporal and spatial coverage, particularly in intertidal zones, prompting the addition of satellite data within monitoring programs. Yet, satellite remote sensing can be limited by too coarse spatial and/or spectral resolutions, making it difficult to discriminate seagrass from other macrophytes in highly heterogeneous meadows. Drone (unmanned aerial vehicle—UAV) images at a very high spatial resolution offer a promising solution to address challenges related to spatial heterogeneity and the intrapixel mixture. This study focuses on using drone acquisitions with a ten spectral band sensor similar to that onboard Sentinel-2 for mapping intertidal macrophytes at low tide (i.e., during a period of emersion) and effectively discriminating between seagrass and green macroalgae. Nine drone flights were conducted at two different altitudes (12 m and 120 m) across heterogeneous intertidal European habitats in France and Portugal, providing multispectral reflectance observation at very high spatial resolution (8 mm and 80 mm, respectively). Taking advantage of their extremely high spatial resolution, the low altitude flights were used to train a Neural Network classifier to discriminate five taxonomic classes of intertidal vegetation: Magnoliopsida (Seagrass), Chlorophyceae (Green macroalgae), Phaeophyceae (Brown algae), Rhodophyceae (Red macroalgae), and benthic Bacillariophyceae (Benthic diatoms), and validated using concomitant field measurements. Classification of drone imagery resulted in an overall accuracy of 94;,citation_author=Simon Oiry;,citation_author=Bede Ffinian Rowe Davies;,citation_author=Ana I. Sousa;,citation_author=Philippe Rosa;,citation_author=Maria Laura Zoffoli;,citation_author=Guillaume Brunier;,citation_author=Pierre Gernez;,citation_author=Laurent Barillé;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://www.mdpi.com/2072-4292/16/23/4383;,citation_issue=23;,citation_doi=10.3390/rs16234383;,citation_issn=2072-4292;,citation_volume=16;,citation_journal_title=Remote Sensing;">
<meta name="citation_reference" content="citation_title=Systematics of the gracilariales (rhodophyta) including new subfamilies, tribes, subgenera, and two new genera, agarophyton gen. Nov. And crassa gen. nov.;,citation_author=Carlos Frederico D Gurgel;,citation_author=James N Norris;,citation_author=William E Schmidt;,citation_author=Hau Nhu Le;,citation_author=Suzanne Fredericq;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_issue=1;,citation_volume=374;,citation_journal_title=Phytotaxa;">
<meta name="citation_reference" content="citation_title=CONTRIBUTIONS TO THE KNOWLEDGE OF GRACILARIACEAE FROM JAPAN: Ⅱ. On a new species of the genus gracilariopsis, with some considerations on its ecology;,citation_author=Hikoei OHMI;,citation_publication_date=1956;,citation_cover_date=1956;,citation_year=1956;,citation_issue=4;,citation_volume=6;,citation_journal_title=北海道大學水産學部研究彙報;,citation_publisher=北海道大学水産学部;">
<meta name="citation_reference" content="citation_title=Notes on algal nomenclature - v. Various chlorophyceae and rhodophyceae;,citation_author=G. F. Papenfuss;,citation_publication_date=1967;,citation_cover_date=1967;,citation_year=1967;,citation_fulltext_html_url=https://marinespecies.org/aphia.php?p=sourcedetails&amp;amp;amp;id=303450;,citation_volume=5;,citation_journal_title=Phykos;">
<meta name="citation_reference" content="citation_title=Assessment of the segmentation of RGB remote sensing images: A subjective approach;,citation_author=Giruta Kazakeviciute-Januskeviciene;,citation_author=Edgaras Janusonis;,citation_author=Romualdas Bausys;,citation_author=Tadas Limba;,citation_author=Mindaugas Kiskis;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_issue=24;,citation_volume=12;,citation_journal_title=Remote Sensing;,citation_publisher=MDPI;">
<meta name="citation_reference" content="citation_title=Precision aquaculture drone mapping of the spatial distribution of kappaphycus alvarezii biomass and carrageenan;,citation_abstract=The aquaculture of Kappaphycus alvarezii (Kappaphycus hereafter) seaweed has rapidly expanded among coastal communities in Indonesia due to its relatively simple farming process, low capital costs and short production cycles. This species is mainly cultivated for its carrageenan content used as a gelling agent in the food industry. To further assist producers in improving cultivation management and providing quantitative information about the yield, a novel approach involving remote sensing techniques was tested. In this study, multispectral images obtained from a drone (Unoccupied Aerial Vehicle, UAV) were processed to estimate the fresh and carrageenan weights of Kappaphycus at a cultivation site in South Sulawesi. The UAV imagery was geometrically and radiometrically corrected, and the resulting orthomosaics were used for detecting and classifying Kappaphycus using a random forest algorithm. The classification results were combined with in situ measurements of Kappaphycus fresh weight and carrageenan content using empirical relations between the area and weight of fresh seaweed/carrageenan. This approach allowed quantifying seaweed biometry and biochemistry at single cultivation lines and cultivation plot scales. Fresh seaweed and carrageenan weights were estimated for different dates within three distinct cultivation cycles, and the daily growth rate for each cycle was derived. Data were upscaled to a small family-scale farm and a large-scale leader farm and compared with previous estimations. To our knowledge, this study provides, for the first time, an estimation of yield at the scale of cultivation lines by exploiting the very high spatial resolution of drone data. Overall, the use of UAV remote sensing proved to be a promising approach for seaweed monitoring, opening the way to precision aquaculture of Kappaphycus.;,citation_author=Nurjannah Nurdin;,citation_author=Evangelos Alevizos;,citation_author=Rajuddin Syamsuddin;,citation_author=Hasni Asis;,citation_author=Elmi Nurhaidah Zainuddin;,citation_author=Agus Aris;,citation_author=Simon Oiry;,citation_author=Guillaume Brunier;,citation_author=Teruhisa Komatsu;,citation_author=Laurent Barillé;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.mdpi.com/2072-4292/15/14/3674;,citation_issue=14;,citation_doi=10.3390/rs15143674;,citation_issn=2072-4292;,citation_volume=15;,citation_journal_title=Remote Sensing;">
<meta name="citation_reference" content="citation_title=Remote sensing image scene classification: Benchmark and state of the art;,citation_author=Gong Cheng;,citation_author=Junwei Han;,citation_author=Xiaoqiang Lu;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_issue=10;,citation_volume=105;,citation_journal_title=Proceedings of the IEEE;,citation_publisher=IEEE;">
<meta name="citation_reference" content="citation_title=Variability of kappaphycus alvarezii cultivation in south-sulawesi (indonesia) related to the monsoon shift: Water quality, growth and colour quantification;,citation_author=Laurent Barillé;,citation_author=Iona LR Paterson;,citation_author=Simon Oiry;,citation_author=Agus Aris;,citation_author=Elizabeth J Cook-Cottier;,citation_author=Nurjannah Nurdin;,citation_publication_date=2025;,citation_cover_date=2025;,citation_year=2025;,citation_volume=40;,citation_journal_title=Aquaculture Reports;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=A global review of the distribution, taxonomy, and impacts of introduced seaweeds;,citation_author=Susan L Williams;,citation_author=Jennifer E Smith;,citation_publication_date=2007;,citation_cover_date=2007;,citation_year=2007;,citation_issue=1;,citation_volume=38;,citation_journal_title=Annu. Rev. Ecol. Evol. Syst.;,citation_publisher=Annual Reviews;">
<meta name="citation_reference" content="citation_title=Remote sensing in seagrass ecology: Coupled dynamics between migratory herbivorous birds and intertidal meadows observed by satellite during four decades;,citation_author=Maria Laura Zoffoli;,citation_author=Pierre Gernez;,citation_author=Simon Oiry;,citation_author=Laurent Godet;,citation_author=Sébastien Dalloyau;,citation_author=Bede Ffinian Rowe Davies;,citation_author=Laurent Barillé;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=3;,citation_volume=9;,citation_journal_title=Remote Sensing in Ecology and Conservation;,citation_publisher=Wiley Online Library;">
<meta name="citation_reference" content="citation_title=Status and trends in the rate of introduction of marine non-indigenous species in european seas;,citation_abstract=Invasive alien species are a major worldwide driver of biodiversity change. The current study lists verified records of non-indigenous species (NIS) in European marine waters until 2020, with the purpose of establishing a baseline, assessing trends, and discussing appropriate threshold values for good environmental status (GES) according to the relevant European legislation. All NIS records were verified by national experts and trends are presented in six-year assessment periods from 1970 to 2020 according to the European Union Marine Strategy Framework Directive. Altogether, 874 NIS have been introduced to European marine waters until 2020 with the Mediterranean Sea and North-East Atlantic Ocean hosting most of the introductions. Overall, the number of new introductions has steadily increased since 2000. The annual rate of new introductions reached 21 new NIS in European seas within the last six-year assessment period (2012–2017). This increase is likely due to increased human activities and research efforts that have intensified during the early 21st century within European Seas. As Europe seas are not environmentally, nor geographically homogenous, the setting of threshold values for assessing GES requires regional expertise. Further, once management measures are operational, pathway-specific threshold values would enable assessing the effectiveness of such measures.;,citation_author=Argyro Zenetos;,citation_author=Konstantinos Tsiamis;,citation_author=Marika Galanidi;,citation_author=Natacha Carvalho;,citation_author=Cátia Bartilotti;,citation_author=João Canning-Clode;,citation_author=Luca Castriota;,citation_author=Paula Chainho;,citation_author=Robert Comas-González;,citation_author=Ana C. Costa;,citation_author=Branko Dragičević;,citation_author=Jakov Dulčić;,citation_author=Marco Faasse;,citation_author=Ann-Britt Florin;,citation_author=Arjan Gittenberger;,citation_author=Hans Jakobsen;,citation_author=Anders Jelmert;,citation_author=Francis Kerckhof;,citation_author=Maiju Lehtiniemi;,citation_author=Silvia Livi;,citation_author=Kim Lundgreen;,citation_author=Vesna Macic;,citation_author=Cécile Massé;,citation_author=Borut Mavrič;,citation_author=Rahmat Naddafi;,citation_author=Martina Orlando-Bonaca;,citation_author=Slavica Petovic;,citation_author=Lydia Png-Gonzalez;,citation_author=Aina Carbonell Quetglas;,citation_author=Romeu S. Ribeiro;,citation_author=Tiago Cidade;,citation_author=Sander Smolders;,citation_author=Peter A. U. Stæhr;,citation_author=Frederique Viard;,citation_author=Okko Outinen;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://www.mdpi.com/1424-2818/14/12/1077;,citation_issue=12;,citation_doi=10.3390/d14121077;,citation_issn=1424-2818;,citation_volume=14;,citation_journal_title=Diversity;">
<meta name="citation_reference" content="citation_title=DISCOV: Drone Intertidal Sediment Classification Of Vegetation;,citation_author=Simon Oiry;,citation_author=Bede Ffinian Rowe Davies;,citation_author=Gernez Pierre;,citation_author=Barillé Laurent;,citation_publication_date=2024-11;,citation_cover_date=2024-11;,citation_year=2024;,citation_fulltext_html_url=https://doi.org/10.5281/zenodo.14218984;,citation_doi=10.5281/zenodo.14218984;,citation_publisher=Zenodo;">
<meta name="citation_reference" content="citation_title=Impact of hydrodynamics on community structure and metabolic production of marine biofouling formed in a highly energetic estuary;,citation_author=Aurélie Portas;,citation_author=Nathan Carriot;,citation_author=Annick Ortalo-Magné;,citation_author=Guillaume Damblans;,citation_author=Maxime Thiébaut;,citation_author=Gérald Culioli;,citation_author=Nolwenn Quillien;,citation_author=Jean-François Briand;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_doi=10.1016/j.marenvres.2023.106241;,citation_volume=192;,citation_journal_title=Marine Environmental Research;">
<meta name="citation_reference" content="citation_title=How the origin of sedimentary organic matter impacts the benthic nutrient fluxes in shallow coastal mudflats;,citation_author=Laurent Jeanneau;,citation_author=Emilie Jardé;,citation_author=Justine Louis;,citation_author=Alexandrine Pannard;,citation_author=Marine Liotaud;,citation_author=Françoise Andrieux-Loyer;,citation_author=Gérard Gruau;,citation_author=Florian Caradec;,citation_author=Emilie Rabiller;,citation_author=Nathalie Lebris;,citation_author=Anniet Laverman;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_doi=10.5802/crgeos.228;,citation_volume=355;,citation_journal_title=Comptes Rendus Géoscience;">
<meta name="citation_reference" content="citation_title=Plant landscape and models of french atlantic estuarine systems: Extended summary of the doctoral thesis;,citation_author=Kevin Cianfaglione;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_doi=10.2478/trser-2021-0002;,citation_volume=23;,citation_journal_title=Transylvanian Review of Systematical and Ecological Research;">
<meta name="citation_reference" content="citation_title=Bacterial load mitigation of the shellfish magallana gigas by the marine sponge hymeniacidon perlevis (montagu 1818);,citation_author=Charline Gentric;,citation_author=Pierre Sauleau;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_doi=10.1016/j.rsma.2024.103564;,citation_volume=75;,citation_journal_title=Regional Studies in Marine Science;">
<meta name="citation_reference" content="citation_title=Morpho-sedimentological and dynamic patterns in a ria type estuary: The belon estuary (south brittany, france);,citation_author=Guillaume Michel;,citation_author=Sophie Le Bot;,citation_author=Sandric Lesourd;,citation_author=Robert Lafite;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issue=2;,citation_doi=10.1080/17445647.2021.1925170;,citation_volume=17;,citation_journal_title=Journal of Maps;">
<meta name="citation_reference" content="citation_title=Morphosedimentary evolution of ria-type estuaries;,citation_author=Pierre Castaing;,citation_author=André Guilcher;,citation_publication_date=1995;,citation_cover_date=1995;,citation_year=1995;,citation_issue=4;,citation_doi=10.1002/esp.3290200408;,citation_volume=20;,citation_journal_title=Earth Surface Processes and Landforms;">
<meta name="citation_reference" content="citation_title=Potential influence of confounding factors (size, salinity) on biomarkers in the sentinel species scrobicularia plana used in programmes monitoring estuarine quality;,citation_author=Olivia Fossi Tankoua;,citation_author=Pierre-Emmanuel Buffet;,citation_author=Jean-Claude Amiard;,citation_author=Claude Amiard-Triquet;,citation_author=Catherine Mouneyrac;,citation_author=Brigitte Berthet;,citation_publication_date=2011;,citation_cover_date=2011;,citation_year=2011;,citation_issue=7;,citation_doi=10.1007/s11356-011-0479-3;,citation_volume=18;,citation_journal_title=Environmental Science and Pollution Research;">
<meta name="citation_reference" content="citation_title=Multiscale patterns in the diversity and organization of benthic intertidal fauna among french atlantic estuaries;,citation_author=Hugues Blanchet;,citation_author=Benoît Gouillieux;,citation_author=Sandrine Alizier;,citation_author=others;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_doi=10.1016/j.seares.2014.02.014;,citation_volume=90;,citation_journal_title=Journal of Sea Research;">
<meta name="citation_reference" content="citation_title=Fluxes of dissolved inorganic carbon in three estuarine systems of the cantabrian sea (north of spain);,citation_author=T Ortega;,citation_author=R Ponce;,citation_author=J Forja;,citation_author=A Gómez-Parra;,citation_publication_date=2005;,citation_cover_date=2005;,citation_year=2005;,citation_issue=1-4;,citation_volume=53;,citation_journal_title=Journal of Marine Systems;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Sintering behaviour of ceramic bodies from contaminated marine sediments;,citation_author=Maximina Romero;,citation_author=A Andrés;,citation_author=Rebeca Alonso;,citation_author=Javier Viguri;,citation_author=J Ma Rincón;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_issue=8;,citation_volume=34;,citation_journal_title=Ceramics International;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Using a UAV-mounted multispectral camera for the monitoring of marine macrophytes;,citation_author=Alejandro Román;,citation_author=Antonio Tovar-Sánchez;,citation_author=Irene Olivé;,citation_author=Gabriel Navarro;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_volume=8;,citation_journal_title=Frontiers in Marine Science;,citation_publisher=Frontiers Media SA;">
<meta name="citation_reference" content="citation_title=Mapping intertidal oyster farms using unmanned aerial vehicles (UAV) high-resolution multispectral data;,citation_author=Alejandro Román;,citation_author=Hermansyah Prasyad;,citation_author=Simon Oiry;,citation_author=Bede FR Davies;,citation_author=Guillaume Brunier;,citation_author=Laurent Barillé;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_volume=291;,citation_journal_title=Estuarine, Coastal and Shelf Science;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Mapping intertidal microphytobenthic biomass with very high-resolution remote sensing imagery in an estuarine system;,citation_author=Alejandro Román;,citation_author=Simon Oiry;,citation_author=Bede FR Davies;,citation_author=Philippe Rosa;,citation_author=Pierre Gernez;,citation_author=Antonio Tovar-Sánchez;,citation_author=Gabriel Navarro;,citation_author=Vona Méléder;,citation_author=Laurent Barillé;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_journal_title=Science of The Total Environment;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Agisoft metashape;,citation_author=undefined Agisoft;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=https://www.agisoft.com/;">
<meta name="citation_reference" content="citation_title=A review of photogrammetry and photorealistic 3D models in education from a psychological perspective;,citation_author=Steve Nebel;,citation_author=Maik Beege;,citation_author=Sascha Schneider;,citation_author=Günter Daniel Rey;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_volume=5;,citation_conference_title=Frontiers in education;,citation_conference=Frontiers Media SA;">
<meta name="citation_reference" content="citation_title=Decadal increase in the ecological status of a north-atlantic intertidal seagrass meadow observed with multi-mission satellite time-series;,citation_author=Maria Laura Zoffoli;,citation_author=Pierre Gernez;,citation_author=Laurent Godet;,citation_author=Steef Peters;,citation_author=Simon Oiry;,citation_author=Laurent Barillé;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_volume=130;,citation_journal_title=Ecological Indicators;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=A sentinel watching over inter-tidal seagrass phenology across western europe and north africa;,citation_author=Bede Ffinian Rowe Davies;,citation_author=Simon Oiry;,citation_author=Philippe Rosa;,citation_author=Maria Laura Zoffoli;,citation_author=Ana I Sousa;,citation_author=Oliver R Thomas;,citation_author=Dan A Smale;,citation_author=Melanie C Austen;,citation_author=Lauren Biermann;,citation_author=Martin J Attrill;,citation_author=others;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_issue=1;,citation_volume=5;,citation_journal_title=Communications Earth &amp;amp;amp; Environment;,citation_publisher=Nature Publishing Group UK London;">
<meta name="citation_reference" content="citation_title=Intertidal seagrass extent from sentinel-2 time-series show distinct trajectories in western europe;,citation_author=Bede Ffinian Rowe Davies;,citation_author=Simon Oiry;,citation_author=Philippe Rosa;,citation_author=Maria Laura Zoffoli;,citation_author=Ana I Sousa;,citation_author=Oliver R Thomas;,citation_author=Dan A Smale;,citation_author=Melanie C Austen;,citation_author=Lauren Biermann;,citation_author=Martin J Attrill;,citation_author=others;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_volume=312;,citation_journal_title=Remote Sensing of Environment;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Climate change, human impacts, and coastal ecosystems in the anthropocene;,citation_author=Qiang He;,citation_author=Brian R Silliman;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=19;,citation_volume=29;,citation_journal_title=Current Biology;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Ocean acidification impacts on coastal ecosystem services due to habitat degradation;,citation_author=Jason M Hall-Spencer;,citation_author=Ben P Harvey;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=2;,citation_volume=3;,citation_journal_title=Emerging Topics in Life Sciences;,citation_publisher=Portland Press Ltd.;">
<meta name="citation_reference" content="citation_title=Major threats of pollution and climate change to global coastal ecosystems and enhanced management for sustainability;,citation_author=Yonglong Lu;,citation_author=Jingjing Yuan;,citation_author=Xiaotian Lu;,citation_author=Chao Su;,citation_author=Yueqing Zhang;,citation_author=Chenchen Wang;,citation_author=Xianghui Cao;,citation_author=Qifeng Li;,citation_author=Jilan Su;,citation_author=Venugopalan Ittekkot;,citation_author=others;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_volume=239;,citation_journal_title=Environmental Pollution;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=A global analysis of human settlement in coastal zones;,citation_author=Christopher Small;,citation_author=Robert J Nicholls;,citation_publication_date=2003;,citation_cover_date=2003;,citation_year=2003;,citation_journal_title=Journal of coastal research;,citation_publisher=JSTOR;">
<meta name="citation_reference" content="citation_title=Coastal pollution—an overview;,citation_author=Swapna Mukherjee;,citation_author=Kaushik Kiran Ghosh;,citation_author=Abhra Chanda;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_journal_title=Environmental Oceanography and Coastal Dynamics: Current Scenario and Future Trends;,citation_publisher=Springer;">
<meta name="citation_reference" content="citation_title=Valuing ecosystem services in complex coastal settings: An extended ecosystem accounting framework for improved decision-making;,citation_author=Jeremy De Valck;,citation_author=Diane Jarvis;,citation_author=Anthea Coggan;,citation_author=Ella Schirru;,citation_author=Petina Pert;,citation_author=Victoria Graham;,citation_author=Maxine Newlands;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_volume=155;,citation_journal_title=Marine Policy;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Ecological value of coastal habitats for commercially and ecologically important species;,citation_author=Rochelle D Seitz;,citation_author=Håkan Wennhage;,citation_author=Ulf Bergström;,citation_author=Romuald N Lipcius;,citation_author=Tom Ysebaert;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_issue=3;,citation_volume=71;,citation_journal_title=ICES Journal of Marine Science;,citation_publisher=Oxford University Press;">
<meta name="citation_reference" content="citation_title=Quantifying economic value of coastal ecosystem services: A review;,citation_author=Seyedabdolhossein Mehvar;,citation_author=Tatiana Filatova;,citation_author=Ali Dastgheib;,citation_author=Erik De Ruyter van Steveninck;,citation_author=Roshanka Ranasinghe;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_issue=1;,citation_volume=6;,citation_journal_title=Journal of marine science and engineering;,citation_publisher=MDPI;">
<meta name="citation_reference" content="citation_title=Current status and future prospects for the assessment of marine and coastal ecosystem services: A systematic review;,citation_author=Camino Liquete;,citation_author=Chiara Piroddi;,citation_author=Evangelia G Drakou;,citation_author=Leigh Gurney;,citation_author=Stelios Katsanevakis;,citation_author=Aymen Charef;,citation_author=Benis Egoh;,citation_publication_date=2013;,citation_cover_date=2013;,citation_year=2013;,citation_issue=7;,citation_volume=8;,citation_journal_title=PloS one;,citation_publisher=Public Library of Science San Francisco, USA;">
<meta name="citation_reference" content="citation_title=Introduction: The coastal ecosystem complex as a unit of structure and function of biological productivity in coastal areas;,citation_author=Yoshiro Watanabe;,citation_author=Tomohiko Kawamura;,citation_author=Yoh Yamashita;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_volume=84;,citation_journal_title=Fisheries science;,citation_publisher=Springer;">
<meta name="citation_reference" content="citation_title=The value of estuarine and coastal ecosystem services;,citation_author=Edward B Barbier;,citation_author=Sally D Hacker;,citation_author=Chris Kennedy;,citation_author=Evamaria W Koch;,citation_author=Adrian C Stier;,citation_author=Brian R Silliman;,citation_publication_date=2011;,citation_cover_date=2011;,citation_year=2011;,citation_issue=2;,citation_volume=81;,citation_journal_title=Ecological monographs;,citation_publisher=Wiley Online Library;">
<meta name="citation_reference" content="citation_title=The planetary role of seagrass conservation;,citation_author=Richard KF Unsworth;,citation_author=Leanne C Cullen-Unsworth;,citation_author=Benjamin LH Jones;,citation_author=Richard J Lilley;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=6606;,citation_volume=377;,citation_journal_title=Science;,citation_publisher=American Association for the Advancement of Science;">
<meta name="citation_reference" content="citation_title=Invasive macroalgae in native seagrass beds: Vectors of spread and impacts;,citation_author=Louise B Firth;,citation_author=Andy Foggo;,citation_author=Thomas Watts;,citation_author=Antony M Knights;,citation_author=Stacey DeAmicis;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_issue=1;,citation_volume=133;,citation_journal_title=Annals of Botany;,citation_publisher=Oxford University Press US;">
<meta name="citation_reference" content="citation_title=Nitrogen uptake responses of gracilaria vermiculophylla (ohmi) papenfuss under combined and single addition of nitrate and ammonium;,citation_author=Maria H Abreu;,citation_author=R Pereira;,citation_author=AH Buschmann;,citation_author=I Sousa-Pinto;,citation_author=C Yarish;,citation_publication_date=2011;,citation_cover_date=2011;,citation_year=2011;,citation_issue=2;,citation_volume=407;,citation_journal_title=Journal of Experimental Marine Biology and Ecology;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Multiple effects of a gracilaria vermiculophylla invasion on estuarine mudflat functioning and diversity;,citation_author=D Davoult;,citation_author=G Surget;,citation_author=Valérie Stiger-Pouvreau;,citation_author=F Noisette;,citation_author=P Riera;,citation_author=D Stagnol;,citation_author=T Androuin;,citation_author=N Poupart;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_volume=131;,citation_journal_title=Marine Environmental Research;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Flora and fauna associated with the introduced red alga gracilaria vermiculophylla;,citation_author=Cecilia D Nyberg;,citation_author=Mads S Thomsen;,citation_author=Inger Wallentinus;,citation_publication_date=2009;,citation_cover_date=2009;,citation_year=2009;,citation_issue=3;,citation_volume=44;,citation_journal_title=European Journal of Phycology;,citation_publisher=Taylor &amp;amp;amp; Francis;">
<meta name="citation_reference" content="citation_title=Combining niche shift and population genetic analyses predicts rapid phenotypic evolution during invasion;,citation_author=Erik E Sotka;,citation_author=Aaron W Baumgardner;,citation_author=Paige M Bippus;,citation_author=Christophe Destombe;,citation_author=Elizabeth A Duermit;,citation_author=Hikaru Endo;,citation_author=Ben A Flanagan;,citation_author=Mits Kamiya;,citation_author=Lauren E Lees;,citation_author=Courtney J Murren;,citation_author=others;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_issue=5;,citation_volume=11;,citation_journal_title=Evolutionary Applications;,citation_publisher=Wiley Online Library;">
<meta name="citation_reference" content="citation_title=The invasive red alga gracilaria vermiculophylla in the baltic sea: Adaptation to brackish water may compensate for light limitation;,citation_author=Florian Weinberger;,citation_author=Björn Buchholz;,citation_author=Rolf Karez;,citation_author=Martin Wahl;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_issue=3;,citation_volume=3;,citation_journal_title=Aquatic Biology;">
<meta name="citation_reference" content="citation_title=Genetic identification of source and likely vector of a widespread marine invader;,citation_author=Stacy A Krueger-Hadfield;,citation_author=Nicole M Kollars;,citation_author=Allan E Strand;,citation_author=James E Byers;,citation_author=Sarah J Shainker;,citation_author=Ryuta Terada;,citation_author=Thomas W Greig;,citation_author=Mareike Hammann;,citation_author=David C Murray;,citation_author=Florian Weinberger;,citation_author=others;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_issue=12;,citation_volume=7;,citation_journal_title=Ecology and evolution;,citation_publisher=Wiley Online Library;">
<meta name="citation_reference" content="citation_title=Everywhere you look, everywhere you go, there’s an estuary invaded by the red seaweed gracilaria vermiculophylla (ohmi) papenfuss, 1967;,citation_author=Stacy Krueger-Hadfield;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_issue=4;,citation_volume=7;,citation_journal_title=BioInvasions Records;">
<meta name="citation_reference" content="citation_title=Gracilaria vermiculophylla (ohmi) papenfuss, 1967 (rhodophyta, gracilariaceae) in northern europe, with emphasis on danish conditions, and what to expect in the future;,citation_author=Mads S Thomsen;,citation_author=Peter A Staehr;,citation_author=Cecilia D Nyberg;,citation_author=Steen Schwærter;,citation_author=Dorte Krause-Jensen;,citation_author=Brian R Silliman;,citation_publication_date=2007;,citation_cover_date=2007;,citation_year=2007;,citation_issue=2;,citation_volume=2;,citation_journal_title=Aquatic invasions;,citation_publisher=Regional Euro-Asian Biological Invasions Centre;">
<meta name="citation_reference" content="citation_title=Ocean warming alters the responses to eutrophication in a commercially farmed seaweed, gracilariopsis lemaneiformis;,citation_author=Chunxiang Liu;,citation_author=Dinghui Zou;,citation_author=Zhiwei Liu;,citation_author=Changpeng Ye;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_volume=847;,citation_journal_title=Hydrobiologia;,citation_publisher=Springer;">
<meta name="citation_reference" content="citation_title=Effects of the invasive macroalgae gracilaria vermiculophylla on two co-occurring foundation species and associated invertebrates;,citation_author=Mads S Thomsen;,citation_author=Peter A Stæhr;,citation_author=Lars Nejrup;,citation_author=David R Schiel;,citation_publication_date=2013;,citation_cover_date=2013;,citation_year=2013;,citation_issue=2;,citation_volume=8;,citation_journal_title=Aquatic Invasions;,citation_publisher=Regional Euro-Asian Biological Invasions Centre;">
<meta name="citation_reference" content="citation_title=Reintroduction of eelgrass (zostera marina l.) in the dutch wadden sea: A research overview and management vision;,citation_author=MM Van Katwijk;,citation_publication_date=2003;,citation_cover_date=2003;,citation_year=2003;,citation_conference_title=Challenges to the wadden sea area. In: Proceedings of the 10th international scientific wadden sea symposium, groningen, the netherlands;">
<meta name="citation_reference" content="citation_title=Warming impacts on early life stages increase the vulnerability and delay the population recovery of a long-lived habitat-forming macroalga;,citation_author=Pol Capdevila;,citation_author=Bernat Hereu;,citation_author=Roberto Salguero-Gómez;,citation_author=Graciel Rovira;,citation_author=Alba Medrano;,citation_author=Emma Cebrian;,citation_author=Joaquim Garrabou;,citation_author=Diego K Kersting;,citation_author=Cristina Linares;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=3;,citation_volume=107;,citation_journal_title=Journal of Ecology;,citation_publisher=Wiley Online Library;">
<meta name="citation_reference" content="citation_title=An invasive foundation species enhances multifunctionality in a coastal ecosystem;,citation_author=Aaron P Ramus;,citation_author=Brian R Silliman;,citation_author=Mads S Thomsen;,citation_author=Zachary T Long;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_issue=32;,citation_volume=114;,citation_journal_title=Proceedings of the national academy of sciences;,citation_publisher=National Acad Sciences;">
<meta name="citation_reference" content="citation_title=The global dispersal of the non-endemic invasive red alga gracilaria vermiculophylla in the ecosystems of the euro-asia coastal waters including the wadden sea unesco world heritage coastal area: Awful or awesome?;,citation_author=Vincent Ginneken;,citation_author=Evert Vries;,citation_author=others;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_issue=1;,citation_volume=8;,citation_journal_title=Oceanography &amp;amp;amp; Fisheries Open Access Journal;,citation_publisher=Juniper Publishers Inc.;">
</head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> Article Notebook</h6>

            <a href="./index.qmd" class="btn btn-primary quarto-download-embed" download="index.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Spatial and temporal distribution of the invasive <em>Gracilaria vermiculophylla</em> through remote sensing at its first described European site</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
            <div class="quarto-title-meta-author">
          <div class="quarto-title-meta-heading">Authors</div>
          <div class="quarto-title-meta-heading">Affiliations</div>
          
                <div class="quarto-title-meta-contents">
            <p class="author">Simon Oiry¹ <a href="mailto:oirysimon@gmail.com" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0001-7161-5246" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        
                      </p>
                    <p class="affiliation">
                        Institut des Substances et Organismes de la Mer, ISOMer
                      </p>
                    <p class="affiliation">
                        
                      </p>
                    <p class="affiliation">
                        
                      </p>
                    <p class="affiliation">
                        
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Bede Ffinian Rowe Davies¹ <a href="https://orcid.org/0000-0001-6462-4347" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Pierre Gernez¹ <a href="https://orcid.org/0000-0003-2055-410X" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Laurent Barillé¹ <a href="https://orcid.org/0000-0001-5138-2684" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        
                      </p>
                  </div>
                    </div>
        
        <div class="quarto-title-meta">

                      
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">December 17, 2024</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        To be Written
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>Remote Sensing, Invasive species, Coastal Ecosystems, Biodiversity</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#title-proposition" id="toc-title-proposition" class="nav-link active" data-scroll-target="#title-proposition"><span class="header-section-number">1</span> Title proposition</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction"><span class="header-section-number">2</span> Introduction</a></li>
  <li><a href="#materiel-methods" id="toc-materiel-methods" class="nav-link" data-scroll-target="#materiel-methods"><span class="header-section-number">3</span> Materiel &amp; Methods</a>
  <ul class="collapse">
  <li><a href="#study-sites" id="toc-study-sites" class="nav-link" data-scroll-target="#study-sites"><span class="header-section-number">3.1</span> Study sites</a></li>
  <li><a href="#sec-DroneFlights" id="toc-sec-DroneFlights" class="nav-link" data-scroll-target="#sec-DroneFlights"><span class="header-section-number">3.2</span> Remote sensing data acquisition and pre-processing</a></li>
  <li><a href="#hyperspectral-measurements" id="toc-hyperspectral-measurements" class="nav-link" data-scroll-target="#hyperspectral-measurements"><span class="header-section-number">3.3</span> Hyperspectral measurements</a></li>
  <li><a href="#drone-data" id="toc-drone-data" class="nav-link" data-scroll-target="#drone-data"><span class="header-section-number">3.4</span> Drone data</a></li>
  <li><a href="#scene-classification" id="toc-scene-classification" class="nav-link" data-scroll-target="#scene-classification"><span class="header-section-number">3.5</span> Scene classification</a></li>
  <li><a href="#historical-presence-of-gracilaria-vermiculophylla-in-the-bélon-esturay" id="toc-historical-presence-of-gracilaria-vermiculophylla-in-the-bélon-esturay" class="nav-link" data-scroll-target="#historical-presence-of-gracilaria-vermiculophylla-in-the-bélon-esturay"><span class="header-section-number">3.6</span> Historical Presence of <em>Gracilaria vermiculophylla</em> in the Bélon esturay</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">4</span> Results</a>
  <ul class="collapse">
  <li><a href="#historical-records-in-the-belon-estuary" id="toc-historical-records-in-the-belon-estuary" class="nav-link" data-scroll-target="#historical-records-in-the-belon-estuary"><span class="header-section-number">4.1</span> Historical records in the Belon estuary</a></li>
  <li><a href="#spectral-description" id="toc-spectral-description" class="nav-link" data-scroll-target="#spectral-description"><span class="header-section-number">4.2</span> Spectral description</a></li>
  <li><a href="#spatial-distribution" id="toc-spatial-distribution" class="nav-link" data-scroll-target="#spatial-distribution"><span class="header-section-number">4.3</span> Spatial distribution</a></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion"><span class="header-section-number">5</span> Discussion</a>
  <ul class="collapse">
  <li><a href="#drone-mapping-g.-vermiculophylla-with-machine-learning" id="toc-drone-mapping-g.-vermiculophylla-with-machine-learning" class="nav-link" data-scroll-target="#drone-mapping-g.-vermiculophylla-with-machine-learning"><span class="header-section-number">5.1</span> Drone mapping <em>G. vermiculophylla</em> with machine learning</a></li>
  <li><a href="#g.-vermiculophylla-spatial-distribution-and-mudflat-topography" id="toc-g.-vermiculophylla-spatial-distribution-and-mudflat-topography" class="nav-link" data-scroll-target="#g.-vermiculophylla-spatial-distribution-and-mudflat-topography"><span class="header-section-number">5.2</span> <em>G. vermiculophylla</em> spatial distribution and mudflat topography</a></li>
  <li><a href="#spatio-temporal-monitoring-of-invasive-macroalgae" id="toc-spatio-temporal-monitoring-of-invasive-macroalgae" class="nav-link" data-scroll-target="#spatio-temporal-monitoring-of-invasive-macroalgae"><span class="header-section-number">5.3</span> Spatio-temporal monitoring of invasive macroalgae</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">6</span> Conclusion</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><span class="header-section-number">7</span> References</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      

       <p><a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></p>
<div class="cell-container"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rnaturalearth) </span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rnaturalearthdata) </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Utilities.Package)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flextable)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: le package 'flextable' a été compilé avec la version R 4.3.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attachement du package : 'flextable'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Les objets suivants sont masqués depuis 'package:terra':

    align, colorize, rotate, width</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>L'objet suivant est masqué depuis 'package:purrr':

    compose</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>my_comma<span class="ot">&lt;-</span>scales<span class="sc">::</span><span class="fu">label_comma</span>(<span class="at">accuracy =</span> <span class="cn">NULL</span>, <span class="at">big.mark =</span> <span class="st">","</span>,<span class="at">decimal.mark =</span> <span class="st">"."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fs)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the size threshold in bytes (100 MB)</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>size_threshold <span class="ot">&lt;-</span> <span class="dv">100</span> <span class="sc">*</span> <span class="dv">1024</span> <span class="sc">*</span> <span class="dv">1024</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Get a list of all files in the repository</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">dir_ls</span>(<span class="at">recurse =</span> <span class="cn">TRUE</span>, <span class="at">type =</span> <span class="st">"file"</span>)</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter files larger than the size threshold</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>large_files <span class="ot">&lt;-</span> files[<span class="fu">file_info</span>(files)<span class="sc">$</span>size <span class="sc">&gt;</span> size_threshold]</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if there are large files</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(large_files) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read the existing .gitignore file if it exists</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  gitignore_path <span class="ot">&lt;-</span> <span class="st">".gitignore"</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  gitignore_content <span class="ot">&lt;-</span> <span class="cf">if</span> (<span class="fu">file_exists</span>(gitignore_path)) {</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">readLines</span>(gitignore_path)</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">character</span>(<span class="dv">0</span>)</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify files not already in .gitignore</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>  files_to_add <span class="ot">&lt;-</span> large_files[<span class="sc">!</span>large_files <span class="sc">%in%</span> gitignore_content]</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Append new large files to .gitignore</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(files_to_add) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">writeLines</span>(<span class="fu">c</span>(gitignore_content, files_to_add), gitignore_path)</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="fu">length</span>(files_to_add), <span class="st">" file(s) added to .gitignore."</span>)</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"No new files to add to .gitignore."</span>)</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"No files larger than 100 MB found."</span>)</span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="title-proposition" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Title proposition</h1>
<ul>
<li><p>Remote Sensing of <em>Gracilaria vermiculophylla</em> in the site of its First European Observation</p></li>
<li><p>Monitoring the marine invasive alien species <em>Gracilaria vermiculophylla</em> using unmanned aerial vehicles</p></li>
<li><p>Remote Sensing of <em>Gracilaria vermiculophylla</em>: Mapping its Distribution at the Site of its First European Description</p></li>
<li><p>Mapping the Spread of the Invasive Species <em>Gracilaria vermiculophylla</em> using Remote Sensing at the Site of its Initial Description in Europe</p></li>
<li><p>Quantifying <em>Gracilaria vermiculophylla</em> Spatial and Temporal Distribution Through Remote Sensing at its First Recorded European Site</p></li>
</ul>
</section>
<section id="introduction" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Introduction</h1>
<p>The introduction of Non-Indigenous Species (NIS) in terrestrial, freshwater, and marine ecosystems is one of the major threats to biodiversity worldwide. In particular, the proliferation and rapid spread of Invasive Alien Species (IAS) can radically change the structure and functioning of marine ecosystems, , requiring effective inventorying and monitoring programs <span class="citation" data-cites="d15020161">(<a href="#ref-d15020161" role="doc-biblioref">Massé et al., 2023</a>)</span>. In Europe, 874 NIS have been introduced to the marine environment so far (i.e.&nbsp;until 2020) and it is expected that the rate of biological invasions will continue to increase in the coming years <span class="citation" data-cites="d14121077">(<a href="#ref-d14121077" role="doc-biblioref">Zenetos et al., 2022</a>)</span>. Macroalgae represent more than 40 % of the NIS introduced to Europe waters, with many species native to the Temperate Northern Pacific <span class="citation" data-cites="williams2007global">(<a href="#ref-williams2007global" role="doc-biblioref">Williams and Smith, 2007</a>)</span>. Amongst all invasive macroalgae, <em>Gracilaria vermiculophylla</em> <span class="citation" data-cites="WoRMS303450">(<a href="#ref-WoRMS303450" role="doc-biblioref">Papenfuss, 1967</a>)</span> (original name <em>Gracilariopsis vermiculophylla</em> <span class="citation" data-cites="ohmi1956contributions">(<a href="#ref-ohmi1956contributions" role="doc-biblioref">OHMI, 1956</a>)</span>; also known as <em>Agarophyton vermiculophyllum</em> <span class="citation" data-cites="gurgel2018systematics">(<a href="#ref-gurgel2018systematics" role="doc-biblioref">Gurgel et al., 2018</a>)</span>), has spread extensively from its native distribution range in Japan and Korea <span class="citation" data-cites="terada2002review">(<a href="#ref-terada2002review" role="doc-biblioref">Terada and Yamamoto, 2002</a>)</span> across temperate estuaries in North America, Europe, and other regions, facilitated by aquaculture and maritime activities <span class="citation" data-cites="rueness2005life weinberger2008invasive krueger2017genetic">(<a href="#ref-krueger2017genetic" role="doc-biblioref">Krueger-Hadfield et al., 2017</a>; <a href="#ref-rueness2005life" role="doc-biblioref">Rueness, 2005</a>; <a href="#ref-weinberger2008invasive" role="doc-biblioref">Weinberger et al., 2008</a>)</span>. While <em>G. vermiculophylla</em> can provide some ecosystem services, such as habitat for invertebrates and juvenile fish <span class="citation" data-cites="davoult2017multiple">(<a href="#ref-davoult2017multiple" role="doc-biblioref">Davoult et al., 2017</a>)</span>, it often outcompetes native vegetation, alters sediment composition <span class="citation" data-cites="nyberg2009flora">(<a href="#ref-nyberg2009flora" role="doc-biblioref">Nyberg et al., 2009</a>)</span>, and disrupts trophic interactions <span class="citation" data-cites="van2018global">(<a href="#ref-van2018global" role="doc-biblioref">Ginneken et al., 2018</a>)</span>. In regions like the Baltic Sea and the eastern United States, it has been documented to negatively affect native fucoids and seagrasses <span class="citation" data-cites="van2003reintroduction thomsen2013effects firth2024invasive">(<a href="#ref-firth2024invasive" role="doc-biblioref">Firth et al., 2024</a>; <a href="#ref-thomsen2013effects" role="doc-biblioref">Thomsen et al., 2013</a>; <a href="#ref-van2003reintroduction" role="doc-biblioref">Van Katwijk, 2003</a>)</span>. These impacts underscore the importance of monitoring and managing the spread of <em>G. vermiculophylla</em>, particularly as climate change and anthropogenic pressures continue to facilitate biological invasions. <em>G. vermiculophylla</em> success as an invader stems from its tolerance to a wide range of environmental conditions, including temperature <span class="citation" data-cites="sotka2018combining">(<a href="#ref-sotka2018combining" role="doc-biblioref">Sotka et al., 2018</a>)</span>, nutrient variability <span class="citation" data-cites="abreu2011nitrogen">(<a href="#ref-abreu2011nitrogen" role="doc-biblioref">Abreu et al., 2011</a>)</span> and salinity <span class="citation" data-cites="weinberger2008invasive">(<a href="#ref-weinberger2008invasive" role="doc-biblioref">Weinberger et al., 2008</a>)</span>. Its growth capacity at low salinities <span class="citation" data-cites="rueness2005life nyberg2007introduced">(<a href="#ref-nyberg2007introduced" role="doc-biblioref">Nyberg, 2007</a>; <a href="#ref-rueness2005life" role="doc-biblioref">Rueness, 2005</a>)</span> explains its presence in the brackish waters of the Baltic Sea <span class="citation" data-cites="weinberger2008invasive">(<a href="#ref-weinberger2008invasive" role="doc-biblioref">Weinberger et al., 2008</a>)</span> but also in the mesohaline sheltered part of estuaries of the Atlantic coast of Europe <strong>(Surget et al., 2017)</strong>. It is also present in confined areas of lagoons characterized by low hydrodynamism <span class="citation" data-cites="abreu2011nitrogen sfriso2012spreading">(<a href="#ref-abreu2011nitrogen" role="doc-biblioref">Abreu et al., 2011</a>; <a href="#ref-sfriso2012spreading" role="doc-biblioref">Sfriso et al., 2012</a>)</span>. In Europe, it was first observed in 1996 in the Belon estuary (France) and later in many other estuaries on the Brittany coast of France <span class="citation" data-cites="rueness2005life">(<a href="#ref-rueness2005life" role="doc-biblioref">Rueness, 2005</a>)</span>. It can be found on hard substrates such as invertebrate’s tubes and shells providing a substratum <span class="citation" data-cites="thomsen2007gracilaria">(<a href="#ref-thomsen2007gracilaria" role="doc-biblioref">Thomsen et al., 2007</a>)</span> or attached to pebbles and rocks <span class="citation" data-cites="terada2002review">(<a href="#ref-terada2002review" role="doc-biblioref">Terada and Yamamoto, 2002</a>)</span> but the largest populations are colonizing soft-bottom sediment and particularly estuarine intertidal mudflats <strong>(Surget et al., 2017)</strong>. In this habitat, extensive dark red mats are observed at low tide, covering vast areas that have largely been unquantified in most studies. Therefore, <em>G. vermiculophylla</em> is capable of establishing populations in soft-bottom sediment habitats that were previously devoid of macroalgae <span class="citation" data-cites="ramus2017invasive">(<a href="#ref-ramus2017invasive" role="doc-biblioref">Ramus et al., 2017</a>)</span>. These mats are usually monospecific with the alga thalli partially buried into the mud <span class="citation" data-cites="rueness2005life surget2017processus">(<a href="#ref-rueness2005life" role="doc-biblioref">Rueness, 2005</a>; <a href="#ref-surget2017processus" role="doc-biblioref">Surget, 2017</a>)</span>. Intertidal mats can however be temporarily overgrown by ephemeral green macroalgae <span class="citation" data-cites="weinberger2008invasive">(<a href="#ref-weinberger2008invasive" role="doc-biblioref">Weinberger et al., 2008</a>)</span>. In the estuaries where <em>G. vermiculophylla</em> was first documented, large monospecific mats were reported to be confined to the upper intertidal zones <span class="citation" data-cites="rueness2005life">(<a href="#ref-rueness2005life" role="doc-biblioref">Rueness, 2005</a>)</span>; however, their spatial distribution relative to the mudflat topography had not been quantitatively assessed. In fact, <em>G. vermiculophylla</em> has never been mapped using remote sensing techniques, and existing descriptions of its distribution lack spatially explicit mapping <span class="citation" data-cites="thomsen2007gracilaria weinberger2008invasive abreu2011nitrogen sfriso2012spreading">(<a href="#ref-abreu2011nitrogen" role="doc-biblioref">Abreu et al., 2011</a>; <a href="#ref-sfriso2012spreading" role="doc-biblioref">Sfriso et al., 2012</a>; <a href="#ref-thomsen2007gracilaria" role="doc-biblioref">Thomsen et al., 2007</a>; <a href="#ref-weinberger2008invasive" role="doc-biblioref">Weinberger et al., 2008</a>)</span></p>
<p>Remote sensing has revolutionized our ability to monitor and manage coastal ecosystems, offering efficient and scalable methods for detecting environmental changes in intertidal vegetation across a wide range of spatio-temporal scales <span class="citation" data-cites="davies2024intertidal calleja2017long davies2024sentinel zoffoli2021decadal valle2015mapping">(<a href="#ref-calleja2017long" role="doc-biblioref">Calleja et al., 2017</a>; <a href="#ref-davies2024intertidal" role="doc-biblioref">Davies et al., 2024a</a>, <a href="#ref-davies2024sentinel" role="doc-biblioref">2024b</a>; <a href="#ref-valle2015mapping" role="doc-biblioref">Valle et al., 2015</a>; <a href="#ref-zoffoli2021decadal" role="doc-biblioref">Zoffoli et al., 2021</a>)</span>. Among remote-sensing technologies, drone-based imagery has recently emerged as a particularly promising tool for studying the spatial distribution of intertidal primary producers such as benthic microalgae <span class="citation" data-cites="roman2021using roman2024mapping">(<a href="#ref-roman2024mapping" role="doc-biblioref">Román et al., 2024</a>, <a href="#ref-roman2021using" role="doc-biblioref">2021</a>)</span>, seagrass <span class="citation" data-cites="chand2021low duffy2018spatial roman2021using">(<a href="#ref-chand2021low" role="doc-biblioref">Chand and Bollard, 2021</a>; <a href="#ref-duffy2018spatial" role="doc-biblioref">Duffy et al., 2018</a>; <a href="#ref-roman2021using" role="doc-biblioref">Román et al., 2021</a>)</span> and macroalgae <span class="citation" data-cites="rs14133124 peidro2024quantifying">(<a href="#ref-rs14133124" role="doc-biblioref">Diruit et al., 2022</a>; <a href="#ref-peidro2024quantifying" role="doc-biblioref">Peidro-Devesa et al., 2024</a>)</span>. While it lacks the temporal consistency of satellite missions, drone remote sensing makes it possible to acquire at extremely high spatial resolution (i.e.&nbsp;cm-scale), rapidly target specific areas of interest, and to provide observations in overcast conditions. In particular, the potential of drone remote sensing for monitoring the surface area occupied by IAS has been demonstrated <span class="citation" data-cites="roca2022monitoring">(<a href="#ref-roca2022monitoring" role="doc-biblioref">Roca et al., 2022</a>)</span>. Drone-based photogrammetry also makes it possible to characterize the distribution of intertidal vegetation together with mudflat geomorphology, thus improving our understanding of primary producers patterning <span class="citation" data-cites="brunier2022evolution douglas2024linking">(<a href="#ref-brunier2022evolution" role="doc-biblioref">Brunier et al., 2022</a>; <a href="#ref-douglas2024linking" role="doc-biblioref">Douglas et al., 2024</a>)</span>.</p>
<p>In this study, a drone-based multispectral remote sensing approach was applied to map <em>G. vermiculophylla</em> spatial distribution at a very-high spatial resolution in three intertidal estuaries of European Atlantic coast. We adapted the neural network classification model DISCOV (Drone Intertidal Substrate Classification Of Vegetation, <span class="citation" data-cites="rs16234383">Oiry et al. (<a href="#ref-rs16234383" role="doc-biblioref">2024b</a>)</span>, <span class="citation" data-cites="oiry_2024_14218984">Oiry et al. (<a href="#ref-oiry_2024_14218984" role="doc-biblioref">2024a</a>)</span>) by specifically training the model with a new class corresponding to <em>G. vermiculophylla</em>. A validation dataset was obtained from <em>in situ</em> data to estimate the classification accuracy. LIDAR data were concurrently acquired to accurately map the intertidal elevation. We used a Generalized Additive Model (GAM) to examine the relationship between the seaweed spatial distribution and spatial metrics quantifying the mudflat topography. We expected the presence of G. vermiculophylla in mudflats to be associated to a specific height range as well as being more closely related to flat areas of the intertidal zone.</p>
</section>
<section id="materiel-methods" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Materiel &amp; Methods</h1>
<div class="cell-container"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>building <span class="ot">&lt;-</span> T   </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>layout<span class="ot">&lt;-</span><span class="st">"Data/Workflow_Figure"</span> <span class="sc">%&gt;%</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list.files</span>(<span class="st">"V2.txt"</span>,<span class="at">full.names =</span> T, <span class="at">recursive =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_delim</span>(<span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">trim_ws =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Evaluate <span class="sc">==</span> T)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>layout<span class="sc">$</span>Text<span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alaligne"</span>,<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, layout<span class="sc">$</span>Text)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>boxes<span class="ot">&lt;-</span> layout <span class="sc">%&gt;%</span> </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Type <span class="sc">==</span> <span class="st">"Box"</span>,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>                shape <span class="sc">==</span> <span class="st">"square"</span>)</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>text<span class="ot">&lt;-</span> layout <span class="sc">%&gt;%</span> </span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Type <span class="sc">==</span> <span class="st">"Box"</span>)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>diamonds<span class="ot">&lt;-</span> layout <span class="sc">%&gt;%</span> </span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Type <span class="sc">==</span> <span class="st">"Box"</span>,</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>                shape <span class="sc">==</span> <span class="st">"diamond"</span>)</span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>diff <span class="ot">=</span> <span class="fl">0.1</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(diamonds)) {</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>  diam_a<span class="ot">&lt;-</span>diamonds[i,]</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>df<span class="ot">&lt;-</span><span class="fu">data.frame</span>(  </span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="fu">c</span>(diam_a<span class="sc">$</span>xmin <span class="sc">-</span> diff, diam_a<span class="sc">$</span>xmin <span class="sc">+</span> diff, diam_a<span class="sc">$</span>xmax <span class="sc">+</span> diff, diam_a<span class="sc">$</span>xmax <span class="sc">-</span> diff),</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="fu">c</span>(diam_a<span class="sc">$</span>ymin, diam_a<span class="sc">$</span>ymax, diam_a<span class="sc">$</span>ymax, diam_a<span class="sc">$</span>ymin),</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">Type =</span> <span class="fu">rep</span>(diam_a<span class="sc">$</span>Type,<span class="dv">4</span>),</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">Text =</span> <span class="fu">rep</span>(diam_a<span class="sc">$</span>Text,<span class="dv">4</span>),</span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">Color =</span> <span class="fu">rep</span>(diam_a<span class="sc">$</span>color,<span class="dv">4</span>),</span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">Evaluate =</span> <span class="fu">rep</span>(diam_a<span class="sc">$</span>Evaluate,<span class="dv">4</span>),</span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">Resolution =</span>  <span class="fu">rep</span>(diam_a<span class="sc">$</span>Resolution,<span class="dv">4</span>),</span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">Evaluate_bigbox =</span> <span class="fu">rep</span>(diam_a<span class="sc">$</span>Evaluate_bigbox,<span class="dv">4</span>),</span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fu">rep</span>(diam_a<span class="sc">$</span>alpha,<span class="dv">4</span>),</span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">shape =</span>  <span class="fu">rep</span>(diam_a<span class="sc">$</span>shape,<span class="dv">4</span>)</span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(i <span class="sc">==</span> <span class="dv">1</span>){</span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>    diamond_list<span class="ot">&lt;-</span><span class="fu">list</span>(df)</span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a>  }<span class="cf">else</span>{</span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a>    diamond_list[[i]]<span class="ot">&lt;-</span>df</span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a>Arrows<span class="ot">&lt;-</span> layout <span class="sc">%&gt;%</span> </span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Type <span class="sc">==</span> <span class="st">"Arrow"</span>,</span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>                Evaluate <span class="sc">==</span> T)</span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a>Group_box<span class="ot">&lt;-</span>layout <span class="sc">%&gt;%</span> </span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Type <span class="sc">==</span> <span class="st">"Box"</span>,</span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a>                Evaluate_bigbox <span class="sc">==</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Group) <span class="sc">%&gt;%</span> </span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">xmin =</span> <span class="fu">min</span>(xmin)<span class="sc">-</span><span class="fl">0.1</span>,</span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a>                   <span class="at">xmax =</span> <span class="fu">max</span>(xmax)<span class="sc">+</span><span class="fl">0.1</span>,</span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ymin =</span> <span class="fu">min</span>(ymin)<span class="sc">-</span><span class="fl">0.1</span>,</span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ymax =</span> <span class="fu">max</span>(ymax)<span class="sc">+</span><span class="fl">0.1</span>,</span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a>                   <span class="at">alpha =</span> <span class="fl">0.1</span>,</span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a>                   <span class="at">color =</span> color)</span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a>resolution_box<span class="ot">&lt;-</span>layout <span class="sc">%&gt;%</span> </span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Type <span class="sc">==</span> <span class="st">"Box"</span>,</span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a>                Resolution <span class="sc">!=</span> <span class="st">"NA"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Resolution) <span class="sc">%&gt;%</span> </span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">xmin =</span> <span class="fu">min</span>(xmin)<span class="sc">-</span><span class="fl">0.1</span>,</span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a>                   <span class="at">xmax =</span> <span class="fu">max</span>(xmax)<span class="sc">+</span><span class="fl">0.1</span>,</span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ymin =</span> <span class="fu">min</span>(ymin)<span class="sc">-</span><span class="fl">0.1</span>,</span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ymax =</span> <span class="fu">max</span>(ymax)<span class="sc">+</span><span class="fl">0.1</span>,</span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a>                   <span class="at">alpha =</span> <span class="fl">0.1</span>,</span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a>                   <span class="at">color =</span> color)</span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a>legend <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">color =</span> <span class="fu">unique</span>(layout<span class="sc">$</span>color)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-75"><a href="#cb8-75" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(color)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-76"><a href="#cb8-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">case_when</span>(color <span class="sc">==</span> <span class="st">"#EA6155"</span> <span class="sc">~</span> <span class="st">"Input Data"</span>,</span>
<span id="cb8-77"><a href="#cb8-77" aria-hidden="true" tabindex="-1"></a>                           color <span class="sc">==</span> <span class="st">"#DC9614"</span> <span class="sc">~</span> <span class="st">"Pre-processing"</span>,</span>
<span id="cb8-78"><a href="#cb8-78" aria-hidden="true" tabindex="-1"></a>                           color <span class="sc">==</span> <span class="st">"#52A4DA"</span> <span class="sc">~</span> <span class="st">"Model Building"</span>,</span>
<span id="cb8-79"><a href="#cb8-79" aria-hidden="true" tabindex="-1"></a>                           color <span class="sc">==</span> <span class="st">"#79DA52"</span> <span class="sc">~</span> <span class="st">"Validation"</span>,</span>
<span id="cb8-80"><a href="#cb8-80" aria-hidden="true" tabindex="-1"></a>                           color <span class="sc">==</span> <span class="st">"#DA9652"</span> <span class="sc">~</span> <span class="st">"Output Data"</span>,</span>
<span id="cb8-81"><a href="#cb8-81" aria-hidden="true" tabindex="-1"></a>                           color <span class="sc">==</span> <span class="st">"darkcyan"</span> <span class="sc">~</span> <span class="st">"Final Product"</span>,</span>
<span id="cb8-82"><a href="#cb8-82" aria-hidden="true" tabindex="-1"></a>                           color <span class="sc">==</span> <span class="st">"#DA5273"</span>  <span class="sc">~</span> <span class="st">"Prediction"</span>,</span>
<span id="cb8-83"><a href="#cb8-83" aria-hidden="true" tabindex="-1"></a>                           <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"NA"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-84"><a href="#cb8-84" aria-hidden="true" tabindex="-1"></a>  arrange</span>
<span id="cb8-85"><a href="#cb8-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-86"><a href="#cb8-86" aria-hidden="true" tabindex="-1"></a>segment<span class="ot">&lt;-</span>layout <span class="sc">%&gt;%</span> </span>
<span id="cb8-87"><a href="#cb8-87" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Type <span class="sc">==</span> <span class="st">"Segment"</span>)</span>
<span id="cb8-88"><a href="#cb8-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-89"><a href="#cb8-89" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(building <span class="sc">==</span> T){</span>
<span id="cb8-90"><a href="#cb8-90" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-91"><a href="#cb8-91" aria-hidden="true" tabindex="-1"></a>text_size <span class="ot">=</span> <span class="dv">6</span> <span class="do">#### 6 to plot it fullscreen in R,  1.8 for exporting with ggsave</span></span>
<span id="cb8-92"><a href="#cb8-92" aria-hidden="true" tabindex="-1"></a>line_size <span class="ot">=</span><span class="dv">1</span> <span class="do">####arrow_size 0.3 to export  1 to for R plotting</span></span>
<span id="cb8-93"><a href="#cb8-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-94"><a href="#cb8-94" aria-hidden="true" tabindex="-1"></a>legend_size <span class="ot">=</span> <span class="dv">1</span> <span class="do">#### 0.3 to export, 1 for R ploting</span></span>
<span id="cb8-95"><a href="#cb8-95" aria-hidden="true" tabindex="-1"></a>legend_text <span class="ot">=</span> <span class="dv">10</span> <span class="do">#### 4 to export, 10 for R ploting</span></span>
<span id="cb8-96"><a href="#cb8-96" aria-hidden="true" tabindex="-1"></a>}<span class="cf">else</span>{</span>
<span id="cb8-97"><a href="#cb8-97" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-98"><a href="#cb8-98" aria-hidden="true" tabindex="-1"></a>text_size <span class="ot">=</span> <span class="fl">1.85</span> <span class="do">#### 6 to plot it fullscreen in R,  1.8 for exporting with ggsave</span></span>
<span id="cb8-99"><a href="#cb8-99" aria-hidden="true" tabindex="-1"></a>line_size <span class="ot">=</span><span class="fl">0.3</span> <span class="do">####arrow_size 0.3 to export  1 to for R plotting</span></span>
<span id="cb8-100"><a href="#cb8-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-101"><a href="#cb8-101" aria-hidden="true" tabindex="-1"></a>legend_size <span class="ot">=</span> <span class="fl">0.3</span> <span class="do">#### 0.3 to export, 1 for R ploting</span></span>
<span id="cb8-102"><a href="#cb8-102" aria-hidden="true" tabindex="-1"></a>legend_text <span class="ot">=</span> <span class="dv">5</span> <span class="do">#### 5 to export, 10 for R ploting</span></span>
<span id="cb8-103"><a href="#cb8-103" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-104"><a href="#cb8-104" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-105"><a href="#cb8-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-106"><a href="#cb8-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-107"><a href="#cb8-107" aria-hidden="true" tabindex="-1"></a>a<span class="ot">&lt;-</span><span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb8-108"><a href="#cb8-108" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-109"><a href="#cb8-109" aria-hidden="true" tabindex="-1"></a>  <span class="co"># statebins:::geom_rrect(data = Group_box,</span></span>
<span id="cb8-110"><a href="#cb8-110" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           mapping = aes(xmin = xmin,</span></span>
<span id="cb8-111"><a href="#cb8-111" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                         xmax= xmax,</span></span>
<span id="cb8-112"><a href="#cb8-112" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                         ymin = ymin,</span></span>
<span id="cb8-113"><a href="#cb8-113" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                         ymax = ymax),</span></span>
<span id="cb8-114"><a href="#cb8-114" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           alpha = 0.05,</span></span>
<span id="cb8-115"><a href="#cb8-115" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           size = 0.2,</span></span>
<span id="cb8-116"><a href="#cb8-116" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           show.legend = F,</span></span>
<span id="cb8-117"><a href="#cb8-117" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           fill = Group_box$color,</span></span>
<span id="cb8-118"><a href="#cb8-118" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           color = "black")+</span></span>
<span id="cb8-119"><a href="#cb8-119" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-120"><a href="#cb8-120" aria-hidden="true" tabindex="-1"></a>    <span class="co"># statebins:::geom_rrect(data = resolution_box,</span></span>
<span id="cb8-121"><a href="#cb8-121" aria-hidden="true" tabindex="-1"></a>    <span class="co">#         mapping = aes(xmin = xmin,</span></span>
<span id="cb8-122"><a href="#cb8-122" aria-hidden="true" tabindex="-1"></a>    <span class="co">#                       xmax= xmax,</span></span>
<span id="cb8-123"><a href="#cb8-123" aria-hidden="true" tabindex="-1"></a>    <span class="co">#                       ymin = ymin,</span></span>
<span id="cb8-124"><a href="#cb8-124" aria-hidden="true" tabindex="-1"></a>    <span class="co">#                       ymax = ymax,</span></span>
<span id="cb8-125"><a href="#cb8-125" aria-hidden="true" tabindex="-1"></a>    <span class="co">#                       alpha = rep(c("Input Data", "Pre-processing", "Model Building", "Prediction", "Validation", "Output Data", "Final Product"),2)),</span></span>
<span id="cb8-126"><a href="#cb8-126" aria-hidden="true" tabindex="-1"></a>    <span class="co">#         fill = "NA",</span></span>
<span id="cb8-127"><a href="#cb8-127" aria-hidden="true" tabindex="-1"></a>    <span class="co">#         linetype = 2,</span></span>
<span id="cb8-128"><a href="#cb8-128" aria-hidden="true" tabindex="-1"></a>    <span class="co">#         size = 0.4,</span></span>
<span id="cb8-129"><a href="#cb8-129" aria-hidden="true" tabindex="-1"></a>    <span class="co">#         color = "black")+</span></span>
<span id="cb8-130"><a href="#cb8-130" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-131"><a href="#cb8-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_alpha_manual</span>(<span class="at">name =</span> <span class="st">""</span>, </span>
<span id="cb8-132"><a href="#cb8-132" aria-hidden="true" tabindex="-1"></a>                     <span class="at">values =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>),</span>
<span id="cb8-133"><a href="#cb8-133" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Input Data"</span>, <span class="st">"Pre-Processing"</span>, <span class="st">"Model Building"</span>, <span class="st">"Prediction"</span>, <span class="st">"Validation"</span>, <span class="st">"Output Data"</span>, <span class="st">"Final Product"</span>),</span>
<span id="cb8-134"><a href="#cb8-134" aria-hidden="true" tabindex="-1"></a>                     <span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"#f94144"</span>, <span class="st">"#f3722c"</span>, <span class="st">"#f8961e"</span>, <span class="st">"#f9c74f"</span>, <span class="st">"#90be6d"</span>, <span class="st">"#43aa8b"</span>, <span class="st">"#577590"</span>),</span>
<span id="cb8-135"><a href="#cb8-135" aria-hidden="true" tabindex="-1"></a>                                                                        <span class="at">shape =</span> <span class="fu">c</span>(<span class="dv">22</span>),</span>
<span id="cb8-136"><a href="#cb8-136" aria-hidden="true" tabindex="-1"></a>                                                                        <span class="at">linetype =</span> <span class="fu">c</span>(<span class="dv">1</span>),</span>
<span id="cb8-137"><a href="#cb8-137" aria-hidden="true" tabindex="-1"></a>                                                                        <span class="at">alpha =</span> <span class="fu">c</span>(<span class="fl">0.5</span>)</span>
<span id="cb8-138"><a href="#cb8-138" aria-hidden="true" tabindex="-1"></a>                                                                        )</span>
<span id="cb8-139"><a href="#cb8-139" aria-hidden="true" tabindex="-1"></a>                                          )</span>
<span id="cb8-140"><a href="#cb8-140" aria-hidden="true" tabindex="-1"></a>                     )<span class="sc">+</span></span>
<span id="cb8-141"><a href="#cb8-141" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-142"><a href="#cb8-142" aria-hidden="true" tabindex="-1"></a>  statebins<span class="sc">:::</span><span class="fu">geom_rrect</span>(<span class="at">data =</span> boxes,</span>
<span id="cb8-143"><a href="#cb8-143" aria-hidden="true" tabindex="-1"></a>            <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">xmin =</span> xmin,</span>
<span id="cb8-144"><a href="#cb8-144" aria-hidden="true" tabindex="-1"></a>                          <span class="at">xmax=</span> xmax,</span>
<span id="cb8-145"><a href="#cb8-145" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ymin =</span> ymin,</span>
<span id="cb8-146"><a href="#cb8-146" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ymax =</span> ymax</span>
<span id="cb8-147"><a href="#cb8-147" aria-hidden="true" tabindex="-1"></a>                          ),</span>
<span id="cb8-148"><a href="#cb8-148" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="fl">0.2</span>,</span>
<span id="cb8-149"><a href="#cb8-149" aria-hidden="true" tabindex="-1"></a>            <span class="at">alpha =</span> <span class="fu">as.numeric</span>(boxes<span class="sc">$</span>alpha),</span>
<span id="cb8-150"><a href="#cb8-150" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> boxes<span class="sc">$</span>color,</span>
<span id="cb8-151"><a href="#cb8-151" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">"black"</span>)</span>
<span id="cb8-152"><a href="#cb8-152" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-153"><a href="#cb8-153" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(diamond_list)){</span>
<span id="cb8-154"><a href="#cb8-154" aria-hidden="true" tabindex="-1"></a>    poly_i<span class="ot">&lt;-</span>diamond_list[[i]]</span>
<span id="cb8-155"><a href="#cb8-155" aria-hidden="true" tabindex="-1"></a>    a<span class="ot">&lt;-</span>a<span class="sc">+</span></span>
<span id="cb8-156"><a href="#cb8-156" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_polygon</span>(<span class="at">data =</span> poly_i, </span>
<span id="cb8-157"><a href="#cb8-157" aria-hidden="true" tabindex="-1"></a>                   <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> x, </span>
<span id="cb8-158"><a href="#cb8-158" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">y =</span> y</span>
<span id="cb8-159"><a href="#cb8-159" aria-hidden="true" tabindex="-1"></a>                                 ),</span>
<span id="cb8-160"><a href="#cb8-160" aria-hidden="true" tabindex="-1"></a>                  <span class="at">size =</span> <span class="fl">0.2</span>,</span>
<span id="cb8-161"><a href="#cb8-161" aria-hidden="true" tabindex="-1"></a>                   <span class="at">alpha =</span> poly_i<span class="sc">$</span>alpha,</span>
<span id="cb8-162"><a href="#cb8-162" aria-hidden="true" tabindex="-1"></a>                   <span class="at">fill =</span> poly_i<span class="sc">$</span>Color,</span>
<span id="cb8-163"><a href="#cb8-163" aria-hidden="true" tabindex="-1"></a>                   <span class="at">color =</span> <span class="st">"black"</span>)</span>
<span id="cb8-164"><a href="#cb8-164" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-165"><a href="#cb8-165" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb8-166"><a href="#cb8-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-167"><a href="#cb8-167" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(building <span class="sc">==</span> T){</span>
<span id="cb8-168"><a href="#cb8-168" aria-hidden="true" tabindex="-1"></a>    a<span class="ot">&lt;-</span>a<span class="sc">+</span></span>
<span id="cb8-169"><a href="#cb8-169" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> text,</span>
<span id="cb8-170"><a href="#cb8-170" aria-hidden="true" tabindex="-1"></a>            <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> (xmax<span class="sc">+</span>xmin)<span class="sc">/</span><span class="dv">2</span>, <span class="at">y =</span> (ymax <span class="sc">+</span> ymin)<span class="sc">/</span><span class="dv">2</span>, <span class="at">label =</span> (ID)), <span class="at">size =</span> text_size)</span>
<span id="cb8-171"><a href="#cb8-171" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-172"><a href="#cb8-172" aria-hidden="true" tabindex="-1"></a>}<span class="cf">else</span>{</span>
<span id="cb8-173"><a href="#cb8-173" aria-hidden="true" tabindex="-1"></a>    a<span class="ot">&lt;-</span>a<span class="sc">+</span></span>
<span id="cb8-174"><a href="#cb8-174" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> text,</span>
<span id="cb8-175"><a href="#cb8-175" aria-hidden="true" tabindex="-1"></a>            <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> (xmax<span class="sc">+</span>xmin)<span class="sc">/</span><span class="dv">2</span>, <span class="at">y =</span> (ymax <span class="sc">+</span> ymin)<span class="sc">/</span><span class="dv">2</span>, <span class="at">label =</span> (Text)), <span class="at">size =</span> text_size)</span>
<span id="cb8-176"><a href="#cb8-176" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-177"><a href="#cb8-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-178"><a href="#cb8-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-179"><a href="#cb8-179" aria-hidden="true" tabindex="-1"></a>  a<span class="ot">&lt;-</span>a<span class="sc">+</span></span>
<span id="cb8-180"><a href="#cb8-180" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_segment</span>(<span class="at">data =</span> Arrows, <span class="fu">aes</span>( <span class="at">x =</span> xmin, <span class="at">xend =</span> xmax, <span class="at">y =</span> ymin, <span class="at">yend =</span> ymax),</span>
<span id="cb8-181"><a href="#cb8-181" aria-hidden="true" tabindex="-1"></a>    <span class="at">size=</span>line_size, <span class="at">linejoin =</span> <span class="st">"mitre"</span>, <span class="at">lineend =</span> <span class="st">"butt"</span>,</span>
<span id="cb8-182"><a href="#cb8-182" aria-hidden="true" tabindex="-1"></a>    <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"mm"</span>), <span class="at">type=</span> <span class="st">"closed"</span>)</span>
<span id="cb8-183"><a href="#cb8-183" aria-hidden="true" tabindex="-1"></a>    )<span class="sc">+</span></span>
<span id="cb8-184"><a href="#cb8-184" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_segment</span>(<span class="at">data =</span> segment, <span class="fu">aes</span>( <span class="at">x =</span> xmin, <span class="at">xend =</span> xmax, <span class="at">y =</span> ymin, <span class="at">yend =</span> ymax),</span>
<span id="cb8-185"><a href="#cb8-185" aria-hidden="true" tabindex="-1"></a>    <span class="at">size=</span>line_size, <span class="at">linejoin =</span> <span class="st">"mitre"</span>, <span class="at">lineend =</span> <span class="st">"square"</span></span>
<span id="cb8-186"><a href="#cb8-186" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb8-187"><a href="#cb8-187" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-188"><a href="#cb8-188" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(building <span class="sc">==</span> F){</span>
<span id="cb8-189"><a href="#cb8-189" aria-hidden="true" tabindex="-1"></a>  a<span class="ot">&lt;-</span> a<span class="sc">+</span> <span class="fu">theme_void</span>()</span>
<span id="cb8-190"><a href="#cb8-190" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb8-191"><a href="#cb8-191" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-192"><a href="#cb8-192" aria-hidden="true" tabindex="-1"></a>  a<span class="ot">&lt;-</span>a<span class="sc">+</span></span>
<span id="cb8-193"><a href="#cb8-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.88</span>,<span class="fl">0.3</span>),</span>
<span id="cb8-194"><a href="#cb8-194" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> legend_text),</span>
<span id="cb8-195"><a href="#cb8-195" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size =</span> <span class="fu">unit</span>(legend_size, <span class="st">'cm'</span>),</span>
<span id="cb8-196"><a href="#cb8-196" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>,<span class="at">color =</span> <span class="cn">NA</span>)</span>
<span id="cb8-197"><a href="#cb8-197" aria-hidden="true" tabindex="-1"></a>        )<span class="sc">+</span></span>
<span id="cb8-198"><a href="#cb8-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> scales<span class="sc">::</span><span class="fu">breaks_width</span>(<span class="fl">0.5</span>))<span class="sc">+</span></span>
<span id="cb8-199"><a href="#cb8-199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> scales<span class="sc">::</span><span class="fu">breaks_width</span>(<span class="fl">0.5</span>))</span>
<span id="cb8-200"><a href="#cb8-200" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-201"><a href="#cb8-201" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(building <span class="sc">==</span> T){</span>
<span id="cb8-202"><a href="#cb8-202" aria-hidden="true" tabindex="-1"></a>a</span>
<span id="cb8-203"><a href="#cb8-203" aria-hidden="true" tabindex="-1"></a>}<span class="cf">else</span>{</span>
<span id="cb8-204"><a href="#cb8-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Workflow.png"</span>,a, <span class="at">units =</span> <span class="st">"px"</span>)</span>
<span id="cb8-205"><a href="#cb8-205" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="study-sites" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="study-sites"><span class="header-section-number">3.1</span> Study sites</h2>
<p>Field campaigns were conducted at three study sites in France and Spain. At each site, two locations were investigated <a href="#fig-location_sites" class="quarto-xref">Figure&nbsp;1</a>. The Aven &amp; Belon Estuary in South Brittany, France (<a href="#fig-location_sites" class="quarto-xref">Figure&nbsp;1</a> A &amp; C), is a dynamic ria-type system hosting diverse habitats, including sandy tidal flats and subtidal zones with coarse, marine-origin sediments <span class="citation" data-cites="Michel2021 Castaing1995">(<a href="#ref-Castaing1995" role="doc-biblioref">Castaing and Guilcher, 1995</a>; <a href="#ref-Michel2021" role="doc-biblioref">Michel et al., 2021</a>)</span>. These habitats support key benthic species such as <em>Scrobicularia plana</em>, <em>Cerastoderma edule</em>, and <em>Tellina tenuis</em>, which play essential roles in sediment bioturbation and nutrient cycling <span class="citation" data-cites="Tankoua2011 Blanchet2014">(<a href="#ref-Blanchet2014" role="doc-biblioref">Blanchet et al., 2014</a>; <a href="#ref-Tankoua2011" role="doc-biblioref">Tankoua et al., 2011</a>)</span>. The estuary serves as a nursery for juvenile fish and a feeding ground for migratory birds, with its ecological productivity driven by a mix of euryhaline and marine species adapted to salinity gradients <span class="citation" data-cites="Blanchet2014">(<a href="#ref-Blanchet2014" role="doc-biblioref">Blanchet et al., 2014</a>)</span>. Oyster farming, particularly <em>Crassostrea gigas</em>, is a dominant activity, altering sediment dynamics and local biodiversity <span class="citation" data-cites="Michel2021">(<a href="#ref-Michel2021" role="doc-biblioref">Michel et al., 2021</a>)</span>. Despite its ecological richness, the estuary faces pressures from nutrient loading and physical alterations, with bioindicators like <em>S. plana</em> used to monitor the impacts of salinity, sediment quality, and pollution <span class="citation" data-cites="Tankoua2011">(<a href="#ref-Tankoua2011" role="doc-biblioref">Tankoua et al., 2011</a>)</span>.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="st">"The Ria d'Étel, located in Brittany, France, is a macrotidal estuary characterized by its unique hydrodynamics and biodiversity (@fig-location_sites B). Influenced predominantly by tidal regimes, the estuary exhibits high-energy zones with strong currents reaching up to 2.5 m/s, shaping both sediment deposition and ecological habitats [@Portas2023]. The estuary supports diverse benthic communities, with sedimentary organic matter originating from both terrestrial inputs and marine sources, contributing to nutrient cycling and benthic fluxes [@Jeanneau2023]. Vegetation gradients transition from halophytic plants in saline zones to freshwater species upstream, reflecting the estuary's salinity dynamics and ecological complexity [@Cianfaglione2018]. This estuary is also notable for its shellfish farming, with species like *Crassostrea gigas* cultivated extensively. The presence of filter-feeding organisms such as sponges (*Hymeniacidon perlevis*) enhances water quality by mitigating bacterial loads and promoting bioremediation [@Gentric2024]. However, the estuary faces environmental pressures, including nutrient enrichment from agricultural runoff and anthropogenic impacts on sedimentary processes.</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>The Saja-Besaya Estuary, situated along the Cantabrian Sea in northern Spain, is characterized by the confluence of the Saja and Besaya rivers near Torrelavega (<a href="#fig-location_sites" class="quarto-xref">Figure&nbsp;1</a> C). The estuary, also known as San Martín de la Arena or Suances Estuary, has been subject to significant anthropogenic pressures, including industrial developments throughout the 20th century. These activities have led to contamination from mining, paper manufacturing, and carbonate discharges, classifying the estuary as highly polluted near its upper reaches <span class="citation" data-cites="ortega2005fluxes">(<a href="#ref-ortega2005fluxes" role="doc-biblioref">Ortega et al., 2005</a>)</span>. This contamination impacts the estuarine ecosystem, including water quality and biodiversity, with minimal aquatic life and sparse riverbank vegetation in its lower sections <span class="citation" data-cites="romero2008sintering">(<a href="#ref-romero2008sintering" role="doc-biblioref">Romero et al., 2008</a>)</span>.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">######### Aven Belon </span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>mask_aven_belon <span class="ot">&lt;-</span> <span class="fu">read_sf</span>(<span class="st">"Data/shp/mask/mask_site_Aven_Belon_32630.shp"</span>)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>HT <span class="ot">&lt;-</span>  <span class="fu">list.files</span>(<span class="st">"Data/Sentinel2/Aven_Belon/S2A_MSIL2A_20231107T111251_N0509_R137_T30TVT_20231107T144858.SAFE/GRANULE/L2A_T30TVT_A043749_20231107T111246/IMG_DATA/R10m/"</span>, <span class="at">pattern =</span> <span class="st">".jp2"</span>,<span class="at">recursive =</span>T,<span class="at">full.names =</span> T)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>B8_HT <span class="ot">&lt;-</span> <span class="fu">rast</span>(HT[<span class="dv">5</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B08"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_aven_belon, <span class="at">mask =</span> T) </span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>B8_HT <span class="ot">&lt;-</span> B8_HT <span class="sc">-</span> <span class="dv">1000</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>B4_HT <span class="ot">&lt;-</span> <span class="fu">rast</span>(HT[<span class="dv">4</span>])<span class="sc">%&gt;%</span> </span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B04"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_aven_belon, <span class="at">mask =</span> T)</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>B4_HT <span class="ot">&lt;-</span> B4_HT <span class="sc">-</span><span class="dv">1000</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>NDVI_HT <span class="ot">&lt;-</span> (B8_HT<span class="sc">-</span>B4_HT)<span class="sc">/</span>(B8_HT<span class="sc">+</span>B4_HT)</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>LT <span class="ot">&lt;-</span>  <span class="fu">list.files</span>(<span class="st">"Data/Sentinel2/Aven_Belon/S2A_MSIL2A_20240624T110641_N0510_R137_T30TVT_20240624T153247.SAFE/GRANULE/L2A_T30TVT_A047038_20240624T111001/IMG_DATA/R10m/"</span>, <span class="at">pattern =</span> <span class="st">".jp2"</span>,<span class="at">recursive =</span>T,<span class="at">full.names =</span> T)</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>B8_LT <span class="ot">&lt;-</span> <span class="fu">rast</span>(LT[<span class="dv">5</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B08"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_aven_belon, <span class="at">mask =</span> T) </span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>B8_LT <span class="ot">&lt;-</span> B8_LT <span class="sc">-</span> <span class="dv">1000</span></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>B4_LT <span class="ot">&lt;-</span> <span class="fu">rast</span>(LT[<span class="dv">4</span>])<span class="sc">%&gt;%</span> </span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B04"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_aven_belon, <span class="at">mask =</span> T)</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>B4_LT <span class="ot">&lt;-</span> B4_LT <span class="sc">-</span><span class="dv">1000</span></span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a>NDVI_LT <span class="ot">&lt;-</span> (B8_LT<span class="sc">-</span>B4_LT)<span class="sc">/</span>(B8_LT<span class="sc">+</span>B4_LT)</span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a>intertidal_Aven_Belon <span class="ot">&lt;-</span> NDVI_LT <span class="sc">&gt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> NDVI_HT <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.05</span></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(intertidal_Aven_Belon)[<span class="fu">values</span>(intertidal_Aven_Belon) <span class="sc">==</span> F] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a>mask_sf <span class="ot">&lt;-</span> <span class="fu">as.polygons</span>(intertidal_Aven_Belon) <span class="sc">%&gt;%</span> </span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_cast</span>(<span class="st">"POLYGON"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">area =</span> <span class="fu">st_area</span>(geometry)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">as.numeric</span>(area) <span class="sc">&gt;</span> <span class="dv">6500</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a>  nngeo<span class="sc">::</span><span class="fu">st_remove_holes</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a>  smoothr<span class="sc">::</span><span class="fu">smooth</span>(<span class="at">method =</span> <span class="st">"ksmooth"</span>, <span class="at">smoothness =</span> <span class="dv">3</span>)</span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(mask_sf)</span></span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a><span class="fu">write_sf</span>(mask_sf,<span class="st">"Data/shp/Intertidal_sites/mask_intertidal_Aven_Belon.shp"</span>)</span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-52"><a href="#cb10-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-53"><a href="#cb10-53" aria-hidden="true" tabindex="-1"></a>Land_Aven_Belon <span class="ot">&lt;-</span> NDVI_LT <span class="sc">&gt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> NDVI_HT <span class="sc">&gt;</span> <span class="fl">0.05</span></span>
<span id="cb10-54"><a href="#cb10-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-55"><a href="#cb10-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-56"><a href="#cb10-56" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(Land_Aven_Belon)[<span class="fu">values</span>(Land_Aven_Belon) <span class="sc">==</span> F] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb10-57"><a href="#cb10-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-58"><a href="#cb10-58" aria-hidden="true" tabindex="-1"></a>mask_sf <span class="ot">&lt;-</span> <span class="fu">as.polygons</span>(Land_Aven_Belon) <span class="sc">%&gt;%</span> </span>
<span id="cb10-59"><a href="#cb10-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-60"><a href="#cb10-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_cast</span>(<span class="st">"POLYGON"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-61"><a href="#cb10-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">area =</span> <span class="fu">st_area</span>(geometry)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-62"><a href="#cb10-62" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">as.numeric</span>(area) <span class="sc">&gt;</span> <span class="dv">6500</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-63"><a href="#cb10-63" aria-hidden="true" tabindex="-1"></a>  nngeo<span class="sc">::</span><span class="fu">st_remove_holes</span>()</span>
<span id="cb10-64"><a href="#cb10-64" aria-hidden="true" tabindex="-1"></a>  <span class="co"># smoothr::smooth(method = "ksmooth", smoothness = 3)</span></span>
<span id="cb10-65"><a href="#cb10-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-66"><a href="#cb10-66" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(mask_sf)</span></span>
<span id="cb10-67"><a href="#cb10-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-68"><a href="#cb10-68" aria-hidden="true" tabindex="-1"></a><span class="fu">write_sf</span>(mask_sf,<span class="st">"Data/shp/Intertidal_sites/mask_Land_Aven_Belon.shp"</span>)</span>
<span id="cb10-69"><a href="#cb10-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-70"><a href="#cb10-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-71"><a href="#cb10-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-72"><a href="#cb10-72" aria-hidden="true" tabindex="-1"></a><span class="do">######### ETEL</span></span>
<span id="cb10-73"><a href="#cb10-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-74"><a href="#cb10-74" aria-hidden="true" tabindex="-1"></a>mask_Etel <span class="ot">&lt;-</span> <span class="fu">read_sf</span>(<span class="st">"Data/shp/mask/mask_site_Etel_32630.shp"</span>)</span>
<span id="cb10-75"><a href="#cb10-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-76"><a href="#cb10-76" aria-hidden="true" tabindex="-1"></a>HT <span class="ot">&lt;-</span>  <span class="fu">list.files</span>(<span class="st">"Data/Sentinel2/Etel/S2A_MSIL2A_20231008T110941_N0509_R137_T30TVT_20231008T171550.SAFE/GRANULE/L2A_T30TVT_A043320_20231008T111641/IMG_DATA/R10m/"</span>, <span class="at">pattern =</span> <span class="st">".jp2"</span>,<span class="at">recursive =</span>T,<span class="at">full.names =</span> T)</span>
<span id="cb10-77"><a href="#cb10-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-78"><a href="#cb10-78" aria-hidden="true" tabindex="-1"></a>B8_HT <span class="ot">&lt;-</span> <span class="fu">rast</span>(HT[<span class="dv">5</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb10-79"><a href="#cb10-79" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B08"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-80"><a href="#cb10-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_Etel, <span class="at">mask =</span> T) </span>
<span id="cb10-81"><a href="#cb10-81" aria-hidden="true" tabindex="-1"></a>B8_HT <span class="ot">&lt;-</span> B8_HT <span class="sc">-</span> <span class="dv">1000</span></span>
<span id="cb10-82"><a href="#cb10-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-83"><a href="#cb10-83" aria-hidden="true" tabindex="-1"></a>B4_HT <span class="ot">&lt;-</span> <span class="fu">rast</span>(HT[<span class="dv">4</span>])<span class="sc">%&gt;%</span> </span>
<span id="cb10-84"><a href="#cb10-84" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B04"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-85"><a href="#cb10-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_Etel, <span class="at">mask =</span> T)</span>
<span id="cb10-86"><a href="#cb10-86" aria-hidden="true" tabindex="-1"></a>B4_HT <span class="ot">&lt;-</span> B4_HT <span class="sc">-</span><span class="dv">1000</span></span>
<span id="cb10-87"><a href="#cb10-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-88"><a href="#cb10-88" aria-hidden="true" tabindex="-1"></a>NDVI_HT <span class="ot">&lt;-</span> (B8_HT<span class="sc">-</span>B4_HT)<span class="sc">/</span>(B8_HT<span class="sc">+</span>B4_HT)</span>
<span id="cb10-89"><a href="#cb10-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-90"><a href="#cb10-90" aria-hidden="true" tabindex="-1"></a>LT <span class="ot">&lt;-</span>  <span class="fu">list.files</span>(<span class="st">"Data/Sentinel2/Etel/S2B_MSIL2A_20230718T112119_N0509_R037_T30TVT_20230718T124824.SAFE/GRANULE/L2A_T30TVT_A033239_20230718T112118/IMG_DATA/R10m/"</span>, <span class="at">pattern =</span> <span class="st">".jp2"</span>,<span class="at">recursive =</span>T,<span class="at">full.names =</span> T)</span>
<span id="cb10-91"><a href="#cb10-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-92"><a href="#cb10-92" aria-hidden="true" tabindex="-1"></a>B8_LT <span class="ot">&lt;-</span> <span class="fu">rast</span>(LT[<span class="dv">5</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb10-93"><a href="#cb10-93" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B08"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-94"><a href="#cb10-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_Etel, <span class="at">mask =</span> T) </span>
<span id="cb10-95"><a href="#cb10-95" aria-hidden="true" tabindex="-1"></a>B8_LT <span class="ot">&lt;-</span> B8_LT <span class="sc">-</span> <span class="dv">1000</span></span>
<span id="cb10-96"><a href="#cb10-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-97"><a href="#cb10-97" aria-hidden="true" tabindex="-1"></a>B4_LT <span class="ot">&lt;-</span> <span class="fu">rast</span>(LT[<span class="dv">4</span>])<span class="sc">%&gt;%</span> </span>
<span id="cb10-98"><a href="#cb10-98" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B04"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-99"><a href="#cb10-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_Etel, <span class="at">mask =</span> T)</span>
<span id="cb10-100"><a href="#cb10-100" aria-hidden="true" tabindex="-1"></a>B4_LT <span class="ot">&lt;-</span> B4_LT <span class="sc">-</span><span class="dv">1000</span></span>
<span id="cb10-101"><a href="#cb10-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-102"><a href="#cb10-102" aria-hidden="true" tabindex="-1"></a>NDVI_LT <span class="ot">&lt;-</span> (B8_LT<span class="sc">-</span>B4_LT)<span class="sc">/</span>(B8_LT<span class="sc">+</span>B4_LT)</span>
<span id="cb10-103"><a href="#cb10-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-104"><a href="#cb10-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-105"><a href="#cb10-105" aria-hidden="true" tabindex="-1"></a>intertidal_Etel <span class="ot">&lt;-</span> NDVI_LT <span class="sc">&gt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> NDVI_HT <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.05</span></span>
<span id="cb10-106"><a href="#cb10-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-107"><a href="#cb10-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-108"><a href="#cb10-108" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(intertidal_Etel)[<span class="fu">values</span>(intertidal_Etel) <span class="sc">==</span> F] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb10-109"><a href="#cb10-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-110"><a href="#cb10-110" aria-hidden="true" tabindex="-1"></a>mask_sf <span class="ot">&lt;-</span> <span class="fu">as.polygons</span>(intertidal_Etel) <span class="sc">%&gt;%</span> </span>
<span id="cb10-111"><a href="#cb10-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-112"><a href="#cb10-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_cast</span>(<span class="st">"POLYGON"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-113"><a href="#cb10-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">area =</span> <span class="fu">st_area</span>(geometry)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-114"><a href="#cb10-114" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">as.numeric</span>(area) <span class="sc">&gt;</span> <span class="dv">6500</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-115"><a href="#cb10-115" aria-hidden="true" tabindex="-1"></a>  nngeo<span class="sc">::</span><span class="fu">st_remove_holes</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-116"><a href="#cb10-116" aria-hidden="true" tabindex="-1"></a>  smoothr<span class="sc">::</span><span class="fu">smooth</span>(<span class="at">method =</span> <span class="st">"ksmooth"</span>, <span class="at">smoothness =</span> <span class="dv">3</span>)</span>
<span id="cb10-117"><a href="#cb10-117" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-118"><a href="#cb10-118" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(mask_sf)</span></span>
<span id="cb10-119"><a href="#cb10-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-120"><a href="#cb10-120" aria-hidden="true" tabindex="-1"></a><span class="fu">write_sf</span>(mask_sf,<span class="st">"Data/shp/Intertidal_sites/mask_intertidal_Etel.shp"</span>)</span>
<span id="cb10-121"><a href="#cb10-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-122"><a href="#cb10-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-123"><a href="#cb10-123" aria-hidden="true" tabindex="-1"></a>Land_Etel <span class="ot">&lt;-</span> NDVI_LT <span class="sc">&gt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> NDVI_HT <span class="sc">&gt;</span> <span class="fl">0.05</span></span>
<span id="cb10-124"><a href="#cb10-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-125"><a href="#cb10-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-126"><a href="#cb10-126" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(Land_Etel)[<span class="fu">values</span>(Land_Etel) <span class="sc">==</span> F] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb10-127"><a href="#cb10-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-128"><a href="#cb10-128" aria-hidden="true" tabindex="-1"></a>mask_sf <span class="ot">&lt;-</span> <span class="fu">as.polygons</span>(Land_Etel) <span class="sc">%&gt;%</span> </span>
<span id="cb10-129"><a href="#cb10-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-130"><a href="#cb10-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_cast</span>(<span class="st">"POLYGON"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-131"><a href="#cb10-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">area =</span> <span class="fu">st_area</span>(geometry)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-132"><a href="#cb10-132" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">as.numeric</span>(area) <span class="sc">&gt;</span> <span class="dv">6500</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-133"><a href="#cb10-133" aria-hidden="true" tabindex="-1"></a>  nngeo<span class="sc">::</span><span class="fu">st_remove_holes</span>() </span>
<span id="cb10-134"><a href="#cb10-134" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-135"><a href="#cb10-135" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mask_sf)</span>
<span id="cb10-136"><a href="#cb10-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-137"><a href="#cb10-137" aria-hidden="true" tabindex="-1"></a><span class="fu">write_sf</span>(mask_sf,<span class="st">"Data/shp/Intertidal_sites/mask_Land_Etel.shp"</span>)</span>
<span id="cb10-138"><a href="#cb10-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-139"><a href="#cb10-139" aria-hidden="true" tabindex="-1"></a><span class="do">######### SAJA</span></span>
<span id="cb10-140"><a href="#cb10-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-141"><a href="#cb10-141" aria-hidden="true" tabindex="-1"></a>mask_SAJA <span class="ot">&lt;-</span> <span class="fu">read_sf</span>(<span class="st">"Data/shp/mask/mask_site_Saja_32630.shp"</span>)</span>
<span id="cb10-142"><a href="#cb10-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-143"><a href="#cb10-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-144"><a href="#cb10-144" aria-hidden="true" tabindex="-1"></a>HT <span class="ot">&lt;-</span>  <span class="fu">list.files</span>(<span class="st">"Data/Sentinel2/Saja/S2A_MSIL2A_20181004T110911_N0500_R137_T30TVP_20230620T014212.SAFE/GRANULE/L2A_T30TVP_A017151_20181004T111207/IMG_DATA/R10m/"</span>, <span class="at">pattern =</span> <span class="st">".jp2"</span>,<span class="at">recursive =</span>T,<span class="at">full.names =</span> T)</span>
<span id="cb10-145"><a href="#cb10-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-146"><a href="#cb10-146" aria-hidden="true" tabindex="-1"></a>B8_HT <span class="ot">&lt;-</span> <span class="fu">rast</span>(HT[<span class="dv">5</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb10-147"><a href="#cb10-147" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B08"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-148"><a href="#cb10-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_SAJA, <span class="at">mask =</span> T) </span>
<span id="cb10-149"><a href="#cb10-149" aria-hidden="true" tabindex="-1"></a>B8_HT <span class="ot">&lt;-</span> B8_HT <span class="sc">-</span> <span class="dv">1000</span></span>
<span id="cb10-150"><a href="#cb10-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-151"><a href="#cb10-151" aria-hidden="true" tabindex="-1"></a>B4_HT <span class="ot">&lt;-</span> <span class="fu">rast</span>(HT[<span class="dv">4</span>])<span class="sc">%&gt;%</span> </span>
<span id="cb10-152"><a href="#cb10-152" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B04"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-153"><a href="#cb10-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_SAJA, <span class="at">mask =</span> T)</span>
<span id="cb10-154"><a href="#cb10-154" aria-hidden="true" tabindex="-1"></a>B4_HT <span class="ot">&lt;-</span> B4_HT <span class="sc">-</span><span class="dv">1000</span></span>
<span id="cb10-155"><a href="#cb10-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-156"><a href="#cb10-156" aria-hidden="true" tabindex="-1"></a>NDVI_HT <span class="ot">&lt;-</span> (B8_HT<span class="sc">-</span>B4_HT)<span class="sc">/</span>(B8_HT<span class="sc">+</span>B4_HT)</span>
<span id="cb10-157"><a href="#cb10-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-158"><a href="#cb10-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-159"><a href="#cb10-159" aria-hidden="true" tabindex="-1"></a>LT <span class="ot">&lt;-</span>  <span class="fu">list.files</span>(<span class="st">"Data/Sentinel2/Saja/S2A_MSIL2A_20230928T110831_N0509_R137_T30TVP_20230928T171657.SAFE/GRANULE/L2A_T30TVP_A043177_20230928T111532/IMG_DATA/R10m/"</span>, <span class="at">pattern =</span> <span class="st">".jp2"</span>,<span class="at">recursive =</span>T,<span class="at">full.names =</span> T)</span>
<span id="cb10-160"><a href="#cb10-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-161"><a href="#cb10-161" aria-hidden="true" tabindex="-1"></a>B8_LT <span class="ot">&lt;-</span> <span class="fu">rast</span>(LT[<span class="dv">5</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb10-162"><a href="#cb10-162" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B08"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-163"><a href="#cb10-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_SAJA, <span class="at">mask =</span> T) </span>
<span id="cb10-164"><a href="#cb10-164" aria-hidden="true" tabindex="-1"></a>B8_LT <span class="ot">&lt;-</span> B8_LT <span class="sc">-</span> <span class="dv">1000</span></span>
<span id="cb10-165"><a href="#cb10-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-166"><a href="#cb10-166" aria-hidden="true" tabindex="-1"></a>B4_LT <span class="ot">&lt;-</span> <span class="fu">rast</span>(LT[<span class="dv">4</span>])<span class="sc">%&gt;%</span> </span>
<span id="cb10-167"><a href="#cb10-167" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="st">"B04"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-168"><a href="#cb10-168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(mask_SAJA, <span class="at">mask =</span> T)</span>
<span id="cb10-169"><a href="#cb10-169" aria-hidden="true" tabindex="-1"></a>B4_LT <span class="ot">&lt;-</span> B4_LT <span class="sc">-</span><span class="dv">1000</span></span>
<span id="cb10-170"><a href="#cb10-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-171"><a href="#cb10-171" aria-hidden="true" tabindex="-1"></a>NDVI_LT <span class="ot">&lt;-</span> (B8_LT<span class="sc">-</span>B4_LT)<span class="sc">/</span>(B8_LT<span class="sc">+</span>B4_LT)</span>
<span id="cb10-172"><a href="#cb10-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-173"><a href="#cb10-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-174"><a href="#cb10-174" aria-hidden="true" tabindex="-1"></a>intertidal_Saja <span class="ot">&lt;-</span> NDVI_LT <span class="sc">&gt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> NDVI_HT <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.05</span></span>
<span id="cb10-175"><a href="#cb10-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-176"><a href="#cb10-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-177"><a href="#cb10-177" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(intertidal_Saja)[<span class="fu">values</span>(intertidal_Saja) <span class="sc">==</span> F] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb10-178"><a href="#cb10-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-179"><a href="#cb10-179" aria-hidden="true" tabindex="-1"></a>mask_sf <span class="ot">&lt;-</span> <span class="fu">as.polygons</span>(intertidal_Saja) <span class="sc">%&gt;%</span> </span>
<span id="cb10-180"><a href="#cb10-180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-181"><a href="#cb10-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_cast</span>(<span class="st">"POLYGON"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-182"><a href="#cb10-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">area =</span> <span class="fu">st_area</span>(geometry)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-183"><a href="#cb10-183" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">as.numeric</span>(area) <span class="sc">&gt;</span> <span class="dv">6500</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-184"><a href="#cb10-184" aria-hidden="true" tabindex="-1"></a>  nngeo<span class="sc">::</span><span class="fu">st_remove_holes</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-185"><a href="#cb10-185" aria-hidden="true" tabindex="-1"></a>  smoothr<span class="sc">::</span><span class="fu">smooth</span>(<span class="at">method =</span> <span class="st">"ksmooth"</span>, <span class="at">smoothness =</span> <span class="dv">3</span>)</span>
<span id="cb10-186"><a href="#cb10-186" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-187"><a href="#cb10-187" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(mask_sf)</span></span>
<span id="cb10-188"><a href="#cb10-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-189"><a href="#cb10-189" aria-hidden="true" tabindex="-1"></a><span class="fu">write_sf</span>(mask_sf,<span class="st">"Data/shp/Intertidal_sites/mask_intertidal_Saja.shp"</span>)</span>
<span id="cb10-190"><a href="#cb10-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-191"><a href="#cb10-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-192"><a href="#cb10-192" aria-hidden="true" tabindex="-1"></a>Land_Saja <span class="ot">&lt;-</span> NDVI_LT <span class="sc">&gt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> NDVI_HT <span class="sc">&gt;</span> <span class="fl">0.05</span></span>
<span id="cb10-193"><a href="#cb10-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-194"><a href="#cb10-194" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(Land_Saja)[<span class="fu">values</span>(Land_Saja) <span class="sc">==</span> F] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb10-195"><a href="#cb10-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-196"><a href="#cb10-196" aria-hidden="true" tabindex="-1"></a>mask_sf <span class="ot">&lt;-</span> <span class="fu">as.polygons</span>(Land_Saja) <span class="sc">%&gt;%</span> </span>
<span id="cb10-197"><a href="#cb10-197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-198"><a href="#cb10-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_cast</span>(<span class="st">"POLYGON"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-199"><a href="#cb10-199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">area =</span> <span class="fu">st_area</span>(geometry)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-200"><a href="#cb10-200" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">as.numeric</span>(area) <span class="sc">&gt;</span> <span class="dv">6500</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-201"><a href="#cb10-201" aria-hidden="true" tabindex="-1"></a>  nngeo<span class="sc">::</span><span class="fu">st_remove_holes</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-202"><a href="#cb10-202" aria-hidden="true" tabindex="-1"></a>  smoothr<span class="sc">::</span><span class="fu">smooth</span>(<span class="at">method =</span> <span class="st">"ksmooth"</span>, <span class="at">smoothness =</span> <span class="dv">3</span>)</span>
<span id="cb10-203"><a href="#cb10-203" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-204"><a href="#cb10-204" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mask_sf)</span>
<span id="cb10-205"><a href="#cb10-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-206"><a href="#cb10-206" aria-hidden="true" tabindex="-1"></a><span class="fu">write_sf</span>(mask_sf,<span class="st">"Data/shp/Intertidal_sites/mask_Land_Saja.shp"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [7]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>sovereignty10 <span class="ot">&lt;-</span> <span class="fu">ne_countries</span>(<span class="at">scale =</span> <span class="dv">10</span>, <span class="at">returnclass =</span> <span class="st">"sf"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>world_map <span class="ot">&lt;-</span> sovereignty10 <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(sovereignt<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">"Spain"</span>,<span class="st">"France"</span>,<span class="st">"Portugal"</span>,</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"Italy"</span>,<span class="st">"Andorra"</span>,</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"United Kingdom"</span>,</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"Switzerland"</span>,<span class="st">"Belgium"</span>,</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"Germany"</span>,<span class="st">"Luxembourg"</span>) ) </span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>bbox_europe <span class="ot">&lt;-</span> <span class="fu">st_bbox</span>(<span class="fu">c</span>(<span class="at">xmin =</span> <span class="sc">-</span><span class="dv">20</span>, <span class="at">ymin =</span> <span class="dv">34</span>,</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>                         <span class="at">xmax =</span> <span class="dv">20</span>, <span class="at">ymax =</span> <span class="dv">55</span>) ,</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">crs =</span> <span class="fu">st_crs</span>(world_map) ) </span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>world_map<span class="ot">&lt;-</span><span class="fu">st_make_valid</span>(world_map) </span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>european_union_map_cropped <span class="ot">&lt;-</span> <span class="fu">st_crop</span>(world_map, bbox_europe)  <span class="sc">%&gt;%</span> </span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_transform</span>(<span class="st">"+proj=laea +lat_0=52 +lon_0=10 +x_0=4321000 +y_0=3210000 +ellps=GRS80 +units=m +no_defs "</span>)  </span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>UnitedKingdom<span class="ot">&lt;-</span>sovereignty10 <span class="sc">%&gt;%</span> </span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(sovereignt<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">"United Kingdom"</span>) ) <span class="sc">%&gt;%</span> </span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_cast</span>(<span class="st">"POLYGON"</span>) </span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>bbox_UK <span class="ot">&lt;-</span> <span class="fu">st_bbox</span>(<span class="fu">c</span>(<span class="at">xmin =</span> <span class="sc">-</span><span class="dv">20</span>, <span class="at">ymin =</span> <span class="dv">45</span>,</span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>                         <span class="at">xmax =</span> <span class="dv">20</span>, <span class="at">ymax =</span> <span class="dv">55</span>) ,</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>                       <span class="at">crs =</span> <span class="fu">st_crs</span>(UnitedKingdom) ) </span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>UnitedKingdom<span class="ot">&lt;-</span><span class="fu">st_make_valid</span>(UnitedKingdom) </span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>UK_map_cropped <span class="ot">&lt;-</span> <span class="fu">st_crop</span>(UnitedKingdom, bbox_UK)  <span class="sc">%&gt;%</span> </span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_transform</span>(<span class="st">"+proj=laea +lat_0=52 +lon_0=10 +x_0=4321000 +y_0=3210000 +ellps=GRS80 +units=m +no_defs "</span>) </span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>Europe_sf<span class="ot">&lt;-</span>european_union_map_cropped <span class="sc">%&gt;%</span> </span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(UK_map_cropped) </span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a>Projects<span class="ot">&lt;-</span><span class="fu">data.frame</span>(</span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>  <span class="at">Name=</span><span class="fu">c</span>(</span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>  <span class="st">"France - Aven"</span>,</span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>  <span class="st">"France - Belon"</span>,</span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Spain - Saja Estuary"</span></span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>  ) ,</span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a>  <span class="at">Long=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">3.187818</span>,</span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span><span class="fl">3.187818</span>, </span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span><span class="fl">4.025402</span>) ,</span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>  <span class="at">Lat=</span><span class="fu">c</span>(<span class="fl">47.697566</span>, </span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a>        <span class="fl">47.697566</span>,</span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>        <span class="fl">43.408356</span>) </span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a>  )  <span class="sc">%&gt;%</span> </span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>(<span class="at">coords=</span><span class="fu">c</span>(<span class="st">"Long"</span>,<span class="st">"Lat"</span>) )  <span class="sc">%&gt;%</span> </span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_set_crs</span>(<span class="st">"EPSG:4326"</span>)  <span class="sc">%&gt;%</span> </span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_transform</span>(<span class="st">"+proj=laea +lat_0=52 +lon_0=10 +x_0=4321000 +y_0=3210000 +ellps=GRS80 +units=m +no_defs "</span>)  </span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a>Projects_df<span class="ot">&lt;-</span>Projects <span class="sc">%&gt;%</span></span>
<span id="cb11-57"><a href="#cb11-57" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">lon =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(.) [,<span class="dv">1</span>],</span>
<span id="cb11-58"><a href="#cb11-58" aria-hidden="true" tabindex="-1"></a>                <span class="at">lat =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(.) [,<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb11-59"><a href="#cb11-59" aria-hidden="true" tabindex="-1"></a>  sf<span class="sc">::</span><span class="fu">st_set_geometry</span>(<span class="cn">NULL</span>)</span>
<span id="cb11-60"><a href="#cb11-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-61"><a href="#cb11-61" aria-hidden="true" tabindex="-1"></a>Miniworld_map <span class="ot">&lt;-</span> sovereignty10 <span class="sc">%&gt;%</span> </span>
<span id="cb11-62"><a href="#cb11-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>()</span>
<span id="cb11-63"><a href="#cb11-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-64"><a href="#cb11-64" aria-hidden="true" tabindex="-1"></a><span class="fu">sf_use_s2</span>(<span class="cn">FALSE</span>)</span>
<span id="cb11-65"><a href="#cb11-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-66"><a href="#cb11-66" aria-hidden="true" tabindex="-1"></a>bbox_EU <span class="ot">&lt;-</span> <span class="fu">st_bbox</span>(<span class="fu">c</span>(<span class="at">xmin =</span> <span class="sc">-</span><span class="dv">30</span>, <span class="at">ymin =</span> <span class="dv">20</span>,</span>
<span id="cb11-67"><a href="#cb11-67" aria-hidden="true" tabindex="-1"></a>                         <span class="at">xmax =</span> <span class="dv">50</span>, <span class="at">ymax =</span> <span class="dv">70</span>) ,</span>
<span id="cb11-68"><a href="#cb11-68" aria-hidden="true" tabindex="-1"></a>                       <span class="at">crs =</span> <span class="fu">st_crs</span>(Miniworld_map) ) </span>
<span id="cb11-69"><a href="#cb11-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-70"><a href="#cb11-70" aria-hidden="true" tabindex="-1"></a>  MiniEU_map<span class="ot">&lt;-</span><span class="fu">st_crop</span>(Miniworld_map, bbox_EU)  <span class="sc">%&gt;%</span> </span>
<span id="cb11-71"><a href="#cb11-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_transform</span>(<span class="st">"+proj=laea +lat_0=52 +lon_0=10 +x_0=4321000 +y_0=3210000 +ellps=GRS80 +units=m +no_defs "</span>)  </span>
<span id="cb11-72"><a href="#cb11-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-73"><a href="#cb11-73" aria-hidden="true" tabindex="-1"></a>  df_P1<span class="ot">&lt;-</span>Projects_df <span class="sc">%&gt;%</span> </span>
<span id="cb11-74"><a href="#cb11-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">ID =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(.))) <span class="sc">%&gt;%</span> </span>
<span id="cb11-75"><a href="#cb11-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Site =</span> <span class="fu">case_when</span>(ID <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"Aven Estuary"</span>,</span>
<span id="cb11-76"><a href="#cb11-76" aria-hidden="true" tabindex="-1"></a>                            ID <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">"Bélon Estuary"</span>,</span>
<span id="cb11-77"><a href="#cb11-77" aria-hidden="true" tabindex="-1"></a>                            ID <span class="sc">==</span> <span class="dv">3</span> <span class="sc">~</span> <span class="st">"Saja Estuary"</span>,</span>
<span id="cb11-78"><a href="#cb11-78" aria-hidden="true" tabindex="-1"></a>                            <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"NA"</span>),</span>
<span id="cb11-79"><a href="#cb11-79" aria-hidden="true" tabindex="-1"></a>           <span class="at">ID =</span> <span class="fu">case_when</span>(ID <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"A"</span>,</span>
<span id="cb11-80"><a href="#cb11-80" aria-hidden="true" tabindex="-1"></a>                          ID <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">"B"</span>,</span>
<span id="cb11-81"><a href="#cb11-81" aria-hidden="true" tabindex="-1"></a>                          <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"C"</span>),</span>
<span id="cb11-82"><a href="#cb11-82" aria-hidden="true" tabindex="-1"></a>           <span class="at">lon_label =</span> <span class="dv">2900000</span>,</span>
<span id="cb11-83"><a href="#cb11-83" aria-hidden="true" tabindex="-1"></a>           <span class="at">lat_label =</span> <span class="fu">case_when</span>(Site <span class="sc">==</span> <span class="st">"Aven Estuary"</span> <span class="sc">~</span> lat  <span class="sc">-</span> <span class="dv">200000</span>,</span>
<span id="cb11-84"><a href="#cb11-84" aria-hidden="true" tabindex="-1"></a>                                 Site <span class="sc">==</span> <span class="st">"Bélon Estuary"</span> <span class="sc">~</span> lat <span class="sc">+</span> <span class="dv">50000</span>,</span>
<span id="cb11-85"><a href="#cb11-85" aria-hidden="true" tabindex="-1"></a>                                 Site <span class="sc">==</span> <span class="st">"Saja Estuary"</span> <span class="sc">~</span> lat  <span class="sc">-</span> <span class="dv">250000</span>,),</span>
<span id="cb11-86"><a href="#cb11-86" aria-hidden="true" tabindex="-1"></a>           <span class="at">lat_ID =</span> lat_label <span class="sc">+</span> <span class="dv">100000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-87"><a href="#cb11-87" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(lon,lat,lon_label,lat_label,lat_ID,ID,Site))</span>
<span id="cb11-88"><a href="#cb11-88" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-89"><a href="#cb11-89" aria-hidden="true" tabindex="-1"></a>    scaleFUN <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">paste0</span>(<span class="fu">sprintf</span>(<span class="st">"%.2f"</span>, x),<span class="st">"°N"</span>)</span>
<span id="cb11-90"><a href="#cb11-90" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb11-91"><a href="#cb11-91" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(MiniEU_map) <span class="sc">+</span></span>
<span id="cb11-92"><a href="#cb11-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">linewidth =</span> <span class="fl">0.5</span>, <span class="at">alpha =</span> <span class="fl">0.93</span>, <span class="at">fill =</span> <span class="st">"#CFCFCF"</span>, <span class="at">colour =</span> <span class="st">"grey30"</span>) <span class="sc">+</span></span>
<span id="cb11-93"><a href="#cb11-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">data =</span> df_P1, <span class="fu">aes</span>(<span class="at">x =</span> lon_label, <span class="at">xend =</span> lon, <span class="at">yend =</span> lat, <span class="at">y =</span>lat_label, <span class="at">group =</span> ID), <span class="at">color =</span> <span class="st">"goldenrod4"</span>,<span class="at">size =</span> <span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb11-94"><a href="#cb11-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> df_P1, <span class="fu">aes</span>(<span class="at">x =</span> lon, <span class="at">y =</span>lat), <span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb11-95"><a href="#cb11-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label</span>(<span class="at">data =</span> df_P1, <span class="fu">aes</span>(<span class="at">x =</span> lon_label, <span class="at">y =</span>lat_label, <span class="at">label =</span> Site), <span class="at">fill =</span> <span class="st">"#FFD27D"</span>,<span class="at">size =</span> <span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb11-96"><a href="#cb11-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label</span>(<span class="at">data =</span> df_P1, <span class="fu">aes</span>(<span class="at">x =</span> lon_label, <span class="at">y =</span> lat_ID, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">"B"</span>,<span class="st">"A"</span>,<span class="st">"C"</span>)), <span class="at">size =</span> <span class="dv">6</span>)<span class="sc">+</span></span>
<span id="cb11-97"><a href="#cb11-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(</span>
<span id="cb11-98"><a href="#cb11-98" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">2600000</span>, <span class="dv">4100000</span>),</span>
<span id="cb11-99"><a href="#cb11-99" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">1600000</span>, <span class="dv">3100000</span>), </span>
<span id="cb11-100"><a href="#cb11-100" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="cn">FALSE</span></span>
<span id="cb11-101"><a href="#cb11-101" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb11-102"><a href="#cb11-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede_Map</span>() <span class="sc">+</span></span>
<span id="cb11-103"><a href="#cb11-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Longitude"</span>, <span class="at">y =</span> <span class="st">"Latitude"</span>) <span class="sc">+</span></span>
<span id="cb11-104"><a href="#cb11-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb11-105"><a href="#cb11-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="dv">42</span>, <span class="at">labels =</span> scaleFUN) <span class="sc">+</span></span>
<span id="cb11-106"><a href="#cb11-106" aria-hidden="true" tabindex="-1"></a>  <span class="co"># theme_void() +</span></span>
<span id="cb11-107"><a href="#cb11-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb11-108"><a href="#cb11-108" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">"cm"</span>),</span>
<span id="cb11-109"><a href="#cb11-109" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-110"><a href="#cb11-110" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-111"><a href="#cb11-111" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>, <span class="at">vjust =</span> <span class="fl">7.1</span>, <span class="at">hjust =</span> <span class="fl">0.8</span>),</span>
<span id="cb11-112"><a href="#cb11-112" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">5</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb11-113"><a href="#cb11-113" aria-hidden="true" tabindex="-1"></a>    <span class="co"># panel.border = element_rect(color = "black", fill = NA),</span></span>
<span id="cb11-114"><a href="#cb11-114" aria-hidden="true" tabindex="-1"></a>    <span class="co"># panel.background = element_rect(fill = "white"),</span></span>
<span id="cb11-115"><a href="#cb11-115" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plot.background = element_rect(fill = "white", color = NA),</span></span>
<span id="cb11-116"><a href="#cb11-116" aria-hidden="true" tabindex="-1"></a>    <span class="co"># legend.background = element_rect(fill = "transparent"),</span></span>
<span id="cb11-117"><a href="#cb11-117" aria-hidden="true" tabindex="-1"></a>    <span class="co"># panel.grid.major = element_line(color = "grey", linetype = "dashed", linewidth = 0.5),</span></span>
<span id="cb11-118"><a href="#cb11-118" aria-hidden="true" tabindex="-1"></a>    <span class="co"># panel.grid.minor = element_line(color = "grey", linetype = "dotted", linewidth = 0.25)</span></span>
<span id="cb11-119"><a href="#cb11-119" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb11-120"><a href="#cb11-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-121"><a href="#cb11-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-122"><a href="#cb11-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-123"><a href="#cb11-123" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Figure1/Map_Drone_Sites.png"</span>,p1,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">400</span>)  </span>
<span id="cb11-124"><a href="#cb11-124" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Low_res/Figure1/Map_Drone_Sites.png"</span>,p1,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">200</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [8]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>create_square_sf <span class="ot">&lt;-</span> <span class="cf">function</span>(side_length, center_lonlat) {</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Validate input</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(center_lonlat) <span class="sc">!=</span> <span class="dv">2</span>) {</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Please provide the center coordinates as a vector: c(longitude, latitude)."</span>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.numeric</span>(side_length) <span class="sc">||</span> side_length <span class="sc">&lt;=</span> <span class="dv">0</span>) {</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Please provide a positive numeric value for the side length."</span>)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define the center point in lon/lat</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  center_point <span class="ot">&lt;-</span> <span class="fu">st_sfc</span>(<span class="fu">st_point</span>(center_lonlat), <span class="at">crs =</span> <span class="dv">4326</span>)</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Transform to EPSG:32630</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  center_point_utm <span class="ot">&lt;-</span> <span class="fu">st_transform</span>(center_point, <span class="dv">32630</span>)</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  center_coords <span class="ot">&lt;-</span> <span class="fu">st_coordinates</span>(center_point_utm)</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate half side length in meters</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>  half_side <span class="ot">&lt;-</span> side_length <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the square corners in UTM</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>  square_coords <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>    center_coords[<span class="dv">1</span>] <span class="sc">-</span> half_side, center_coords[<span class="dv">2</span>] <span class="sc">-</span> half_side,</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>    center_coords[<span class="dv">1</span>] <span class="sc">+</span> half_side, center_coords[<span class="dv">2</span>] <span class="sc">-</span> half_side,</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>    center_coords[<span class="dv">1</span>] <span class="sc">+</span> half_side, center_coords[<span class="dv">2</span>] <span class="sc">+</span> half_side,</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>    center_coords[<span class="dv">1</span>] <span class="sc">-</span> half_side, center_coords[<span class="dv">2</span>] <span class="sc">+</span> half_side,</span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>    center_coords[<span class="dv">1</span>] <span class="sc">-</span> half_side, center_coords[<span class="dv">2</span>] <span class="sc">-</span> half_side <span class="co"># Close the square</span></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>  ), <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a polygon from the coordinates</span></span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>  square_polygon <span class="ot">&lt;-</span> <span class="fu">st_polygon</span>(<span class="fu">list</span>(square_coords))</span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create an sf object</span></span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a>  square_sf <span class="ot">&lt;-</span> <span class="fu">st_sfc</span>(square_polygon, <span class="at">crs =</span> <span class="dv">32630</span>)</span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(square_sf)</span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>It_Saja <span class="ot">&lt;-</span> <span class="st">"Data/shp/Intertidal_sites/mask_intertidal_Saja.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>Land_Saja <span class="ot">&lt;-</span> <span class="st">"Data/shp/Intertidal_sites/mask_Land_Saja.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>ext_plot_Saja <span class="ot">&lt;-</span> <span class="fu">st_bbox</span>(<span class="fu">create_square_sf</span>(<span class="dv">5000</span>, <span class="fu">c</span>(<span class="sc">-</span><span class="fl">4.025987</span>, <span class="fl">43.4200065</span>)))</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>Flight_Saja<span class="ot">&lt;-</span><span class="fu">data.frame</span>(</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">Name=</span><span class="fu">c</span>(</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Marisma de </span><span class="sc">\n</span><span class="st">Cortiguera"</span>,</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Marisma de </span><span class="sc">\n</span><span class="st">Cudón"</span>),</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">Long=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">4.029591</span>,</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span><span class="fl">4.031800</span>) ,</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">Lat=</span><span class="fu">c</span>(<span class="fl">43.409738</span>,</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>        <span class="fl">43.414434</span>)</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>  )  <span class="sc">%&gt;%</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>(<span class="at">coords=</span><span class="fu">c</span>(<span class="st">"Long"</span>,<span class="st">"Lat"</span>) )  <span class="sc">%&gt;%</span></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_set_crs</span>(<span class="st">"EPSG:4326"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_transform</span>(<span class="fu">crs</span>(Land_Saja))  <span class="sc">%&gt;%</span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">lon =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(.) [,<span class="dv">1</span>],</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>                <span class="at">lat =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(.) [,<span class="dv">2</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>  sf<span class="sc">::</span><span class="fu">st_set_geometry</span>(<span class="cn">NULL</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">lon_label =</span> <span class="fu">case_when</span>(Name <span class="sc">==</span> <span class="st">"Marisma de </span><span class="sc">\n</span><span class="st">Cudón"</span> <span class="sc">~</span> <span class="fl">418639.3</span>,</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>                               Name <span class="sc">==</span> <span class="st">"Marisma de </span><span class="sc">\n</span><span class="st">Cortiguera"</span> <span class="sc">~</span> lon <span class="sc">+</span> <span class="dv">2000</span>),</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">lat_label =</span> <span class="fu">case_when</span>(Name <span class="sc">==</span> <span class="st">"Marisma de </span><span class="sc">\n</span><span class="st">Cudón"</span> <span class="sc">~</span> lat <span class="sc">+</span> <span class="dv">1000</span>,</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>                               Name <span class="sc">==</span> <span class="st">"Marisma de </span><span class="sc">\n</span><span class="st">Cortiguera"</span> <span class="sc">~</span> lat <span class="sc">+</span> <span class="dv">500</span>))</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>(plot_Saja <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> Land_Saja, <span class="at">linewidth =</span> <span class="fl">0.05</span>, <span class="at">alpha =</span> <span class="fl">0.93</span>, <span class="at">fill =</span> <span class="st">"grey80"</span>) <span class="sc">+</span></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> It_Saja, <span class="at">linewidth =</span> <span class="fl">0.05</span>, <span class="at">alpha =</span> <span class="fl">0.93</span>, <span class="at">fill =</span> <span class="st">"goldenrod"</span>)<span class="sc">+</span></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_label</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(ext_plot_Saja[<span class="dv">1</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_Saja[<span class="dv">3</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_Saja[<span class="dv">1</span>]))<span class="sc">*</span><span class="fl">0.1</span> , </span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> <span class="fu">as.numeric</span>(ext_plot_Saja[<span class="dv">2</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_Saja[<span class="dv">4</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_Saja[<span class="dv">2</span>]))<span class="sc">*</span><span class="fl">0.1</span></span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>                   , <span class="at">label =</span> <span class="st">"C"</span>), </span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>               <span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.5</span> )<span class="sc">+</span></span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(ext_plot_Saja[<span class="dv">1</span>], ext_plot_Saja[<span class="dv">3</span>]),</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylim =</span> <span class="fu">c</span>(ext_plot_Saja[<span class="dv">2</span>], ext_plot_Saja[<span class="dv">4</span>]),</span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> F</span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">data =</span> Flight_Saja, <span class="fu">aes</span>(<span class="at">x =</span> lon_label, <span class="at">xend =</span> lon, <span class="at">yend =</span> lat, <span class="at">y =</span>lat_label, <span class="at">group =</span> Name), <span class="at">color =</span> <span class="st">"goldenrod4"</span>,<span class="at">size =</span> <span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> Flight_Saja, <span class="fu">aes</span>(<span class="at">x =</span> lon, <span class="at">y =</span>lat), <span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label</span>(<span class="at">data =</span> Flight_Saja, <span class="fu">aes</span>(<span class="at">x =</span> lon_label, <span class="at">y =</span>lat_label, <span class="at">label =</span> Name), <span class="at">fill =</span> <span class="st">"#FFD27D"</span>,<span class="at">size =</span> <span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede_Map</span>()<span class="sc">+</span></span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Longitude"</span>,</span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>       <span class="at">y=</span><span class="st">"Latitude"</span>)<span class="sc">+</span></span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="sc">-</span><span class="fl">4.01</span>) <span class="sc">+</span></span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fl">43.415</span>) <span class="sc">+</span></span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Move y-axis to the right</span></span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(<span class="st">"white"</span>, <span class="dv">0</span>)),</span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>),</span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="fl">4.5</span>,  <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">4</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>()</span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a>    )  <span class="co"># Adjust position if needed  )</span></span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Figure1/Map_Saja.png"</span>,plot_Saja,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">400</span>)  </span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Low_res/Figure1/Map_Saja.png"</span>,plot_Saja,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">200</span>)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [10]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>It_AB <span class="ot">&lt;-</span> <span class="st">"Data/shp/Intertidal_sites/mask_intertidal_Aven_Belon.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>Land_AB <span class="ot">&lt;-</span> <span class="st">"Data/shp/Intertidal_sites/mask_Land_Aven_Belon.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>ext_plot_AB <span class="ot">&lt;-</span> <span class="fu">st_bbox</span>(<span class="fu">create_square_sf</span>(<span class="dv">8000</span>, <span class="fu">c</span>(<span class="sc">-</span><span class="fl">3.780146</span>, <span class="fl">47.810000</span>)))</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>Flight_AB<span class="ot">&lt;-</span><span class="fu">data.frame</span>(</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Name=</span><span class="fu">c</span>(</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Notre-Dame De Tremor"</span>),</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">Long=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">3.748634</span>) ,</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">Lat=</span><span class="fu">c</span>(<span class="fl">47.837923</span>)</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  )  <span class="sc">%&gt;%</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>(<span class="at">coords=</span><span class="fu">c</span>(<span class="st">"Long"</span>,<span class="st">"Lat"</span>) )  <span class="sc">%&gt;%</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_set_crs</span>(<span class="st">"EPSG:4326"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_transform</span>(<span class="fu">crs</span>(Land_Saja))  <span class="sc">%&gt;%</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">lon =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(.) [,<span class="dv">1</span>],</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">lat =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(.) [,<span class="dv">2</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>  sf<span class="sc">::</span><span class="fu">st_set_geometry</span>(<span class="cn">NULL</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">lon_label =</span> <span class="fu">case_when</span>(Name <span class="sc">==</span> <span class="st">"Notre-Dame De Tremor"</span> <span class="sc">~</span>  <span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">1</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">4</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">2</span>]))<span class="sc">*</span><span class="fl">0.35</span>),</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">lat_label =</span> <span class="fu">case_when</span>(Name <span class="sc">==</span> <span class="st">"Notre-Dame De Tremor"</span> <span class="sc">~</span>  <span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">2</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">4</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">2</span>]))<span class="sc">*</span><span class="fl">0.85</span>))</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>(Plot_Aven <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> Land_AB, <span class="at">linewidth =</span> <span class="fl">0.05</span>, <span class="at">alpha =</span> <span class="fl">0.93</span>, <span class="at">fill =</span> <span class="st">"grey80"</span>) <span class="sc">+</span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> It_AB, <span class="at">linewidth =</span> <span class="fl">0.05</span>, <span class="at">alpha =</span> <span class="fl">0.93</span>, <span class="at">fill =</span> <span class="st">"goldenrod"</span>) <span class="sc">+</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_label</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">1</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">3</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">1</span>]))<span class="sc">*</span><span class="fl">0.1</span> , </span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> <span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">2</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">4</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_AB[<span class="dv">2</span>]))<span class="sc">*</span><span class="fl">0.1</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>                   , <span class="at">label =</span> <span class="st">"A"</span>), </span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>               <span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.5</span> )<span class="sc">+</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(ext_plot_AB[<span class="dv">1</span>], ext_plot_AB[<span class="dv">3</span>]),</span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylim =</span> <span class="fu">c</span>(ext_plot_AB[<span class="dv">2</span>], ext_plot_AB[<span class="dv">4</span>]),</span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> F</span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">data =</span> Flight_AB, <span class="fu">aes</span>(<span class="at">x =</span> lon_label, <span class="at">xend =</span> lon, <span class="at">yend =</span> lat, <span class="at">y =</span>lat_label, <span class="at">group =</span> Name), <span class="at">color =</span> <span class="st">"goldenrod4"</span>,<span class="at">size =</span> <span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> Flight_AB, <span class="fu">aes</span>(<span class="at">x =</span> lon, <span class="at">y =</span>lat), <span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label</span>(<span class="at">data =</span> Flight_AB, <span class="fu">aes</span>(<span class="at">x =</span> lon_label, <span class="at">y =</span>lat_label, <span class="at">label =</span> Name), <span class="at">fill =</span> <span class="st">"#FFD27D"</span>,<span class="at">size =</span> <span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede_Map</span>()<span class="sc">+</span></span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Longitude"</span>,</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>       <span class="at">y=</span><span class="st">"Latitude"</span>)<span class="sc">+</span></span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="sc">-</span><span class="fl">3.725</span>) <span class="sc">+</span></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fl">47.82</span>) <span class="sc">+</span></span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Move y-axis to the right</span></span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(<span class="st">"white"</span>, <span class="dv">0</span>)),</span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>),</span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="fl">7.5</span>,  <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">7</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.box =</span> <span class="st">"vertical"</span>,  <span class="co"># Stack legends vertically</span></span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.box.just =</span> <span class="st">"left"</span>,  <span class="co"># Align the legends to the left</span></span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.spacing.y =</span> <span class="fu">unit</span>(<span class="fl">0.0</span>, <span class="st">'mm'</span>),  <span class="co"># Adjust the spacing between items</span></span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="st">"black"</span>),  <span class="co"># Ensure clear legend keys</span></span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.box.margin =</span> <span class="fu">margin</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>),</span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major =</span> ggplot2<span class="sc">::</span><span class="fu">element_line</span>(<span class="at">linetype =</span> <span class="st">"dotted"</span>, </span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a>      <span class="at">colour =</span> <span class="fu">alpha</span>(<span class="st">"grey30"</span>,<span class="fl">0.5</span>), <span class="at">linewidth =</span> <span class="fl">0.25</span>)</span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a>    )  <span class="co"># Adjust position if needed  )</span></span>
<span id="cb14-62"><a href="#cb14-62" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb14-63"><a href="#cb14-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-64"><a href="#cb14-64" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Figure1/Map_AB.png"</span>,Plot_Aven,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">400</span>)  </span>
<span id="cb14-65"><a href="#cb14-65" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Low_res/Figure1/Map_AB.png"</span>,Plot_Aven,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">200</span>)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [11]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>It_AB <span class="ot">&lt;-</span> <span class="st">"Data/shp/Intertidal_sites/mask_intertidal_Aven_Belon.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>Land_AB <span class="ot">&lt;-</span> <span class="st">"Data/shp/Intertidal_sites/mask_Land_Aven_Belon.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>ext_plot_B <span class="ot">&lt;-</span> <span class="fu">st_bbox</span>(<span class="fu">create_square_sf</span>(<span class="dv">8000</span>, <span class="fu">c</span>(<span class="sc">-</span><span class="fl">3.690000</span>, <span class="fl">47.800000</span>)))</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>Flight_AB<span class="ot">&lt;-</span><span class="fu">data.frame</span>(</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Name=</span><span class="fu">c</span>(<span class="st">"Pont du Guilly"</span>),</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">Long=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">3.655220</span>) ,</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">Lat=</span><span class="fu">c</span>(<span class="fl">47.822408</span>)</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  )  <span class="sc">%&gt;%</span></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>(<span class="at">coords=</span><span class="fu">c</span>(<span class="st">"Long"</span>,<span class="st">"Lat"</span>) )  <span class="sc">%&gt;%</span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_set_crs</span>(<span class="st">"EPSG:4326"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_transform</span>(<span class="fu">crs</span>(Land_Saja))  <span class="sc">%&gt;%</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">lon =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(.) [,<span class="dv">1</span>],</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>                <span class="at">lat =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(.) [,<span class="dv">2</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>  sf<span class="sc">::</span><span class="fu">st_set_geometry</span>(<span class="cn">NULL</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">lon_label =</span> <span class="fu">case_when</span>(Name <span class="sc">==</span> <span class="st">"Pont du Guilly"</span> <span class="sc">~</span> <span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">1</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">3</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">1</span>]))<span class="sc">*</span><span class="fl">0.7</span>),</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">lat_label =</span> <span class="fu">case_when</span>(Name <span class="sc">==</span>  <span class="st">"Pont du Guilly"</span> <span class="sc">~</span>  <span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">2</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">4</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">2</span>]))<span class="sc">*</span><span class="fl">0.7</span>))</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>(plot_Belon <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> Land_AB, <span class="at">linewidth =</span> <span class="fl">0.05</span>, <span class="at">alpha =</span> <span class="fl">0.93</span>, <span class="at">fill =</span> <span class="st">"grey80"</span>) <span class="sc">+</span></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> It_AB, <span class="at">linewidth =</span> <span class="fl">0.05</span>, <span class="at">alpha =</span> <span class="fl">0.93</span>, <span class="at">fill =</span> <span class="st">"goldenrod"</span>) <span class="sc">+</span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_label</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">1</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">3</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">1</span>]))<span class="sc">*</span><span class="fl">0.1</span> , </span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> <span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">2</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">4</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_B[<span class="dv">2</span>]))<span class="sc">*</span><span class="fl">0.1</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>                   , <span class="at">label =</span> <span class="st">"B"</span>), </span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>               <span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.5</span> )<span class="sc">+</span></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(ext_plot_B[<span class="dv">1</span>], ext_plot_B[<span class="dv">3</span>]),</span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylim =</span> <span class="fu">c</span>(ext_plot_B[<span class="dv">2</span>], ext_plot_B[<span class="dv">4</span>]),</span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> F</span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">data =</span> Flight_AB, <span class="fu">aes</span>(<span class="at">x =</span> lon_label, <span class="at">xend =</span> lon, <span class="at">yend =</span> lat, <span class="at">y =</span>lat_label, <span class="at">group =</span> Name), <span class="at">color =</span> <span class="st">"goldenrod4"</span>,<span class="at">size =</span> <span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> Flight_AB, <span class="fu">aes</span>(<span class="at">x =</span> lon, <span class="at">y =</span>lat), <span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label</span>(<span class="at">data =</span> Flight_AB, <span class="fu">aes</span>(<span class="at">x =</span> lon_label, <span class="at">y =</span>lat_label, <span class="at">label =</span> Name), <span class="at">fill =</span> <span class="st">"#FFD27D"</span>,<span class="at">size =</span> <span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede_Map</span>()<span class="sc">+</span></span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Longitude"</span>,</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>       <span class="at">y=</span><span class="st">"Latitude"</span>)<span class="sc">+</span></span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="sc">-</span><span class="fl">3.67</span>) <span class="sc">+</span></span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fl">47.82</span>) <span class="sc">+</span></span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Move y-axis to the right</span></span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(<span class="st">"white"</span>, <span class="dv">0</span>)),</span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>),</span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="fl">7.5</span>,  <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">7</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.box =</span> <span class="st">"vertical"</span>,  <span class="co"># Stack legends vertically</span></span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.box.just =</span> <span class="st">"left"</span>,  <span class="co"># Align the legends to the left</span></span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.spacing.y =</span> <span class="fu">unit</span>(<span class="fl">0.0</span>, <span class="st">'mm'</span>),  <span class="co"># Adjust the spacing between items</span></span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="st">"black"</span>),  <span class="co"># Ensure clear legend keys</span></span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.box.margin =</span> <span class="fu">margin</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>),</span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major =</span> ggplot2<span class="sc">::</span><span class="fu">element_line</span>(<span class="at">linetype =</span> <span class="st">"dotted"</span>, </span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a>      <span class="at">colour =</span> <span class="fu">alpha</span>(<span class="st">"grey30"</span>,<span class="fl">0.5</span>), <span class="at">linewidth =</span> <span class="fl">0.25</span>)</span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a>    )  <span class="co"># Adjust position if needed  )</span></span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Figure1/Map_AB.png"</span>,plot_AB,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">400</span>)  </span>
<span id="cb15-64"><a href="#cb15-64" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Low_res/Figure1/Map_AB.png"</span>,plot_AB,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">200</span>)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [12]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>It_Etel <span class="ot">&lt;-</span> <span class="st">"Data/shp/Intertidal_sites/mask_intertidal_Etel.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>Land_Etel <span class="ot">&lt;-</span> <span class="st">"Data/shp/Intertidal_sites/mask_Land_Etel.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>ext_plot_Etel <span class="ot">&lt;-</span> <span class="fu">st_bbox</span>(<span class="fu">create_square_sf</span>(<span class="dv">10000</span>, <span class="fu">c</span>(<span class="sc">-</span><span class="fl">3.186821</span>, <span class="fl">47.683477</span>)))</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>Flight_Etel<span class="ot">&lt;-</span><span class="fu">data.frame</span>(</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">Name=</span><span class="fu">c</span>(</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Berringue"</span>,</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Lorois"</span>),</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">Long=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">3.194274</span>,</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span><span class="fl">3.205714</span>) ,</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">Lat=</span><span class="fu">c</span>(<span class="fl">47.698978</span>,</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>        <span class="fl">47.675159</span>)</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>  )  <span class="sc">%&gt;%</span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_as_sf</span>(<span class="at">coords=</span><span class="fu">c</span>(<span class="st">"Long"</span>,<span class="st">"Lat"</span>) )  <span class="sc">%&gt;%</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_set_crs</span>(<span class="st">"EPSG:4326"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_transform</span>(<span class="fu">crs</span>(Land_Saja))  <span class="sc">%&gt;%</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">lon =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(.) [,<span class="dv">1</span>],</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>                <span class="at">lat =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(.) [,<span class="dv">2</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>  sf<span class="sc">::</span><span class="fu">st_set_geometry</span>(<span class="cn">NULL</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">lon_label =</span> <span class="fu">case_when</span>(Name <span class="sc">==</span> <span class="st">"Berringue"</span> <span class="sc">~</span> lon <span class="sc">-</span> <span class="dv">1000</span>,</span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>                               Name <span class="sc">==</span> <span class="st">"Lorois"</span> <span class="sc">~</span> lon <span class="sc">+</span> <span class="dv">2000</span>),</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">lat_label =</span> <span class="fu">case_when</span>(Name <span class="sc">==</span>  <span class="st">"Berringue"</span> <span class="sc">~</span> lat <span class="sc">+</span> <span class="dv">1000</span>,</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>                               Name <span class="sc">==</span> <span class="st">"Lorois"</span> <span class="sc">~</span> lat <span class="sc">-</span> <span class="dv">1000</span>))</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>(plot_Etel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> Land_Etel, <span class="at">linewidth =</span> <span class="fl">0.05</span>, <span class="at">alpha =</span> <span class="fl">0.93</span>, <span class="at">fill =</span> <span class="st">"grey80"</span>) <span class="sc">+</span></span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> It_Etel, <span class="at">linewidth =</span> <span class="fl">0.05</span>, <span class="at">alpha =</span> <span class="fl">0.93</span>, <span class="at">fill =</span> <span class="st">"goldenrod"</span>) <span class="sc">+</span></span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_label</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(ext_plot_Etel[<span class="dv">1</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_Etel[<span class="dv">3</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_Etel[<span class="dv">1</span>]))<span class="sc">*</span><span class="fl">0.1</span> , </span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> <span class="fu">as.numeric</span>(ext_plot_Etel[<span class="dv">2</span>]) <span class="sc">+</span> (<span class="fu">as.numeric</span>(ext_plot_Etel[<span class="dv">4</span>])<span class="sc">-</span><span class="fu">as.numeric</span>(ext_plot_Etel[<span class="dv">2</span>]))<span class="sc">*</span><span class="fl">0.1</span></span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>                   , <span class="at">label =</span> <span class="st">"B"</span>), </span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a>               <span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.5</span> )<span class="sc">+</span></span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(</span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(ext_plot_Etel[<span class="dv">1</span>], ext_plot_Etel[<span class="dv">3</span>]),</span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylim =</span> <span class="fu">c</span>(ext_plot_Etel[<span class="dv">2</span>], ext_plot_Etel[<span class="dv">4</span>]),</span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ndiscr = 1,</span></span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="cn">FALSE</span></span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="sc">-</span><span class="fl">3.2</span>) <span class="sc">+</span></span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fl">47.68</span>) <span class="sc">+</span></span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">data =</span> Flight_Etel, <span class="fu">aes</span>(<span class="at">x =</span> lon_label, <span class="at">xend =</span> lon, <span class="at">yend =</span> lat, <span class="at">y =</span>lat_label, <span class="at">group =</span> Name), <span class="at">color =</span> <span class="st">"goldenrod4"</span>,<span class="at">size =</span> <span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> Flight_Etel, <span class="fu">aes</span>(<span class="at">x =</span> lon, <span class="at">y =</span>lat), <span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label</span>(<span class="at">data =</span> Flight_Etel, <span class="fu">aes</span>(<span class="at">x =</span> lon_label, <span class="at">y =</span>lat_label, <span class="at">label =</span> Name), <span class="at">fill =</span> <span class="st">"#FFD27D"</span>,<span class="at">size =</span> <span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede_Map</span>() <span class="sc">+</span></span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Longitude"</span>, <span class="at">y =</span> <span class="st">"Latitude"</span>)<span class="sc">+</span></span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_x_continuous(breaks = seq(floor(ext_plot_Etel[1]), ceiling(ext_plot_Etel[3]), by = 500)) +</span></span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_y_continuous(breaks = seq(floor(ext_plot_Etel[2]), ceiling(ext_plot_Etel[4]), by = 500)) +</span></span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(<span class="st">"white"</span>, <span class="dv">0</span>)),</span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>),</span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="fl">7.5</span>,  <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">7</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-64"><a href="#cb16-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.box =</span> <span class="st">"vertical"</span>,</span>
<span id="cb16-65"><a href="#cb16-65" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.box.just =</span> <span class="st">"left"</span>,</span>
<span id="cb16-66"><a href="#cb16-66" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.spacing.y =</span> <span class="fu">unit</span>(<span class="fl">0.0</span>, <span class="st">'mm'</span>),</span>
<span id="cb16-67"><a href="#cb16-67" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb16-68"><a href="#cb16-68" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.box.margin =</span> <span class="fu">margin</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>),</span>
<span id="cb16-69"><a href="#cb16-69" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(</span>
<span id="cb16-70"><a href="#cb16-70" aria-hidden="true" tabindex="-1"></a>      <span class="at">linetype =</span> <span class="st">"dotted"</span>,</span>
<span id="cb16-71"><a href="#cb16-71" aria-hidden="true" tabindex="-1"></a>      <span class="at">colour =</span> <span class="fu">alpha</span>(<span class="st">"grey30"</span>, <span class="fl">0.5</span>),</span>
<span id="cb16-72"><a href="#cb16-72" aria-hidden="true" tabindex="-1"></a>      <span class="at">linewidth =</span> <span class="fl">0.25</span></span>
<span id="cb16-73"><a href="#cb16-73" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-74"><a href="#cb16-74" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb16-75"><a href="#cb16-75" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb16-76"><a href="#cb16-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-77"><a href="#cb16-77" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Figure1/Map_Etel.png"</span>,plot_AB,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">400</span>)  </span>
<span id="cb16-78"><a href="#cb16-78" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Low_res/Figure1/Map_Etel.png"</span>,plot_AB,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">200</span>)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [13]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>(final_map <span class="ot">&lt;-</span> ((p1 <span class="sc">|</span> Plot_Aven) <span class="sc">/</span> (plot_Belon <span class="sc">|</span> plot_Saja))  <span class="sc">+</span> </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">"collect"</span>)<span class="sc">&amp;</span> </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># theme_void()&amp;</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">"cm"</span>),</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>, <span class="at">vjust =</span> <span class="dv">6</span>, <span class="at">hjust =</span> <span class="fl">0.8</span>),</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">5</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">angle =</span> <span class="dv">90</span>))</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Figure1/Map_site.png"</span>,final_map,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">400</span>)  </span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Low_res/Figure1/Map_site.png"</span>,final_map,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">200</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [14]:</pre></div><div id="cell-fig-location_sites" class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"Figures/Low_res/Figure1/Map_site.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-location_sites" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-location_sites-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="Figures/Low_res/Figure1/Map_site.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: Location of the drone flights. A: Flights made in Aven Estuary, France; B: Flights made in Bélon Estuary, France; C: Flights made in Saja Estuaries, Spain. Golden polygons represent intertidal areas."><img src="Figures/Low_res/Figure1/Map_site.png" class="img-fluid figure-img" style="width:95.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-location_sites-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Location of the drone flights. A: Flights made in Aven Estuary, France; B: Flights made in Bélon Estuary, France; C: Flights made in Saja Estuaries, Spain. Golden polygons represent intertidal areas.
</figcaption>
</figure>
</div>
</div>
</div></div>
</section>
<section id="sec-DroneFlights" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="sec-DroneFlights"><span class="header-section-number">3.2</span> Remote sensing data acquisition and pre-processing</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [15]:</pre></div><div id="cell-fig-PictureFigure" class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"./Figures/Low_res/FigurePictures.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-PictureFigure" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-PictureFigure-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="./Figures/Low_res/FigurePictures.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: Pictures of Gracilaria vermiculophylla. A: Quadrat of 0.25 m² with a high cover of G. vermiculophylla; B: Single thallus of the algae; C: Landscape view with a high coverage of G. vermiculophylla; D: Recording of the spectral signature of the algae using an ASD FieldSpec HandHeld 2 spectroradiometer."><img src="./Figures/Low_res/FigurePictures.png" class="img-fluid figure-img" style="width:95.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-PictureFigure-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Pictures of <em>Gracilaria vermiculophylla</em>. A: Quadrat of 0.25 m² with a high cover of <em>G. vermiculophylla</em>; B: Single thallus of the algae; C: Landscape view with a high coverage of G. vermiculophylla; D: Recording of the spectral signature of the algae using an ASD FieldSpec HandHeld 2 spectroradiometer.
</figcaption>
</figure>
</div>
</div>
</div></div>
</section>
<section id="hyperspectral-measurements" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="hyperspectral-measurements"><span class="header-section-number">3.3</span> Hyperspectral measurements</h2>
<p>At each location, hyperspectral reflectance signatures were recorded using an ASD FieldSpec HandHeld 2 spectroradiometer (Malvern Panalytical, Worcestershire, UK), which measures reflectance from 325 to 1075 nm with a spectral resolution of approximately 1 nm. Each spectrum was subsequently smoothed using a Savitzky–Golay filter <span class="citation" data-cites="savitzky1964smoothing">(<a href="#ref-savitzky1964smoothing" role="doc-biblioref">Savitzky and Golay, 1964</a>)</span> with a third-order polynomial and an 11-point window, selected to minimize noise while preserving salient spectral features. After this initial smoothing, the first and second derivatives were computed using a central difference approximation (<a href="#eq-SecondDerivative" class="quarto-xref">Equation&nbsp;1</a>).</p>
<p><span id="eq-SecondDerivative"><span class="math display">\[
f''(\lambda_i) \approx \frac{f(\lambda_{i+1}) - 2f(\lambda_i) + f(\lambda_{i-1})}{(\Delta \lambda)^2}
\tag{1}\]</span></span></p>
<p>where $ f(_i) $ is the reflectance at wavelength <span class="math inline">\(\lambda_i\)</span> and <span class="math inline">\(\Delta \lambda\)</span> is the uniform spectral sampling interval.</p>
</section>
<section id="drone-data" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="drone-data"><span class="header-section-number">3.4</span> Drone data</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [16]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flextable)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: le package 'flextable' a été compilé avec la version R 4.3.3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: le package 'tidyverse' a été compilé avec la version R 4.3.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: le package 'ggplot2' a été compilé avec la version R 4.3.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: le package 'tidyr' a été compilé avec la version R 4.3.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: le package 'readr' a été compilé avec la version R 4.3.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: le package 'dplyr' a été compilé avec la version R 4.3.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: le package 'stringr' a été compilé avec la version R 4.3.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: le package 'lubridate' a été compilé avec la version R 4.3.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ purrr::compose() masks flextable::compose()
✖ dplyr::filter()  masks stats::filter()
✖ dplyr::lag()     masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>

<div class="cell-container"><div class="cell-decorator"><pre>In [17]:</pre></div><div><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Country =</span> <span class="fu">c</span>(<span class="st">"France"</span>, <span class="st">"France"</span>,<span class="st">"Spain"</span>, <span class="st">"Spain"</span>),</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Site =</span> <span class="fu">c</span>(<span class="st">"Aven &amp; Belon"</span>,<span class="st">"Aven &amp; Belon"</span>, <span class="st">"Saja Estuary"</span>,<span class="st">"Saja Estuary"</span>),</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Flight  =</span> <span class="fu">c</span>(<span class="st">"Notre-Dame De Tremor"</span>,<span class="st">"Pont du Guilly"</span>,<span class="st">"Marisma de Cortiguera"</span>,<span class="st">"Marisma de Cudón"</span>),</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Date =</span> <span class="fu">c</span>(<span class="st">"2024-04-11"</span>,<span class="st">"2024-04-11"</span>,<span class="st">"2024-06-25"</span>,<span class="st">"2024-06-25"</span>), </span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Area =</span> <span class="fu">c</span>(<span class="dv">266582</span>,<span class="dv">213064</span>,<span class="dv">204133</span>,<span class="dv">83757</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Area =</span> <span class="fu">round</span>(Area <span class="sc">*</span> <span class="fl">0.0001</span>,<span class="dv">1</span>))</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>brdr1 <span class="ot">&lt;-</span> <span class="fu">fp_border_default</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">width =</span> <span class="fl">0.5</span>)</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>brdr2 <span class="ot">&lt;-</span> <span class="fu">fp_border_default</span>(<span class="at">color =</span> <span class="st">"grey"</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>brdr3 <span class="ot">&lt;-</span> <span class="fu">fp_border_default</span>(<span class="at">color =</span> <span class="st">"grey40"</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a> i <span class="ot">=</span> <span class="fl">16.5</span> <span class="co"># width of the side borders in the word_document output (in centimeters)</span></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a> w <span class="ot">=</span> i<span class="sc">*</span><span class="fl">0.3937</span> <span class="co"># width of the side borders in the word_document output (in inches)</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a><span class="fu">flextable</span>(df)  <span class="sc">%&gt;%</span></span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>  flextable<span class="sc">::</span><span class="fu">width</span>(<span class="at">width =</span> (w<span class="sc">/</span>(<span class="fu">ncol</span>(df)))) <span class="sc">%&gt;%</span> </span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge_v</span>(<span class="at">j =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge_v</span>(<span class="at">j =</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">border_remove</span>() <span class="sc">%&gt;%</span></span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">hline_top</span>(<span class="at">border=</span>brdr1) <span class="sc">%&gt;%</span></span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">hline_bottom</span>(<span class="at">border=</span>brdr1) <span class="sc">%&gt;%</span> </span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">hline</span>(<span class="at">i=</span><span class="dv">4</span>, <span class="at">border=</span>brdr1) <span class="sc">%&gt;%</span> </span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">hline</span>(<span class="at">i=</span><span class="dv">2</span>, <span class="at">border=</span>brdr1) <span class="sc">%&gt;%</span> </span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">hline</span>(<span class="at">i=</span><span class="dv">1</span>, <span class="at">border=</span>brdr2)<span class="sc">%&gt;%</span> </span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">hline</span>(<span class="at">i=</span><span class="dv">3</span>, <span class="at">border=</span>brdr2)<span class="sc">%&gt;%</span> </span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># hline(i=5, border=brdr2) %&gt;% </span></span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a>    flextable<span class="sc">::</span><span class="fu">align</span>(<span class="at">align =</span> <span class="st">"center"</span>,<span class="at">part =</span> <span class="st">"all"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a>    flextable<span class="sc">::</span><span class="fu">align</span>(<span class="at">align =</span> <span class="st">"center"</span>,<span class="at">part =</span> <span class="st">"header"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set_caption(caption = "Table 1 List of drone Flight, summarising the date, the altitude and the purpose of each flight.") %&gt;%</span></span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_table_properties</span>(<span class="at">layout =</span> <span class="st">"autofit"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div><div id="tbl-flights" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-flights-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: List of drone flights, summarising the location, the date, and the total extent of each flight (in hectars).
</figcaption>
<div aria-describedby="tbl-flights-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid"><style>.cl-e6dd9f60{table-layout:auto;}.cl-e6d4dca4{font-family:'Arial';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-e6d842e0{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-e6d85a8c{background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e6d85a8d{background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(190, 190, 190, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e6d85a96{background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(0, 0, 0, 1.00);border-top: 1pt solid rgba(190, 190, 190, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e6d85a97{background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(0, 0, 0, 1.00);border-top: 0.5pt solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e6d85a98{background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(190, 190, 190, 1.00);border-top: 0.5pt solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-e6dd9f60"><thead><tr style="overflow-wrap:break-word;"><th class="cl-e6d85a8c"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Country</span></p></th><th class="cl-e6d85a8c"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Site</span></p></th><th class="cl-e6d85a8c"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Flight</span></p></th><th class="cl-e6d85a8c"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Date</span></p></th><th class="cl-e6d85a8c"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Area</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td rowspan="2" class="cl-e6d85a8c"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">France</span></p></td><td rowspan="2" class="cl-e6d85a8c"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Aven &amp; Belon</span></p></td><td class="cl-e6d85a8d"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Notre-Dame De Tremor</span></p></td><td class="cl-e6d85a8d"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">2024-04-11</span></p></td><td class="cl-e6d85a8d"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">26.7</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e6d85a96"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Pont du Guilly</span></p></td><td class="cl-e6d85a96"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">2024-04-11</span></p></td><td class="cl-e6d85a96"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">21.3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td rowspan="2" class="cl-e6d85a97"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Spain</span></p></td><td rowspan="2" class="cl-e6d85a97"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Saja Estuary</span></p></td><td class="cl-e6d85a98"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Marisma de Cortiguera</span></p></td><td class="cl-e6d85a98"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">2024-06-25</span></p></td><td class="cl-e6d85a98"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">20.4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e6d85a96"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">Marisma de Cudón</span></p></td><td class="cl-e6d85a96"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">2024-06-25</span></p></td><td class="cl-e6d85a96"><p class="cl-e6d842e0"><span class="cl-e6d4dca4">8.4</span></p></td></tr></tbody></table></div>
</div>
</div>
</figure>
</div></div></div>
</div></div>
<p>A total of 6 drone flights were done spread in the 3 study sites. Each time, flights were done at an altitude of 120 m and at a speed of 10 m.s<sup>-1</sup> (<a href="#tbl-flights" class="quarto-xref">Table&nbsp;1</a>).</p>
<section id="sec-photo" class="level4" data-number="3.4.0.1">
<h4 data-number="3.4.0.1" class="anchored" data-anchor-id="sec-photo"><span class="header-section-number">3.4.0.1</span> Multispectral data</h4>
<p>At each location, reflectance images with a resolution of 1.2 million pixels were captured using a DJI Matrice 300 quadcopter drone equipped with a Micasense RedEdge Dual MX multispectral camera. The camera recorded data across ten spectral bands, spanning from blue to near-infrared (NIR) wavelengths (444, 475, 531, 560, 650, 668, 705, 717, 740, and 840 nm) (). To ensure consistent lighting conditions, the drone’s flight trajectory was aligned to maintain a solar azimuth angle of 90 degrees. Image acquisition was carried out with an overlap of 70% between side-by-side images and 80% between successive images along the flight path. A downwelling light sensor (DLS2) was used to measure real-time irradiance, enabling the correction of reflectance values for variations in light intensity caused by cloud cover during the flight. The raw image data were subsequently calibrated to reflectance using a calibration panel with ~50% reflectivity, provided by the camera’s manufacturer. Images were processed using structure-from-motion photogrammetry software <span class="citation" data-cites="agisoft">(<a href="#ref-agisoft" role="doc-biblioref">Agisoft, 2019</a>)</span> to generate multispectral ortho-mosaics for each flight. The ortho-mosaicking workflow was consistent across all flights. Initially, key tie points were identified within each image and across overlapping images to create a sparse point cloud. This point cloud was refined by removing noisy points using a reprojection accuracy metric. Subsequently, a dense point cloud was generated using a structure-from-motion algorithm. A digital surface model (DSM) was then created through surface interpolation of the dense point cloud, which served as the basis for reconstructing the multispectral ortho-image <span class="citation" data-cites="nebel2020review">(<a href="#ref-nebel2020review" role="doc-biblioref">Nebel et al., 2020</a>)</span>. The resolution of the multispectral ortho-mosaic obtained were 8 cm per pixel.</p>
</section>
<section id="lidar-data" class="level4" data-number="3.4.0.2">
<h4 data-number="3.4.0.2" class="anchored" data-anchor-id="lidar-data"><span class="header-section-number">3.4.0.2</span> LiDAR data</h4>
<p>LiDAR standing for Light Detection and Ranging uses lasers to measure distances by timing reflected pulses, creating detailed 3D maps of surfaces.</p>
<p>Using the Matrice 300 Series Dual Gimbal Connector, a DJI Zenmuse L1 LiDAR and RGB sensor was mounted on the drone alongside a multispectral camera. This setup enabled the simultaneous capture of LiDAR point clouds, high-resolution RGB images, and multispectral images collected by the MicaSense RedEdge Dual MX during the same flight. The same processing workflow as <a href="#sec-photo" class="quarto-xref">Section&nbsp;3.4.0.1</a> was applied to process LiDAR RGB images, resulting in ortho-mosaic with a resolution of 2.5 cm per pixel. Since the mapping focused solely on flat surfaces without dense vegetation, the LiDAR measured only a single return. Operating in repetitive scanning mode with a sampling rate of 240 kHz, the system achieved a point density of 350 points per square meter. The LiDAR point cloud was extracted and converted into LAS format using DJI Terra software. The LAS point cloud was then imported into Agisoft Metashape <span class="citation" data-cites="agisoft">(<a href="#ref-agisoft" role="doc-biblioref">Agisoft, 2019</a>)</span> to generate a Digital Surface Model (DSM) with a resolution of 2.5 cm. From the DSM, the slope of each pixel based on a grid of 8 surrounding pixels were computed using the terrain function of the ‘terra’ package in R <span class="citation" data-cites="terrapck">(<a href="#ref-terrapck" role="doc-biblioref">Hijmans, 2024</a>)</span>. The angle of the mudflat was categorized into three classes: Flat (angle &lt; 10°), Angled (10° ≤ angle ≤ 40°), and Vertical (angle &gt; 40°).</p>
</section>
</section>
<section id="scene-classification" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="scene-classification"><span class="header-section-number">3.5</span> Scene classification</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [18]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flextable)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>df_training <span class="ot">&lt;-</span> <span class="st">"../Data/Training_DISCOV/DISCOV_Invasea_Saja_filtered.txt"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read.delim</span>(<span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>brdr1 <span class="ot">&lt;-</span> <span class="fu">fp_border_default</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">width =</span> <span class="fl">0.5</span>)</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>brdr2 <span class="ot">&lt;-</span> <span class="fu">fp_border_default</span>(<span class="at">color =</span> <span class="st">"grey"</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>brdr3 <span class="ot">&lt;-</span> <span class="fu">fp_border_default</span>(<span class="at">color =</span> <span class="st">"grey40"</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>i <span class="ot">=</span> <span class="fl">16.5</span> <span class="co"># width of the side borders in the word_document output (in centimeters)</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>w <span class="ot">=</span> i<span class="sc">*</span><span class="fl">0.3937</span> <span class="co"># width of the side borders in the word_document output (in inches)</span></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>df_table <span class="ot">&lt;-</span> df_training <span class="sc">%&gt;%</span> </span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(True_Class <span class="sc">!=</span> <span class="st">"SunGlint"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DISCOV1_n =</span> <span class="fu">case_when</span>(True_Class <span class="sc">==</span> <span class="st">"Bacillariophyceae"</span> <span class="sc">~</span> <span class="dv">4475</span>,</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>                          True_Class <span class="sc">==</span> <span class="st">"Chlorophyta"</span> <span class="sc">~</span> <span class="dv">17140</span>,</span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>                          True_Class <span class="sc">==</span> <span class="st">"Magnoliopsida"</span> <span class="sc">~</span> <span class="dv">221065</span>,</span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>                          True_Class <span class="sc">==</span> <span class="st">"Phaeophyta"</span> <span class="sc">~</span> <span class="dv">169936</span>,</span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>                          True_Class <span class="sc">==</span> <span class="st">"Rhodophyta"</span> <span class="sc">~</span> <span class="dv">5771</span>,</span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>                          True_Class <span class="sc">==</span> <span class="st">"Water"</span> <span class="sc">~</span> <span class="dv">83677</span>,</span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>                          True_Class <span class="sc">==</span> <span class="st">"Sediment"</span> <span class="sc">~</span> <span class="dv">95126</span>),</span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">Name =</span> <span class="fu">case_when</span>(True_Class <span class="sc">==</span> <span class="st">"Bacillariophyceae"</span> <span class="sc">~</span> <span class="st">"Benthic Diatoms"</span>,</span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a>                          True_Class <span class="sc">==</span> <span class="st">"Chlorophyta"</span> <span class="sc">~</span> <span class="st">"Green macroalgae"</span>,</span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>                          True_Class <span class="sc">==</span> <span class="st">"Magnoliopsida"</span> <span class="sc">~</span> <span class="st">"Seagrass"</span>,</span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a>                          True_Class <span class="sc">==</span> <span class="st">"Phaeophyta"</span> <span class="sc">~</span> <span class="st">"Brown macroalgae"</span>,</span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a>                          True_Class <span class="sc">==</span> <span class="st">"Rhodophyta"</span> <span class="sc">~</span> <span class="st">"Red macroalgae"</span>,</span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>                          T <span class="sc">~</span> True_Class),</span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">True_Class =</span> <span class="fu">case_when</span>(True_Class <span class="sc">==</span> <span class="st">"Sediment"</span> <span class="sc">~</span> <span class="st">"-"</span>,</span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a>                                True_Class <span class="sc">==</span> <span class="st">"Water"</span> <span class="sc">~</span> <span class="st">"-"</span>,</span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a>                                T <span class="sc">~</span> True_Class),</span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Difference with DISCOV v1.0</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">round</span>((n)<span class="sc">/</span>DISCOV1_n,<span class="dv">2</span>),</span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Difference with DISCOV v1.0</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">case_when</span>(<span class="st">`</span><span class="at">Difference with DISCOV v1.0</span><span class="st">`</span> <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"-"</span>,</span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a>                                                   T <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"x"</span>,<span class="st">`</span><span class="at">Difference with DISCOV v1.0</span><span class="st">`</span>))) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>A neural network classification model (DISCOV; <span class="citation" data-cites="rs16234383">Oiry et al. (<a href="#ref-rs16234383" role="doc-biblioref">2024b</a>)</span>; <span class="citation" data-cites="oiry_2024_14218984">Oiry et al. (<a href="#ref-oiry_2024_14218984" role="doc-biblioref">2024a</a>)</span>), previously applied with success to Micasense reflectance data for mapping intertidal vegetation along the Portuguese and French Atlantic coasts, has been used in this study. The training dataset of DISCOV v1.0 has been updated. As shown by <span class="citation" data-cites="rs16234383">Oiry et al. (<a href="#ref-rs16234383" role="doc-biblioref">2024b</a>)</span> the DISCOV v1.0 model <span class="citation" data-cites="oiry_2024_14218984">(<a href="#ref-oiry_2024_14218984" role="doc-biblioref">Oiry et al., 2024a</a>)</span> was trained using only 5771 Rhodophyceae pixel (3% of the training dataset). To fill this gap the original training dataset of DISCOV v1.0 was updated using new training pixel coming from the 5 drone flights (<a href="#sec-DroneFlights" class="quarto-xref">Section&nbsp;3.2</a>). A total of 427000 pixels where added to the DISCOV training dataset compared to the version 1 (<a href="#tbl-Update_training" class="quarto-xref">Table&nbsp;2</a>).</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [19]:</pre></div><div class="cell">

<div class="cell-container"><div class="cell-decorator"><pre>In [20]:</pre></div><div><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flextable)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>df_table <span class="sc">%&gt;%</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">Taxonomic Class</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"True_Class"</span>,</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Training Pixels</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"n"</span>)  <span class="sc">%&gt;%</span> </span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Name,<span class="st">`</span><span class="at">Taxonomic Class</span><span class="st">`</span>,<span class="st">`</span><span class="at">Training Pixels</span><span class="st">`</span>, <span class="st">`</span><span class="at">Difference with DISCOV v1.0</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">align</span>(<span class="at">align =</span> <span class="st">"center"</span>, <span class="at">part =</span> <span class="st">"all"</span>) <span class="sc">%&gt;%</span> <span class="co"># Align text to center</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_table_properties</span>(<span class="at">layout =</span> <span class="st">"autofit"</span>) <span class="co"># Adjust table layout</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div><div id="tbl-Update_training" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-Update_training-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2: Class of the Neural Network model, with the number of training pixels used to train that class and the differences with the training dataset of DISCOV v1.0
</figcaption>
<div aria-describedby="tbl-Update_training-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid"><style>.cl-e6f9bf10{table-layout:auto;}.cl-e6f14db2{font-family:'Arial';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-e6f455ac{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-e6f46b46{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e6f46b47{background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e6f46b48{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-e6f9bf10"><thead><tr style="overflow-wrap:break-word;"><th class="cl-e6f46b46"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Name</span></p></th><th class="cl-e6f46b46"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Taxonomic Class</span></p></th><th class="cl-e6f46b46"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Training Pixels</span></p></th><th class="cl-e6f46b46"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Difference with DISCOV v1.0</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Benthic Diatoms</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Bacillariophyceae</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">62,436</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">x13.95</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Green macroalgae</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Chlorophyta</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">92,585</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">x5.4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Seagrass</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Magnoliopsida</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">221,065</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">-</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Brown macroalgae</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Phaeophyta</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">169,936</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">-</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Red macroalgae</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Rhodophyta</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">268,637</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">x46.55</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Sediment</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">-</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">117,956</span></p></td><td class="cl-e6f46b47"><p class="cl-e6f455ac"><span class="cl-e6f14db2">x1.24</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e6f46b48"><p class="cl-e6f455ac"><span class="cl-e6f14db2">Water</span></p></td><td class="cl-e6f46b48"><p class="cl-e6f455ac"><span class="cl-e6f14db2">-</span></p></td><td class="cl-e6f46b48"><p class="cl-e6f455ac"><span class="cl-e6f14db2">91,614</span></p></td><td class="cl-e6f46b48"><p class="cl-e6f455ac"><span class="cl-e6f14db2">x1.09</span></p></td></tr></tbody></table></div>
</div>
</div>
</figure>
</div></div></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [21]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: le package 'sf' a été compilé avec la version R 4.3.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Linking to GEOS 3.11.2, GDAL 3.8.2, PROJ 9.3.1; sf_use_s2() is TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>df_validation <span class="ot">&lt;-</span> </span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"../Data/Shiny_Validation/shp_all_validation.shp"</span> <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># "Data/Shiny_Validation/shp_all_validation.shp" %&gt;%</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">user =</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(user_date, <span class="st">"_"</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>]),</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">site =</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(snapshot, <span class="st">"_"</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>To validate the new version of the DISCOV model, a Shiny app was developed, enabling independent users to photo-interpret snapshots of the ortho-mosaic from each drone flight <span class="citation" data-cites="shinypck Simon2024ShinyApp">(<a href="#ref-shinypck" role="doc-biblioref">Chang et al., 2024</a>; <a href="#ref-Simon2024ShinyApp" role="doc-biblioref">Simon, 2024</a>)</span>. Users could click on various parts of the snapshots to indicate the type of vegetation they believed was present. Using this method, 3 independent users contributed to creating a validation dataset of 6755 pixels across 79 snapshots distributed among the four drone flights <a href="#tbl-ValidationDataset" class="quarto-xref">Table&nbsp;3</a>. The validation dataset was then simplify into two classes: Presence of Absence of Red Algae.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [22]:</pre></div><div class="cell">

<div class="cell-container"><div class="cell-decorator"><pre>In [23]:</pre></div><div><div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flextable)</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>df_flx_valid <span class="ot">&lt;-</span> df_validation <span class="sc">%&gt;%</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">site =</span> <span class="fu">case_when</span>(site <span class="sc">==</span> <span class="st">"Aven"</span> <span class="sc">~</span> <span class="st">"Notre-Dame De Tremor"</span>,</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>                          site <span class="sc">==</span> <span class="st">"Belon"</span> <span class="sc">~</span> <span class="st">"Pont de Guilly"</span>,</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>                          site <span class="sc">==</span> <span class="st">"SajaNorth"</span> <span class="sc">~</span> <span class="st">"Marisma de Cudón"</span>,</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>                          site <span class="sc">==</span> <span class="st">"SajaSouth"</span> <span class="sc">~</span> <span class="st">"Marisma de Cortiguera"</span>),</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">PresenceAbs =</span> <span class="fu">case_when</span>(Vegetation <span class="sc">==</span> <span class="st">"Red algae"</span> <span class="sc">~</span> <span class="st">"Present"</span>,</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>                                 T <span class="sc">~</span> <span class="st">"Absent"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Site =</span> <span class="st">"site"</span>) </span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>total <span class="ot">&lt;-</span> df_flx_valid <span class="sc">%&gt;%</span> </span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(PresenceAbs) <span class="sc">%&gt;%</span> </span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> PresenceAbs, <span class="at">values_from =</span> n) <span class="sc">%&gt;%</span> </span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Site =</span> <span class="st">"Total"</span>)</span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a>table_data <span class="ot">&lt;-</span> df_flx_valid <span class="sc">%&gt;%</span> </span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Site, PresenceAbs) <span class="sc">%&gt;%</span> </span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"PresenceAbs"</span>, <span class="at">values_from =</span> <span class="st">"n"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(total) <span class="sc">%&gt;%</span> </span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Total =</span> <span class="fu">sum</span>(<span class="fu">c</span>(Absent, Present)))</span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the number of rows for the table data</span></span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a>n_rows <span class="ot">&lt;-</span> <span class="fu">nrow</span>(table_data)</span>
<span id="cb38-29"><a href="#cb38-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-30"><a href="#cb38-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the flextable with a horizontal line</span></span>
<span id="cb38-31"><a href="#cb38-31" aria-hidden="true" tabindex="-1"></a>table_data <span class="sc">%&gt;%</span></span>
<span id="cb38-32"><a href="#cb38-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-33"><a href="#cb38-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">align</span>(<span class="at">align =</span> <span class="st">"center"</span>, <span class="at">part =</span> <span class="st">"all"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-34"><a href="#cb38-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_table_properties</span>(<span class="at">layout =</span> <span class="st">"autofit"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-35"><a href="#cb38-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hline</span>(<span class="at">i =</span> n_rows <span class="sc">-</span> <span class="dv">1</span>, <span class="at">border =</span> officer<span class="sc">::</span><span class="fu">fp_border</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">width =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div><div id="tbl-ValidationDataset" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-ValidationDataset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3: Presence and absence of red macroalgae for each drone flight
</figcaption>
<div aria-describedby="tbl-ValidationDataset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid"><style>.cl-e7b72e60{table-layout:auto;}.cl-e7ae6c08{font-family:'Arial';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-e7b16fac{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-e7b18488{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e7b18492{background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e7b18493{background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e7b1849c{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-e7b72e60"><thead><tr style="overflow-wrap:break-word;"><th class="cl-e7b18488"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">Site</span></p></th><th class="cl-e7b18488"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">Absent</span></p></th><th class="cl-e7b18488"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">Present</span></p></th><th class="cl-e7b18488"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">Total</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">Marisma de Cortiguera</span></p></td><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">1,531</span></p></td><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">483</span></p></td><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">2,014</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">Marisma de Cudón</span></p></td><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">1,237</span></p></td><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">136</span></p></td><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">1,373</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">Notre-Dame De Tremor</span></p></td><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">1,073</span></p></td><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">463</span></p></td><td class="cl-e7b18492"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">1,536</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7b18493"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">Pont de Guilly</span></p></td><td class="cl-e7b18493"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">1,389</span></p></td><td class="cl-e7b18493"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">443</span></p></td><td class="cl-e7b18493"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">1,832</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7b1849c"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">Total</span></p></td><td class="cl-e7b1849c"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">5,230</span></p></td><td class="cl-e7b1849c"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">1,525</span></p></td><td class="cl-e7b1849c"><p class="cl-e7b16fac"><span class="cl-e7ae6c08">6,755</span></p></td></tr></tbody></table></div>
</div>
</div>
</figure>
</div></div></div>
</div></div>
</section>
<section id="historical-presence-of-gracilaria-vermiculophylla-in-the-bélon-esturay" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="historical-presence-of-gracilaria-vermiculophylla-in-the-bélon-esturay"><span class="header-section-number">3.6</span> Historical Presence of <em>Gracilaria vermiculophylla</em> in the Bélon esturay</h2>
<p>To assess the historical presence of <em>G. vermiculophylla</em> in the Bélon Estuary, aerial imagery from flight campaigns was obtained via the IGN platform “Remonter Le Temps” <span class="citation" data-cites="RemonterLeTempsIGN">(<a href="#ref-RemonterLeTempsIGN" role="doc-biblioref">IGN, 2024</a>)</span>. Nine images were selected between 1952 and 2012 from the IGN plateform and an additional one has been added for the year 2024 (<a href="#tbl-IGNimg" class="quarto-xref">Table&nbsp;4</a>). Since most of the images retrieved from “Remonter Le Temps” were digitized versions of physical photographs, georeferencing were required.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [24]:</pre></div><div class="cell">

<div class="cell-container"><div class="cell-decorator"><pre>In [25]:</pre></div><div><div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flextable)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">data.frame</span>(</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">Date =</span> <span class="fu">c</span>(</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"1952-04-26"</span>,</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"1958-04-22"</span>,</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"1976-07-? "</span>,</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"1978-08-22"</span>,</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"1982-08-11"</span>,</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"1992-05-17"</span>,</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"1997-04-11"</span>,</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">"2012-07-24"</span>,</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">"2024-04-11"</span>),</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">Type =</span> <span class="fu">c</span>(</span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Black and White"</span>,</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Black and White"</span>,</span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Black and White"</span>,</span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Black and White"</span>,</span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Black and White"</span>,</span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"True Color"</span>,</span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Black and White"</span>,</span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">"True Color"</span>,</span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">"True Color"</span>),</span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">DataSource =</span> <span class="fu">c</span>(    </span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a>    <span class="st">"IGN"</span>,</span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">"IGN"</span>,</span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a>    <span class="st">"IGN"</span>,</span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a>    <span class="st">"IGN"</span>,</span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a>    <span class="st">"IGN"</span>,</span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">"IGN"</span>,</span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">"IGN"</span>,</span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a>    <span class="st">"IGN"</span>,</span>
<span id="cb39-33"><a href="#cb39-33" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Drone Flight"</span>),</span>
<span id="cb39-34"><a href="#cb39-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">Res =</span> <span class="fu">c</span>(</span>
<span id="cb39-35"><a href="#cb39-35" aria-hidden="true" tabindex="-1"></a>     <span class="st">"10"</span>,</span>
<span id="cb39-36"><a href="#cb39-36" aria-hidden="true" tabindex="-1"></a>     <span class="st">"90"</span>,</span>
<span id="cb39-37"><a href="#cb39-37" aria-hidden="true" tabindex="-1"></a>     <span class="st">"4"</span>,</span>
<span id="cb39-38"><a href="#cb39-38" aria-hidden="true" tabindex="-1"></a>     <span class="st">"44"</span>,</span>
<span id="cb39-39"><a href="#cb39-39" aria-hidden="true" tabindex="-1"></a>     <span class="st">"44"</span>,</span>
<span id="cb39-40"><a href="#cb39-40" aria-hidden="true" tabindex="-1"></a>     <span class="st">"70"</span>,</span>
<span id="cb39-41"><a href="#cb39-41" aria-hidden="true" tabindex="-1"></a>     <span class="st">"64"</span>,</span>
<span id="cb39-42"><a href="#cb39-42" aria-hidden="true" tabindex="-1"></a>     <span class="st">"18"</span>,</span>
<span id="cb39-43"><a href="#cb39-43" aria-hidden="true" tabindex="-1"></a>     <span class="st">"3"</span></span>
<span id="cb39-44"><a href="#cb39-44" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb39-45"><a href="#cb39-45" aria-hidden="true" tabindex="-1"></a> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">`</span><span class="at">Resolution (cm per Pixel)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Res"</span>,</span>
<span id="cb39-46"><a href="#cb39-46" aria-hidden="true" tabindex="-1"></a>               <span class="st">`</span><span class="at">Data Source</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"DataSource"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-47"><a href="#cb39-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>()<span class="sc">%&gt;%</span></span>
<span id="cb39-48"><a href="#cb39-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">align</span>(<span class="at">align =</span> <span class="st">"center"</span>, <span class="at">part =</span> <span class="st">"all"</span>) <span class="sc">%&gt;%</span> <span class="co"># Align text to center</span></span>
<span id="cb39-49"><a href="#cb39-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_table_properties</span>(<span class="at">layout =</span> <span class="st">"autofit"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div><div id="tbl-IGNimg" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-IGNimg-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;4: Images used to assess the historical presence of Gracilaria vermiculophylla in the Belon esturay. Images from the IGN data source have been retrieved from the “Remonter Le Temps” plateform <span class="citation" data-cites="RemonterLeTempsIGN">(<a href="#ref-RemonterLeTempsIGN" role="doc-biblioref">IGN, 2024</a>)</span>. Drone flight have been performed by the team using a Mavic 3 Entreprise.
</figcaption>
<div aria-describedby="tbl-IGNimg-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid"><style>.cl-e7cc4476{table-layout:auto;}.cl-e7c39420{font-family:'Arial';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-e7c72fb8{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-e7c7441c{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e7c7441d{background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e7c74426{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-e7cc4476"><thead><tr style="overflow-wrap:break-word;"><th class="cl-e7c7441c"><p class="cl-e7c72fb8"><span class="cl-e7c39420">Date</span></p></th><th class="cl-e7c7441c"><p class="cl-e7c72fb8"><span class="cl-e7c39420">Type</span></p></th><th class="cl-e7c7441c"><p class="cl-e7c72fb8"><span class="cl-e7c39420">Data Source</span></p></th><th class="cl-e7c7441c"><p class="cl-e7c72fb8"><span class="cl-e7c39420">Resolution (cm per Pixel)</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">1952-04-26</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">Black and White</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">IGN</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">10</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">1958-04-22</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">Black and White</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">IGN</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">90</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">1976-07-? </span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">Black and White</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">IGN</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">1978-08-22</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">Black and White</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">IGN</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">44</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">1982-08-11</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">Black and White</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">IGN</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">44</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">1992-05-17</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">True Color</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">IGN</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">70</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">1997-04-11</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">Black and White</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">IGN</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">64</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">2012-07-24</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">True Color</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">IGN</span></p></td><td class="cl-e7c7441d"><p class="cl-e7c72fb8"><span class="cl-e7c39420">18</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-e7c74426"><p class="cl-e7c72fb8"><span class="cl-e7c39420">2024-04-11</span></p></td><td class="cl-e7c74426"><p class="cl-e7c72fb8"><span class="cl-e7c39420">True Color</span></p></td><td class="cl-e7c74426"><p class="cl-e7c72fb8"><span class="cl-e7c39420">Drone Flight</span></p></td><td class="cl-e7c74426"><p class="cl-e7c72fb8"><span class="cl-e7c39420">3</span></p></td></tr></tbody></table></div>
</div>
</div>
</figure>
</div></div></div>
</div></div>
<p>For each date, polygons have been drawn around <em>G. vermiculophylla</em> patches by visually photo-interpreting on each images. These polygons were used to calculate the total area of the mudflat covered by macroalgae within a common extent of 30 hectares in Pont de Guilly, located in the Bélon Estuary, South Brittany, France.</p>
</section>
</section>
<section id="results" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Results</h1>
<section id="historical-records-in-the-belon-estuary" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="historical-records-in-the-belon-estuary"><span class="header-section-number">4.1</span> Historical records in the Belon estuary</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [26]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyterra)</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>img_list <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"Data/RemonterLeTemps/img"</span>, <span class="at">full.names =</span> T, <span class="at">include.dirs =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">path =</span><span class="st">"value"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(path, <span class="st">"RAW"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">filename =</span> <span class="fu">gsub</span>(<span class="st">".*/"</span>,<span class="st">""</span>,path),</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">source =</span> <span class="fu">case_when</span>(<span class="fu">str_detect</span>(filename, <span class="st">"IGNF"</span>) <span class="sc">~</span> <span class="st">"IGN"</span>,</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>                          T <span class="sc">~</span> <span class="st">"Other"</span>),</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">year =</span> <span class="fu">as.integer</span>(<span class="fu">case_when</span>(source <span class="sc">==</span> <span class="st">"IGN"</span> <span class="sc">~</span> <span class="fu">substr</span>(filename,<span class="dv">15</span>,<span class="dv">18</span>),</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>                          T <span class="sc">~</span> <span class="fu">substr</span>(filename,<span class="dv">1</span>,<span class="dv">4</span>))))</span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>mask <span class="ot">&lt;-</span> <span class="st">"Data/RemonterLeTemps/shp/mask/IntertidalMask_Belon_RemonterLeTemps.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>Pres_algae <span class="ot">&lt;-</span> <span class="st">"Data/RemonterLeTemps/shp/shape_red_algae/Presence_Absence_RedAlgae_Date.shp"</span> <span class="sc">%&gt;%</span> <span class="fu">read_sf</span>()</span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a>Text_position <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> <span class="fu">rep</span>(<span class="fu">ext</span>(mask)[<span class="dv">1</span>]<span class="sc">+</span><span class="fl">0.9</span><span class="sc">*</span>(<span class="fu">ext</span>(mask)[<span class="dv">2</span>]<span class="sc">-</span><span class="fu">ext</span>(mask)[<span class="dv">1</span>]),<span class="fu">nrow</span>(img_list)),</span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a>                            <span class="at">y =</span> <span class="fu">rep</span>(<span class="fu">ext</span>(mask)[<span class="dv">3</span>]<span class="sc">+</span><span class="fl">0.9</span><span class="sc">*</span>(<span class="fu">ext</span>(mask)[<span class="dv">4</span>]<span class="sc">-</span><span class="fu">ext</span>(mask)[<span class="dv">3</span>]),<span class="fu">nrow</span>(img_list)),</span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a>                            <span class="at">year =</span> img_list<span class="sc">$</span>year)</span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(img_list)){</span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a>  img <span class="ot">&lt;-</span> <span class="fu">rast</span>(img_list<span class="sc">$</span>path[i]) <span class="sc">%&gt;%</span> </span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">project</span>(<span class="fu">crs</span>(mask)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">crop</span>(mask, <span class="at">mask =</span> T)</span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a>  date_i <span class="ot">&lt;-</span> Text_position[i,]</span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># shp_i &lt;- Pres_algae %&gt;% </span></span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   dplyr::filter(date == img_list$year[i])</span></span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># </span></span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># if(img_list$year[i] == 2024){</span></span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   shp_i &lt;- shp_i %&gt;% </span></span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a>  <span class="co">#     st_buffer(dist = 0.3) %&gt;% </span></span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a>  <span class="co">#     st_union()</span></span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># }</span></span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># </span></span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a>Plot <span class="ot">&lt;-</span>  </span>
<span id="cb40-46"><a href="#cb40-46" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nlyr</span>(img) <span class="sc">!=</span> <span class="dv">1</span>){</span>
<span id="cb40-47"><a href="#cb40-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">RGB</span>(img)<span class="ot">&lt;-</span><span class="dv">1</span><span class="sc">:</span><span class="dv">3</span></span>
<span id="cb40-48"><a href="#cb40-48" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-49"><a href="#cb40-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb40-50"><a href="#cb40-50" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_spatraster_rgb</span>(<span class="at">data =</span> img)</span>
<span id="cb40-51"><a href="#cb40-51" aria-hidden="true" tabindex="-1"></a>  }<span class="cf">else</span>{</span>
<span id="cb40-52"><a href="#cb40-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(img) <span class="ot">=</span> <span class="st">"layer"</span></span>
<span id="cb40-53"><a href="#cb40-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb40-54"><a href="#cb40-54" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_spatraster</span>(img, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">fill =</span> layer), <span class="at">show.legend =</span> F)<span class="sc">+</span></span>
<span id="cb40-55"><a href="#cb40-55" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">"black"</span>, <span class="at">high =</span> <span class="st">"white"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">240</span>),<span class="at">na.value =</span> <span class="cn">NA</span>, <span class="at">name =</span> <span class="st">"Greyscale"</span>)</span>
<span id="cb40-56"><a href="#cb40-56" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-57"><a href="#cb40-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-58"><a href="#cb40-58" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> Plot <span class="sc">+</span></span>
<span id="cb40-59"><a href="#cb40-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> date_i, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> year),</span>
<span id="cb40-60"><a href="#cb40-60" aria-hidden="true" tabindex="-1"></a>              <span class="at">size =</span> <span class="dv">7</span>)<span class="sc">+</span></span>
<span id="cb40-61"><a href="#cb40-61" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_sf(data = shp_i, fill = NA, color = "darkred")+ </span></span>
<span id="cb40-62"><a href="#cb40-62" aria-hidden="true" tabindex="-1"></a>   <span class="fu">coord_sf</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="fu">ext</span>(mask)[<span class="dv">1</span>],<span class="fu">ext</span>(mask)[<span class="dv">2</span>]),</span>
<span id="cb40-63"><a href="#cb40-63" aria-hidden="true" tabindex="-1"></a>            <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fu">ext</span>(mask)[<span class="dv">3</span>],<span class="fu">ext</span>(mask)[<span class="dv">4</span>]),</span>
<span id="cb40-64"><a href="#cb40-64" aria-hidden="true" tabindex="-1"></a>            <span class="at">expand =</span> F)<span class="sc">+</span></span>
<span id="cb40-65"><a href="#cb40-65" aria-hidden="true" tabindex="-1"></a>    <span class="co"># geom_text(data = date_i, aes(x = x, y = y, label = year),</span></span>
<span id="cb40-66"><a href="#cb40-66" aria-hidden="true" tabindex="-1"></a>    <span class="co">#           size = 10)+</span></span>
<span id="cb40-67"><a href="#cb40-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_Bede</span>()<span class="sc">+</span></span>
<span id="cb40-68"><a href="#cb40-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-69"><a href="#cb40-69" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-70"><a href="#cb40-70" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-71"><a href="#cb40-71" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-72"><a href="#cb40-72" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-73"><a href="#cb40-73" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-74"><a href="#cb40-74" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.clip =</span> <span class="st">"inherit"</span>,</span>
<span id="cb40-75"><a href="#cb40-75" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">0</span>,<span class="st">"pt"</span>), </span>
<span id="cb40-76"><a href="#cb40-76" aria-hidden="true" tabindex="-1"></a>          <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">"lines"</span>))<span class="sc">+</span></span>
<span id="cb40-77"><a href="#cb40-77" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(img_list<span class="sc">$</span>year[i] <span class="sc">==</span> <span class="dv">2024</span>){</span>
<span id="cb40-78"><a href="#cb40-78" aria-hidden="true" tabindex="-1"></a>    ggspatial<span class="sc">::</span><span class="fu">annotation_scale</span>(<span class="at">location =</span> <span class="st">"bl"</span>,</span>
<span id="cb40-79"><a href="#cb40-79" aria-hidden="true" tabindex="-1"></a>                                <span class="at">pad_y =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">"cm"</span>),</span>
<span id="cb40-80"><a href="#cb40-80" aria-hidden="true" tabindex="-1"></a>                                <span class="at">text_cex=</span><span class="dv">1</span>)</span>
<span id="cb40-81"><a href="#cb40-81" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb40-82"><a href="#cb40-82" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-83"><a href="#cb40-83" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-84"><a href="#cb40-84" aria-hidden="true" tabindex="-1"></a>a </span>
<span id="cb40-85"><a href="#cb40-85" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-86"><a href="#cb40-86" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste0</span>(<span class="st">"plot_"</span>,<span class="fu">as.character</span>(img_list<span class="sc">$</span>year[i])),a)</span>
<span id="cb40-87"><a href="#cb40-87" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-88"><a href="#cb40-88" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-89"><a href="#cb40-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-90"><a href="#cb40-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-91"><a href="#cb40-91" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_all &lt;- (plot_1952 + plot_1958)/</span></span>
<span id="cb40-92"><a href="#cb40-92" aria-hidden="true" tabindex="-1"></a><span class="co"># (plot_1976 + plot_1978)/</span></span>
<span id="cb40-93"><a href="#cb40-93" aria-hidden="true" tabindex="-1"></a><span class="co"># (plot_1982 + plot_1992)/</span></span>
<span id="cb40-94"><a href="#cb40-94" aria-hidden="true" tabindex="-1"></a><span class="co"># (plot_1997 + plot_2012)/</span></span>
<span id="cb40-95"><a href="#cb40-95" aria-hidden="true" tabindex="-1"></a><span class="co"># (plot_2019 + plot_2024)</span></span>
<span id="cb40-96"><a href="#cb40-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-97"><a href="#cb40-97" aria-hidden="true" tabindex="-1"></a>plot_all <span class="ot">&lt;-</span> (plot_1952 <span class="sc">+</span> plot_1958 <span class="sc">+</span> plot_1976)<span class="sc">/</span></span>
<span id="cb40-98"><a href="#cb40-98" aria-hidden="true" tabindex="-1"></a>(plot_1978 <span class="sc">+</span> plot_1982 <span class="sc">+</span> plot_1992)<span class="sc">/</span></span>
<span id="cb40-99"><a href="#cb40-99" aria-hidden="true" tabindex="-1"></a>(plot_1997 <span class="sc">+</span> plot_2012 <span class="sc">+</span> plot_2024)</span>
<span id="cb40-100"><a href="#cb40-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-101"><a href="#cb40-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-102"><a href="#cb40-102" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Historical_maps.png"</span>,plot_all, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="fl">4.52</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span>
<span id="cb40-103"><a href="#cb40-103" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Low_res/Historical_maps.png"</span>,plot_all, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="fl">4.52</span>, <span class="at">dpi =</span> <span class="dv">200</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [27]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Utilities.Package)</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>mask <span class="ot">&lt;-</span> <span class="st">"Data/RemonterLeTemps/shp/mask/IntertidalMask_Belon_RemonterLeTemps.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">area =</span> <span class="fu">st_area</span>(.)) <span class="sc">%&gt;%</span> </span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(<span class="fu">as.numeric</span>(area)[<span class="dv">1</span>])</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>Pres_algae <span class="ot">&lt;-</span> <span class="st">"Data/RemonterLeTemps/shp/shape_red_algae/Presence_Absence_RedAlgae_Date.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">area =</span> <span class="fu">st_area</span>(.)) <span class="sc">%&gt;%</span> </span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(date) <span class="sc">%&gt;%</span> </span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">area =</span> <span class="fu">sum</span> (area)) <span class="sc">%&gt;%</span> </span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">prop =</span> <span class="fu">as.numeric</span>((area<span class="sc">/</span>mask)<span class="sc">*</span><span class="dv">100</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>area)</span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">date =</span> <span class="fu">c</span>(<span class="dv">1952</span>,<span class="dv">1958</span>),</span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a>           <span class="at">prop =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(Pres_algae) <span class="sc">%&gt;%</span> </span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> date, <span class="at">y =</span> prop))<span class="sc">+</span></span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_bar(stat = "identity",size = 10 )+</span></span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Year"</span>)<span class="sc">+</span></span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Cover of Gracilaria (%)"</span>)<span class="sc">+</span></span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_line()+</span></span>
<span id="cb41-28"><a href="#cb41-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">1971</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">linewidth=</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb41-29"><a href="#cb41-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">1994</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"goldenrod"</span>, <span class="at">linewidth=</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb41-30"><a href="#cb41-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_rug</span>(<span class="at">sides =</span> <span class="st">"b"</span>) <span class="sc">+</span>  <span class="co"># Add ticks at the bottom for each x-value</span></span>
<span id="cb41-31"><a href="#cb41-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb41-32"><a href="#cb41-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">1970</span>, </span>
<span id="cb41-33"><a href="#cb41-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="dv">30</span>, </span>
<span id="cb41-34"><a href="#cb41-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"First introduction of "</span>, <span class="fu">italic</span>(<span class="st">"Crassostrea gigas"</span>))), </span>
<span id="cb41-35"><a href="#cb41-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span></span>
<span id="cb41-36"><a href="#cb41-36" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb41-37"><a href="#cb41-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb41-38"><a href="#cb41-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">1972</span>, </span>
<span id="cb41-39"><a href="#cb41-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="dv">30</span>, </span>
<span id="cb41-40"><a href="#cb41-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"in France"</span>)), </span>
<span id="cb41-41"><a href="#cb41-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span></span>
<span id="cb41-42"><a href="#cb41-42" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb41-43"><a href="#cb41-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb41-44"><a href="#cb41-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">1993</span>, </span>
<span id="cb41-45"><a href="#cb41-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="dv">30</span>, </span>
<span id="cb41-46"><a href="#cb41-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"First record of "</span>, <span class="fu">italic</span>(<span class="st">"Gracilaria vermiculophylla"</span>))), </span>
<span id="cb41-47"><a href="#cb41-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span></span>
<span id="cb41-48"><a href="#cb41-48" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb41-49"><a href="#cb41-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb41-50"><a href="#cb41-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">1995</span>, </span>
<span id="cb41-51"><a href="#cb41-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="dv">30</span>, </span>
<span id="cb41-52"><a href="#cb41-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"in the Belon Estuary"</span>)), </span>
<span id="cb41-53"><a href="#cb41-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span></span>
<span id="cb41-54"><a href="#cb41-54" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb41-55"><a href="#cb41-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede</span>()<span class="sc">+</span></span>
<span id="cb41-56"><a href="#cb41-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb41-57"><a href="#cb41-57" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb41-58"><a href="#cb41-58" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb41-59"><a href="#cb41-59" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>), </span>
<span id="cb41-60"><a href="#cb41-60" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>))</span>
<span id="cb41-61"><a href="#cb41-61" aria-hidden="true" tabindex="-1"></a> <span class="co"># geom_line()</span></span>
<span id="cb41-62"><a href="#cb41-62" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Cover_Gracillaria_vs_Time.png"</span>, plot, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span>
<span id="cb41-63"><a href="#cb41-63" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Low_res/Cover_Gracillaria_vs_Time.png"</span>, plot, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">dpi =</span> <span class="dv">200</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>A clear shift in sediment coloration over the paste 70 years were observed, closely aligned with the subsequent proliferation of the invasive red macroalga <em>Gracilaria vermiculophylla</em> (<a href="#fig-HistoricalMap" class="quarto-xref">Figure&nbsp;3</a>). Before 1976, the sediments appeared relatively light, indicating no detectable presence of this species. Following its initial appearance in 1976, subtle darkening of the sediment became discernible, coinciding with the early establishment of <em>G. vermiculophylla</em>. During the subsequent decades, the late 1970s through the 1990s, this darkening trend became more pronounced and widespread, reflecting an increasing spatial coverage and biomass of the algae. By the early 2000s, and especially by 2024, the sediment exhibited consistently darker tones, indicative of extensive and persistent colonization by <em>G. vermiculophylla</em>.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [28]:</pre></div><div id="cell-fig-HistoricalMap" class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"./Figures/Low_res/Historical_maps.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-HistoricalMap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-HistoricalMap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="./Figures/Low_res/Historical_maps.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;3: Historical images of Pont du Guilly between 1952 and 2024."><img src="./Figures/Low_res/Historical_maps.png" class="img-fluid figure-img" style="width:95.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-HistoricalMap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Historical images of Pont du Guilly between 1952 and 2024.
</figcaption>
</figure>
</div>
</div>
</div></div>
<p>From the early recordings in the 1950s through the late 1970s, <em>Gracilaria vermiculophylla</em> coverage remained effectively at 0% (<a href="#fig-HistoricalPlot" class="quarto-xref">Figure&nbsp;4</a>). Shortly after the introduction of <em>Crassostrea gigas</em> in the estuary (see vertical red dashed line in the figure), the first detectable presence of <em>G. vermiculophylla</em> emerged. By 1976, it covered 2.5% (0.7 ha) of the Pont du Guilly area, and by 1978 it had increased slightly to 3.0% (0.9 ha). From 1982 onward, coverage expanded more rapidly, increasing from 6.6% (2.0 ha) in 1982 to 14.7% (4.5 ha) in 1992 and nearly 30% (9.0 ha) by 1997. This upward trend continued into the 21st century, peaking at 43.8% (13.3 ha) in 2012. Although coverage fluctuated somewhat thereafter (40.6% in 2019 and 40.2% in 2024), it remained consistently high, indicating sustained and widespread colonization.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [29]:</pre></div><div id="cell-fig-HistoricalPlot" class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"./Figures/Low_res/Cover_Gracillaria_vs_Time.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-HistoricalPlot" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-HistoricalPlot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="./Figures/Low_res/Cover_Gracillaria_vs_Time.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;4: Evolution of the Gracilaria vermiculophylla cover at Pont du Guilly in the Belon Estuary. The red vertical line indicates the date of Crassostrea gigas introduction in France [@grizel1991introduction], while the golden line represents the date of the first documented mention of Gracilaria vermiculophylla invasion in France in the literature [@rueness2005life]."><img src="./Figures/Low_res/Cover_Gracillaria_vs_Time.png" class="img-fluid figure-img" style="width:95.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-HistoricalPlot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Evolution of the Gracilaria vermiculophylla cover at Pont du Guilly in the Belon Estuary. The red vertical line indicates the date of Crassostrea gigas introduction in France <span class="citation" data-cites="grizel1991introduction">(<a href="#ref-grizel1991introduction" role="doc-biblioref">Grizel and Heral, 1991</a>)</span>, while the golden line represents the date of the first documented mention of Gracilaria vermiculophylla invasion in France in the literature <span class="citation" data-cites="rueness2005life">(<a href="#ref-rueness2005life" role="doc-biblioref">Rueness, 2005</a>)</span>.
</figcaption>
</figure>
</div>
</div>
</div></div>
</section>
<section id="spectral-description" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="spectral-description"><span class="header-section-number">4.2</span> Spectral description</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [30]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(signal)</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Utilities.Package)</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a> <span class="co"># Set the path to the directory containing the CSV files</span></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>csv_directory <span class="ot">&lt;-</span> <span class="st">"Data/ASD"</span></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a><span class="co"># List all CSV files in the directory</span></span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>csv_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> csv_directory, <span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.csv$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Read and combine all CSV files into one dataframe</span></span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> csv_files <span class="sc">%&gt;%</span></span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(read.csv) <span class="sc">%&gt;%</span>  <span class="co"># Read each CSV file into a list of dataframes</span></span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span>         <span class="co"># Combine all dataframes into one</span></span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Letter <span class="sc">==</span> <span class="st">"Field"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(Wavelength, mean_RAW, ID)) <span class="sc">%&gt;%</span> </span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> ID, <span class="at">values_from =</span> mean_RAW)</span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a><span class="co"># User-defined parameters</span></span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">3</span>      <span class="co"># polynomial order for Savitzky-Golay</span></span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">11</span>     <span class="co"># frame length (odd number, e.g. 11)</span></span>
<span id="cb44-26"><a href="#cb44-26" aria-hidden="true" tabindex="-1"></a>factor <span class="ot">&lt;-</span> <span class="dv">5</span> <span class="co"># degrade resolution by a factor of 5</span></span>
<span id="cb44-27"><a href="#cb44-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-28"><a href="#cb44-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming your initial dataframe is called df</span></span>
<span id="cb44-29"><a href="#cb44-29" aria-hidden="true" tabindex="-1"></a><span class="co"># df: first column is wavelength, subsequent columns are spectra</span></span>
<span id="cb44-30"><a href="#cb44-30" aria-hidden="true" tabindex="-1"></a>wavelength <span class="ot">&lt;-</span> df[[<span class="dv">1</span>]]</span>
<span id="cb44-31"><a href="#cb44-31" aria-hidden="true" tabindex="-1"></a>reflectances <span class="ot">&lt;-</span> df[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb44-32"><a href="#cb44-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-33"><a href="#cb44-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute original wavelength spacing</span></span>
<span id="cb44-34"><a href="#cb44-34" aria-hidden="true" tabindex="-1"></a>d_lambda <span class="ot">&lt;-</span> wavelength[<span class="dv">2</span>] <span class="sc">-</span> wavelength[<span class="dv">1</span>]</span>
<span id="cb44-35"><a href="#cb44-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-36"><a href="#cb44-36" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Degrade the spectral resolution</span></span>
<span id="cb44-37"><a href="#cb44-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a new wavelength vector with coarser spacing</span></span>
<span id="cb44-38"><a href="#cb44-38" aria-hidden="true" tabindex="-1"></a>new_wavelength <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span> <span class="fu">min</span>(wavelength), <span class="at">to =</span> <span class="fu">max</span>(wavelength), <span class="at">by =</span> factor <span class="sc">*</span> d_lambda)</span>
<span id="cb44-39"><a href="#cb44-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-40"><a href="#cb44-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Interpolate the spectra onto the new wavelength grid</span></span>
<span id="cb44-41"><a href="#cb44-41" aria-hidden="true" tabindex="-1"></a>degraded_spectra <span class="ot">&lt;-</span> <span class="fu">apply</span>(reflectances, <span class="dv">2</span>, <span class="cf">function</span>(spec) {</span>
<span id="cb44-42"><a href="#cb44-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">approx</span>(<span class="at">x =</span> wavelength, <span class="at">y =</span> spec, <span class="at">xout =</span> new_wavelength)<span class="sc">$</span>y</span>
<span id="cb44-43"><a href="#cb44-43" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb44-44"><a href="#cb44-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-45"><a href="#cb44-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Now our working wavelength and reflectances become:</span></span>
<span id="cb44-46"><a href="#cb44-46" aria-hidden="true" tabindex="-1"></a>wavelength <span class="ot">&lt;-</span> new_wavelength</span>
<span id="cb44-47"><a href="#cb44-47" aria-hidden="true" tabindex="-1"></a>reflectances <span class="ot">&lt;-</span> degraded_spectra</span>
<span id="cb44-48"><a href="#cb44-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-49"><a href="#cb44-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Update the spacing to reflect the new degraded resolution</span></span>
<span id="cb44-50"><a href="#cb44-50" aria-hidden="true" tabindex="-1"></a>d_lambda <span class="ot">&lt;-</span> wavelength[<span class="dv">2</span>] <span class="sc">-</span> wavelength[<span class="dv">1</span>]</span>
<span id="cb44-51"><a href="#cb44-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-52"><a href="#cb44-52" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Perform initial Savitzky-Golay smoothing on each degraded-resolution spectrum</span></span>
<span id="cb44-53"><a href="#cb44-53" aria-hidden="true" tabindex="-1"></a>smooth_spectra <span class="ot">&lt;-</span> <span class="fu">apply</span>(reflectances, <span class="dv">2</span>, sgolayfilt, <span class="at">p =</span> p, <span class="at">n =</span> n)</span>
<span id="cb44-54"><a href="#cb44-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-55"><a href="#cb44-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a function for first derivative using central differences</span></span>
<span id="cb44-56"><a href="#cb44-56" aria-hidden="true" tabindex="-1"></a>first_derivative <span class="ot">&lt;-</span> <span class="cf">function</span>(x, d) {</span>
<span id="cb44-57"><a href="#cb44-57" aria-hidden="true" tabindex="-1"></a>  n <span class="ot">&lt;-</span> <span class="fu">length</span>(x)</span>
<span id="cb44-58"><a href="#cb44-58" aria-hidden="true" tabindex="-1"></a>  d1 <span class="ot">&lt;-</span> <span class="fu">numeric</span>(n)</span>
<span id="cb44-59"><a href="#cb44-59" aria-hidden="true" tabindex="-1"></a>  d1[<span class="dv">2</span><span class="sc">:</span>(n<span class="dv">-1</span>)] <span class="ot">&lt;-</span> (x[<span class="dv">3</span><span class="sc">:</span>n] <span class="sc">-</span> x[<span class="dv">1</span><span class="sc">:</span>(n<span class="dv">-2</span>)]) <span class="sc">/</span> (<span class="dv">2</span><span class="sc">*</span>d)</span>
<span id="cb44-60"><a href="#cb44-60" aria-hidden="true" tabindex="-1"></a>  d1[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb44-61"><a href="#cb44-61" aria-hidden="true" tabindex="-1"></a>  d1[n] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb44-62"><a href="#cb44-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(d1)</span>
<span id="cb44-63"><a href="#cb44-63" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-64"><a href="#cb44-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-65"><a href="#cb44-65" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. Compute the first derivative from the smoothed spectra</span></span>
<span id="cb44-66"><a href="#cb44-66" aria-hidden="true" tabindex="-1"></a>d1_spectra <span class="ot">&lt;-</span> <span class="fu">apply</span>(smooth_spectra, <span class="dv">2</span>, first_derivative, <span class="at">d =</span> d_lambda)</span>
<span id="cb44-67"><a href="#cb44-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-68"><a href="#cb44-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Helper function to safely smooth a vector that may have NAs at boundaries</span></span>
<span id="cb44-69"><a href="#cb44-69" aria-hidden="true" tabindex="-1"></a>safe_sgolay <span class="ot">&lt;-</span> <span class="cf">function</span>(x, p, n) {</span>
<span id="cb44-70"><a href="#cb44-70" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Replace NAs at edges with nearest non-NA values before smoothing</span></span>
<span id="cb44-71"><a href="#cb44-71" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.na</span>(x[<span class="dv">1</span>])) {</span>
<span id="cb44-72"><a href="#cb44-72" aria-hidden="true" tabindex="-1"></a>    first_valid <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(x))[<span class="dv">1</span>]</span>
<span id="cb44-73"><a href="#cb44-73" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">1</span><span class="sc">:</span>(first_valid<span class="dv">-1</span>)] <span class="ot">&lt;-</span> x[first_valid]</span>
<span id="cb44-74"><a href="#cb44-74" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb44-75"><a href="#cb44-75" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.na</span>(x[<span class="fu">length</span>(x)])) {</span>
<span id="cb44-76"><a href="#cb44-76" aria-hidden="true" tabindex="-1"></a>    last_valid <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(x)))[<span class="dv">1</span>]</span>
<span id="cb44-77"><a href="#cb44-77" aria-hidden="true" tabindex="-1"></a>    x[(last_valid<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(x)] <span class="ot">&lt;-</span> x[last_valid]</span>
<span id="cb44-78"><a href="#cb44-78" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb44-79"><a href="#cb44-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sgolayfilt</span>(x, <span class="at">p =</span> p, <span class="at">n =</span> n)</span>
<span id="cb44-80"><a href="#cb44-80" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-81"><a href="#cb44-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-82"><a href="#cb44-82" aria-hidden="true" tabindex="-1"></a><span class="co"># 4. Smooth the first derivative again using Savitzky-Golay</span></span>
<span id="cb44-83"><a href="#cb44-83" aria-hidden="true" tabindex="-1"></a>d1_spectra_smooth <span class="ot">&lt;-</span> <span class="fu">apply</span>(d1_spectra, <span class="dv">2</span>, safe_sgolay, <span class="at">p =</span> p, <span class="at">n =</span> n)</span>
<span id="cb44-84"><a href="#cb44-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-85"><a href="#cb44-85" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a function for second derivative using central differences</span></span>
<span id="cb44-86"><a href="#cb44-86" aria-hidden="true" tabindex="-1"></a>second_derivative <span class="ot">&lt;-</span> <span class="cf">function</span>(x, d) {</span>
<span id="cb44-87"><a href="#cb44-87" aria-hidden="true" tabindex="-1"></a>  n <span class="ot">&lt;-</span> <span class="fu">length</span>(x)</span>
<span id="cb44-88"><a href="#cb44-88" aria-hidden="true" tabindex="-1"></a>  d2 <span class="ot">&lt;-</span> <span class="fu">numeric</span>(n)</span>
<span id="cb44-89"><a href="#cb44-89" aria-hidden="true" tabindex="-1"></a>  d2[<span class="dv">2</span><span class="sc">:</span>(n<span class="dv">-1</span>)] <span class="ot">&lt;-</span> (x[<span class="dv">3</span><span class="sc">:</span>n] <span class="sc">-</span> <span class="dv">2</span><span class="sc">*</span>x[<span class="dv">2</span><span class="sc">:</span>(n<span class="dv">-1</span>)] <span class="sc">+</span> x[<span class="dv">1</span><span class="sc">:</span>(n<span class="dv">-2</span>)]) <span class="sc">/</span> (d<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb44-90"><a href="#cb44-90" aria-hidden="true" tabindex="-1"></a>  d2[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb44-91"><a href="#cb44-91" aria-hidden="true" tabindex="-1"></a>  d2[n] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb44-92"><a href="#cb44-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(d2)</span>
<span id="cb44-93"><a href="#cb44-93" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-94"><a href="#cb44-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-95"><a href="#cb44-95" aria-hidden="true" tabindex="-1"></a><span class="co"># 5. Compute the second derivative from the smoothed first derivative</span></span>
<span id="cb44-96"><a href="#cb44-96" aria-hidden="true" tabindex="-1"></a>d2_spectra <span class="ot">&lt;-</span> <span class="fu">apply</span>(d1_spectra_smooth, <span class="dv">2</span>, second_derivative, <span class="at">d =</span> d_lambda)</span>
<span id="cb44-97"><a href="#cb44-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-98"><a href="#cb44-98" aria-hidden="true" tabindex="-1"></a><span class="co"># 6. Finally, smooth the second derivative again using Savitzky-Golay</span></span>
<span id="cb44-99"><a href="#cb44-99" aria-hidden="true" tabindex="-1"></a>d2_spectra_smooth <span class="ot">&lt;-</span> <span class="fu">apply</span>(d2_spectra, <span class="dv">2</span>, safe_sgolay, <span class="at">p =</span> p, <span class="at">n =</span> n)</span>
<span id="cb44-100"><a href="#cb44-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-101"><a href="#cb44-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results into a final dataframe</span></span>
<span id="cb44-102"><a href="#cb44-102" aria-hidden="true" tabindex="-1"></a>result_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">wavelength =</span> wavelength, d2_spectra_smooth) <span class="sc">%&gt;%</span> </span>
<span id="cb44-103"><a href="#cb44-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb44-104"><a href="#cb44-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>wavelength, <span class="at">names_to =</span> <span class="st">"spectra"</span>, <span class="at">values_to =</span> <span class="st">"deriv"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-105"><a href="#cb44-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(wavelength) <span class="sc">%&gt;%</span> </span>
<span id="cb44-106"><a href="#cb44-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">mean =</span> <span class="fu">mean</span>(deriv, <span class="at">na.rm =</span> T),</span>
<span id="cb44-107"><a href="#cb44-107" aria-hidden="true" tabindex="-1"></a>          <span class="at">sd =</span> <span class="fu">sd</span>(deriv, <span class="at">na.rm=</span>T))</span>
<span id="cb44-108"><a href="#cb44-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-109"><a href="#cb44-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-110"><a href="#cb44-110" aria-hidden="true" tabindex="-1"></a>df_spectra <span class="ot">&lt;-</span> csv_files <span class="sc">%&gt;%</span></span>
<span id="cb44-111"><a href="#cb44-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(read.csv) <span class="sc">%&gt;%</span>  <span class="co"># Read each CSV file into a list of dataframes</span></span>
<span id="cb44-112"><a href="#cb44-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span>         <span class="co"># Combine all dataframes into one</span></span>
<span id="cb44-113"><a href="#cb44-113" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Letter <span class="sc">==</span> <span class="st">"Field"</span>)<span class="sc">%&gt;%</span> </span>
<span id="cb44-114"><a href="#cb44-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Wavelength) <span class="sc">%&gt;%</span> </span>
<span id="cb44-115"><a href="#cb44-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">mean =</span> <span class="fu">mean</span>(mean_RAW, <span class="at">na.rm =</span> T),</span>
<span id="cb44-116"><a href="#cb44-116" aria-hidden="true" tabindex="-1"></a>          <span class="at">sd =</span> <span class="fu">sd</span>(mean_RAW, <span class="at">na.rm=</span>T))</span>
<span id="cb44-117"><a href="#cb44-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-118"><a href="#cb44-118" aria-hidden="true" tabindex="-1"></a>chla<span class="ot">=</span> <span class="dv">678</span></span>
<span id="cb44-119"><a href="#cb44-119" aria-hidden="true" tabindex="-1"></a>PE <span class="ot">=</span> <span class="dv">565</span></span>
<span id="cb44-120"><a href="#cb44-120" aria-hidden="true" tabindex="-1"></a>PC <span class="ot">=</span> <span class="dv">620</span></span>
<span id="cb44-121"><a href="#cb44-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-122"><a href="#cb44-122" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(result_df) <span class="sc">+</span></span>
<span id="cb44-123"><a href="#cb44-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> wavelength<span class="sc">+</span><span class="dv">10</span>, <span class="at">y =</span> mean))<span class="sc">+</span></span>
<span id="cb44-124"><a href="#cb44-124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">x =</span> wavelength<span class="sc">+</span><span class="dv">10</span>, <span class="at">y =</span> mean, <span class="at">ymin =</span> mean<span class="sc">-</span>sd, <span class="at">ymax =</span> mean<span class="sc">+</span>sd), <span class="at">alpha =</span> <span class="fl">0.05</span>)<span class="sc">+</span></span>
<span id="cb44-125"><a href="#cb44-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> chla,</span>
<span id="cb44-126"><a href="#cb44-126" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>)<span class="sc">+</span> <span class="co"># chl-a</span></span>
<span id="cb44-127"><a href="#cb44-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> PE,</span>
<span id="cb44-128"><a href="#cb44-128" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"darkgreen"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>)<span class="sc">+</span> </span>
<span id="cb44-129"><a href="#cb44-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> PC,</span>
<span id="cb44-130"><a href="#cb44-130" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"goldenrod"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>)<span class="sc">+</span></span>
<span id="cb44-131"><a href="#cb44-131" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_vline(xintercept = 645, linetype = "dashed")+</span></span>
<span id="cb44-132"><a href="#cb44-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-133"><a href="#cb44-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">400</span>, <span class="dv">700</span>)<span class="sc">+</span></span>
<span id="cb44-134"><a href="#cb44-134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede</span>()<span class="sc">+</span></span>
<span id="cb44-135"><a href="#cb44-135" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">x=</span> <span class="dv">420</span>, <span class="at">y =</span> <span class="fl">10e-06</span>, <span class="at">label =</span> <span class="st">"B"</span>, <span class="at">size =</span> <span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb44-136"><a href="#cb44-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Second Derivative"</span>)<span class="sc">+</span></span>
<span id="cb44-137"><a href="#cb44-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Wavelength"</span>)<span class="sc">+</span></span>
<span id="cb44-138"><a href="#cb44-138" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_text(</span></span>
<span id="cb44-139"><a href="#cb44-139" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   x = 557, </span></span>
<span id="cb44-140"><a href="#cb44-140" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   y = -0.000006, </span></span>
<span id="cb44-141"><a href="#cb44-141" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   label = "Absorption Phycoerythrin", </span></span>
<span id="cb44-142"><a href="#cb44-142" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   angle = 90,</span></span>
<span id="cb44-143"><a href="#cb44-143" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   color = "darkgreen"</span></span>
<span id="cb44-144"><a href="#cb44-144" aria-hidden="true" tabindex="-1"></a>  <span class="co"># )+</span></span>
<span id="cb44-145"><a href="#cb44-145" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_text(</span></span>
<span id="cb44-146"><a href="#cb44-146" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   x = 612, </span></span>
<span id="cb44-147"><a href="#cb44-147" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   y = -0.000006, </span></span>
<span id="cb44-148"><a href="#cb44-148" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   label = "Absorption Phycocyanin", </span></span>
<span id="cb44-149"><a href="#cb44-149" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   angle = 90,</span></span>
<span id="cb44-150"><a href="#cb44-150" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   color = "goldenrod"</span></span>
<span id="cb44-151"><a href="#cb44-151" aria-hidden="true" tabindex="-1"></a>  <span class="co"># )+</span></span>
<span id="cb44-152"><a href="#cb44-152" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_text(</span></span>
<span id="cb44-153"><a href="#cb44-153" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   x = 670, </span></span>
<span id="cb44-154"><a href="#cb44-154" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   y = -0.000006, </span></span>
<span id="cb44-155"><a href="#cb44-155" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   label = "Absorption Chlorophyll-a", </span></span>
<span id="cb44-156"><a href="#cb44-156" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   angle = 90,</span></span>
<span id="cb44-157"><a href="#cb44-157" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   color = "darkred"</span></span>
<span id="cb44-158"><a href="#cb44-158" aria-hidden="true" tabindex="-1"></a>  <span class="co"># )+</span></span>
<span id="cb44-159"><a href="#cb44-159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">position =</span> <span class="st">"right"</span>,</span>
<span id="cb44-160"><a href="#cb44-160" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">label_number</span>(</span>
<span id="cb44-161"><a href="#cb44-161" aria-hidden="true" tabindex="-1"></a>                       <span class="at">suffix =</span> <span class="st">"e-06"</span>, <span class="at">scale =</span> <span class="fl">1e6</span>))<span class="sc">+</span></span>
<span id="cb44-162"><a href="#cb44-162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> ggplot2<span class="sc">::</span><span class="fu">element_line</span>(<span class="at">linetype =</span> <span class="st">"dotted"</span>, </span>
<span id="cb44-163"><a href="#cb44-163" aria-hidden="true" tabindex="-1"></a>      <span class="at">colour =</span> <span class="fu">alpha</span>(<span class="st">"grey30"</span>,<span class="fl">0.5</span>), <span class="at">linewidth =</span> <span class="fl">0.25</span>),</span>
<span id="cb44-164"><a href="#cb44-164" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>)</span>
<span id="cb44-165"><a href="#cb44-165" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb44-166"><a href="#cb44-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-167"><a href="#cb44-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-168"><a href="#cb44-168" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_spectra) <span class="sc">+</span></span>
<span id="cb44-169"><a href="#cb44-169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> Wavelength, <span class="at">y =</span> mean))<span class="sc">+</span></span>
<span id="cb44-170"><a href="#cb44-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">x =</span> Wavelength, <span class="at">y =</span> mean, <span class="at">ymin =</span> mean<span class="sc">-</span>sd, <span class="at">ymax =</span> mean<span class="sc">+</span>sd), <span class="at">alpha =</span> <span class="fl">0.05</span>)<span class="sc">+</span></span>
<span id="cb44-171"><a href="#cb44-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> chla,</span>
<span id="cb44-172"><a href="#cb44-172" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>)<span class="sc">+</span> <span class="co"># chl-a</span></span>
<span id="cb44-173"><a href="#cb44-173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> PE,</span>
<span id="cb44-174"><a href="#cb44-174" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"darkgreen"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>)<span class="sc">+</span> </span>
<span id="cb44-175"><a href="#cb44-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> PC,</span>
<span id="cb44-176"><a href="#cb44-176" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"goldenrod"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>)<span class="sc">+</span></span>
<span id="cb44-177"><a href="#cb44-177" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_vline(xintercept = 645, linetype = "dashed")+</span></span>
<span id="cb44-178"><a href="#cb44-178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Reflectance"</span>)<span class="sc">+</span></span>
<span id="cb44-179"><a href="#cb44-179" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb44-180"><a href="#cb44-180" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">552</span>, </span>
<span id="cb44-181"><a href="#cb44-181" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fl">0.17</span>, </span>
<span id="cb44-182"><a href="#cb44-182" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Absorption Phycoerythrin"</span>, </span>
<span id="cb44-183"><a href="#cb44-183" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>,</span>
<span id="cb44-184"><a href="#cb44-184" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"darkgreen"</span></span>
<span id="cb44-185"><a href="#cb44-185" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb44-186"><a href="#cb44-186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb44-187"><a href="#cb44-187" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">607</span>, </span>
<span id="cb44-188"><a href="#cb44-188" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fl">0.17</span>, </span>
<span id="cb44-189"><a href="#cb44-189" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Absorption Phycocyanin"</span>, </span>
<span id="cb44-190"><a href="#cb44-190" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>,</span>
<span id="cb44-191"><a href="#cb44-191" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"goldenrod"</span></span>
<span id="cb44-192"><a href="#cb44-192" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb44-193"><a href="#cb44-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb44-194"><a href="#cb44-194" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">665</span>, </span>
<span id="cb44-195"><a href="#cb44-195" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fl">0.17</span>, </span>
<span id="cb44-196"><a href="#cb44-196" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Absorption Chlorophyll-a"</span>, </span>
<span id="cb44-197"><a href="#cb44-197" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>,</span>
<span id="cb44-198"><a href="#cb44-198" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"darkred"</span></span>
<span id="cb44-199"><a href="#cb44-199" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb44-200"><a href="#cb44-200" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">x=</span> <span class="dv">420</span>, <span class="at">y =</span> <span class="fl">0.232</span>, <span class="at">label =</span> <span class="st">"A"</span>, <span class="at">size =</span> <span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb44-201"><a href="#cb44-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-202"><a href="#cb44-202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede</span>()<span class="sc">+</span></span>
<span id="cb44-203"><a href="#cb44-203" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>( <span class="at">panel.grid.major =</span> ggplot2<span class="sc">::</span><span class="fu">element_line</span>(<span class="at">linetype =</span> <span class="st">"dotted"</span>, </span>
<span id="cb44-204"><a href="#cb44-204" aria-hidden="true" tabindex="-1"></a>      <span class="at">colour =</span> <span class="fu">alpha</span>(<span class="st">"grey30"</span>,<span class="fl">0.5</span>), <span class="at">linewidth =</span> <span class="fl">0.25</span>),</span>
<span id="cb44-205"><a href="#cb44-205" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>))</span>
<span id="cb44-206"><a href="#cb44-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-207"><a href="#cb44-207" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> (b)<span class="sc">+</span>(a)</span>
<span id="cb44-208"><a href="#cb44-208" aria-hidden="true" tabindex="-1"></a>plot</span>
<span id="cb44-209"><a href="#cb44-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-210"><a href="#cb44-210" aria-hidden="true" tabindex="-1"></a> <span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/plot_spectral_signature.png"</span>,plot, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">5</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span>
<span id="cb44-211"><a href="#cb44-211" aria-hidden="true" tabindex="-1"></a> <span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Low_res/plot_spectral_signature.png"</span>,plot, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">5</span>, <span class="at">dpi =</span> <span class="dv">200</span>)</span>
<span id="cb44-212"><a href="#cb44-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-213"><a href="#cb44-213" aria-hidden="true" tabindex="-1"></a><span class="do">##### INDEX ######</span></span>
<span id="cb44-214"><a href="#cb44-214" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb44-215"><a href="#cb44-215" aria-hidden="true" tabindex="-1"></a><span class="co"># spec_lib &lt;- readxl::read_excel("Data/speclib/Spectral_library_VEG_smoothed_A_GERAUD.xlsx") %&gt;% </span></span>
<span id="cb44-216"><a href="#cb44-216" aria-hidden="true" tabindex="-1"></a><span class="co">#   pivot_longer(-Wavelength, names_to = "Species", values_to = "Ref") %&gt;% </span></span>
<span id="cb44-217"><a href="#cb44-217" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(ID = Species,</span></span>
<span id="cb44-218"><a href="#cb44-218" aria-hidden="true" tabindex="-1"></a><span class="co">#          Species = sub("\\.\\.\\..*$", "", Species), </span></span>
<span id="cb44-219"><a href="#cb44-219" aria-hidden="true" tabindex="-1"></a><span class="co">#          Class = case_when(str_detect(Species, "nodosum") ~ "Phaeophyceae",</span></span>
<span id="cb44-220"><a href="#cb44-220" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "paludosa") ~ "Bacillariophyceae",</span></span>
<span id="cb44-221"><a href="#cb44-221" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "spiralis") ~ "Phaeophyceae",</span></span>
<span id="cb44-222"><a href="#cb44-222" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "limosum") ~ "Bacillariophyceae",</span></span>
<span id="cb44-223"><a href="#cb44-223" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "ramosissima") ~ "?",</span></span>
<span id="cb44-224"><a href="#cb44-224" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "Ulva ") ~ "Chlorophyceae",</span></span>
<span id="cb44-225"><a href="#cb44-225" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "serratus") ~ "Phaeophyceae",</span></span>
<span id="cb44-226"><a href="#cb44-226" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "Codium ") ~ "Chlorophyceae",</span></span>
<span id="cb44-227"><a href="#cb44-227" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "vesiculosus") ~ "Phaeophyceae",</span></span>
<span id="cb44-228"><a href="#cb44-228" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "okamurae") ~ "Phaeophyceae",</span></span>
<span id="cb44-229"><a href="#cb44-229" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "Vaucheria") ~ "Xanthophyceae",</span></span>
<span id="cb44-230"><a href="#cb44-230" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "marina") ~ "Magnoliopsida",</span></span>
<span id="cb44-231"><a href="#cb44-231" aria-hidden="true" tabindex="-1"></a><span class="co">#                            str_detect(Species, "noltei") ~ "Magnoliopsida")) %&gt;% </span></span>
<span id="cb44-232"><a href="#cb44-232" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(Class, Wavelength) </span></span>
<span id="cb44-233"><a href="#cb44-233" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb44-234"><a href="#cb44-234" aria-hidden="true" tabindex="-1"></a><span class="co">#  </span></span>
<span id="cb44-235"><a href="#cb44-235" aria-hidden="true" tabindex="-1"></a><span class="co">#  indices_speclib &lt;- spec_lib %&gt;% </span></span>
<span id="cb44-236"><a href="#cb44-236" aria-hidden="true" tabindex="-1"></a><span class="co">#    pivot_wider(names_from = "Wavelength", values_from = "Ref") %&gt;% </span></span>
<span id="cb44-237"><a href="#cb44-237" aria-hidden="true" tabindex="-1"></a><span class="co">#    group_by(Species,ID,Class) %&gt;% </span></span>
<span id="cb44-238"><a href="#cb44-238" aria-hidden="true" tabindex="-1"></a><span class="co">#    reframe(NDVI = (`800` - `668`)/(`800` + `668`),</span></span>
<span id="cb44-239"><a href="#cb44-239" aria-hidden="true" tabindex="-1"></a><span class="co">#            GLI = ((`550` - `668`)+(`550` - `451`))/(2*`550`+`668`+`451`),</span></span>
<span id="cb44-240"><a href="#cb44-240" aria-hidden="true" tabindex="-1"></a><span class="co">#            GNDVI =  (`800` - `550`)/(`800` + `550`),</span></span>
<span id="cb44-241"><a href="#cb44-241" aria-hidden="true" tabindex="-1"></a><span class="co">#            EVI = (`800` - `668`)/(`800` - 6*`668` - 7.5*`451`+1),</span></span>
<span id="cb44-242"><a href="#cb44-242" aria-hidden="true" tabindex="-1"></a><span class="co">#            # MSAVI =(2*`800`+ 1 - sqrt(((2*`800`-1)*(2*`800`-1))-8*(`800`-`668`)))/2</span></span>
<span id="cb44-243"><a href="#cb44-243" aria-hidden="true" tabindex="-1"></a><span class="co">#            )</span></span>
<span id="cb44-244"><a href="#cb44-244" aria-hidden="true" tabindex="-1"></a><span class="co">#  </span></span>
<span id="cb44-245"><a href="#cb44-245" aria-hidden="true" tabindex="-1"></a><span class="co">#  </span></span>
<span id="cb44-246"><a href="#cb44-246" aria-hidden="true" tabindex="-1"></a><span class="co">#  df_spectra_indices &lt;- csv_files %&gt;%</span></span>
<span id="cb44-247"><a href="#cb44-247" aria-hidden="true" tabindex="-1"></a><span class="co">#   lapply(read.csv) %&gt;%  # Read each CSV file into a list of dataframes</span></span>
<span id="cb44-248"><a href="#cb44-248" aria-hidden="true" tabindex="-1"></a><span class="co">#   bind_rows() %&gt;%         # Combine all dataframes into one</span></span>
<span id="cb44-249"><a href="#cb44-249" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::filter(Letter == "Field") %&gt;% </span></span>
<span id="cb44-250"><a href="#cb44-250" aria-hidden="true" tabindex="-1"></a><span class="co">#    mutate(Class = "Rhodophyceae",</span></span>
<span id="cb44-251"><a href="#cb44-251" aria-hidden="true" tabindex="-1"></a><span class="co">#           Species = "G.vermiculophylla") %&gt;% </span></span>
<span id="cb44-252"><a href="#cb44-252" aria-hidden="true" tabindex="-1"></a><span class="co">#    dplyr::select(c(Class,Species,ID,Wavelength,mean_RAW)) %&gt;% </span></span>
<span id="cb44-253"><a href="#cb44-253" aria-hidden="true" tabindex="-1"></a><span class="co">#    pivot_wider(names_from = "Wavelength", values_from = "mean_RAW") %&gt;% </span></span>
<span id="cb44-254"><a href="#cb44-254" aria-hidden="true" tabindex="-1"></a><span class="co">#    group_by(Species,ID,Class) %&gt;% </span></span>
<span id="cb44-255"><a href="#cb44-255" aria-hidden="true" tabindex="-1"></a><span class="co">#    reframe(NDVI = (`800` - `668`)/(`800` + `668`),</span></span>
<span id="cb44-256"><a href="#cb44-256" aria-hidden="true" tabindex="-1"></a><span class="co">#            GLI = ((`550` - `668`)+(`550` - `451`))/(2*`550`+`668`+`451`),</span></span>
<span id="cb44-257"><a href="#cb44-257" aria-hidden="true" tabindex="-1"></a><span class="co">#            GNDVI =  (`800` - `550`)/(`800` + `550`),</span></span>
<span id="cb44-258"><a href="#cb44-258" aria-hidden="true" tabindex="-1"></a><span class="co">#            EVI = (`800` - `668`)/(`800` - 6*`668` - 7.5*`451`+1),</span></span>
<span id="cb44-259"><a href="#cb44-259" aria-hidden="true" tabindex="-1"></a><span class="co">#            # MSAVI =(2*`800`+ 1 - sqrt(((2*`800`-1)*(2*`800`-1))-8*(`800`-`668`)))/2</span></span>
<span id="cb44-260"><a href="#cb44-260" aria-hidden="true" tabindex="-1"></a><span class="co">#            )</span></span>
<span id="cb44-261"><a href="#cb44-261" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb44-262"><a href="#cb44-262" aria-hidden="true" tabindex="-1"></a><span class="co">#  all__indices &lt;-df_spectra_indices %&gt;% </span></span>
<span id="cb44-263"><a href="#cb44-263" aria-hidden="true" tabindex="-1"></a><span class="co">#    bind_rows(indices_speclib)</span></span>
<span id="cb44-264"><a href="#cb44-264" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb44-265"><a href="#cb44-265" aria-hidden="true" tabindex="-1"></a><span class="co">#  all__indices %&gt;% </span></span>
<span id="cb44-266"><a href="#cb44-266" aria-hidden="true" tabindex="-1"></a><span class="co">#    dplyr::filter(!Class %in% c("Bacillariophyceae","?")) %&gt;% </span></span>
<span id="cb44-267"><a href="#cb44-267" aria-hidden="true" tabindex="-1"></a><span class="co">#    pivot_longer(c(NDVI,GLI,GNDVI,EVI), names_to = "Indice",values_to = "value") %&gt;% </span></span>
<span id="cb44-268"><a href="#cb44-268" aria-hidden="true" tabindex="-1"></a><span class="co">#    ggplot(aes(x = Class, y = value, color = Class, group = interaction(Indice,Class)))+</span></span>
<span id="cb44-269"><a href="#cb44-269" aria-hidden="true" tabindex="-1"></a><span class="co">#    facet_wrap(~ Indice)+</span></span>
<span id="cb44-270"><a href="#cb44-270" aria-hidden="true" tabindex="-1"></a><span class="co">#    geom_boxplot()+</span></span>
<span id="cb44-271"><a href="#cb44-271" aria-hidden="true" tabindex="-1"></a><span class="co">#    theme_Bede()</span></span>
<span id="cb44-272"><a href="#cb44-272" aria-hidden="true" tabindex="-1"></a><span class="co">#  </span></span>
<span id="cb44-273"><a href="#cb44-273" aria-hidden="true" tabindex="-1"></a><span class="co">#  spec_lib %&gt;% </span></span>
<span id="cb44-274"><a href="#cb44-274" aria-hidden="true" tabindex="-1"></a><span class="co">#   reframe(mean = mean(Ref, na.rm = T),</span></span>
<span id="cb44-275"><a href="#cb44-275" aria-hidden="true" tabindex="-1"></a><span class="co">#           sd = sd(Ref, na.rm = T)) %&gt;% </span></span>
<span id="cb44-276"><a href="#cb44-276" aria-hidden="true" tabindex="-1"></a><span class="co">#   ggplot()+</span></span>
<span id="cb44-277"><a href="#cb44-277" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_line(aes(x = Wavelength, y =mean, group = Class, color = Class))+</span></span>
<span id="cb44-278"><a href="#cb44-278" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_vline(xintercept = chla,</span></span>
<span id="cb44-279"><a href="#cb44-279" aria-hidden="true" tabindex="-1"></a><span class="co">#     color = "darkred", linetype = "dashed")+ # chl-a</span></span>
<span id="cb44-280"><a href="#cb44-280" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_vline(xintercept = PE,</span></span>
<span id="cb44-281"><a href="#cb44-281" aria-hidden="true" tabindex="-1"></a><span class="co">#     color = "darkgreen", linetype = "dashed")+ </span></span>
<span id="cb44-282"><a href="#cb44-282" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_vline(xintercept = PC,</span></span>
<span id="cb44-283"><a href="#cb44-283" aria-hidden="true" tabindex="-1"></a><span class="co">#     color = "goldenrod", linetype = "dashed")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>The spectral signature of <em>Gracilaria vermiculophylla</em> was characterized by a wavy reflectance pattern in the visible region of the spectrum, between 500 and 700 nm (<a href="#fig-SpecDescri" class="quarto-xref">Figure&nbsp;5</a> A). This pattern was primarily driven by the presence of phycoerythrin and phycocyanin, which exhibited maximum absorption peaks at approximately 565 nm and 620 nm, respectively. An additional absorption feature around 495 nm was likely attributable to the presence of accessory carotenoid pigments. The most pronounced absorption peak occurred at 675 nm, corresponding to chlorophyll-a absorption. The second derivative analysis clearly highlighted the inflection points corresponding to the main absorption peaks at 495, 565, 620, and 675 nm, allowing for a more precise identification of the spectral contributions of these pigments (<a href="#fig-SpecDescri" class="quarto-xref">Figure&nbsp;5</a> B).</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [31]:</pre></div><div id="cell-fig-SpecDescri" class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"./Figures/Low_res/plot_spectral_signature.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-SpecDescri" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-SpecDescri-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="./Figures/Low_res/plot_spectral_signature.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Figure&nbsp;5: Hyperspectral signature of Gracilaria vermiculophylla (A) and its second derivative (B). The black line represents the average spectra, while the shaded ribbon indicates the standard deviation. Dashed lines mark the absorption maxima of Phycoerythrin, Phycocyanin, and Chlorophyll-a, shown in green, orange, and red, respectively."><img src="./Figures/Low_res/plot_spectral_signature.png" class="img-fluid figure-img" style="width:95.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-SpecDescri-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Hyperspectral signature of <em>Gracilaria vermiculophylla</em> (A) and its second derivative (B). The black line represents the average spectra, while the shaded ribbon indicates the standard deviation. Dashed lines mark the absorption maxima of Phycoerythrin, Phycocyanin, and Chlorophyll-a, shown in green, orange, and red, respectively.
</figcaption>
</figure>
</div>
</div>
</div></div>
</section>
<section id="spatial-distribution" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="spatial-distribution"><span class="header-section-number">4.3</span> Spatial distribution</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [32]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Utilities.Package)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>msk <span class="ot">&lt;-</span> <span class="st">"Data/shp/mask/Belon_32630.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>DISCOV_rast <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">"Data/DISCOV/Belon_1104_MS_DISCOV_InvaSea_26112024_pred.tif"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">project</span>(<span class="fu">crs</span>(msk), <span class="at">method =</span> <span class="st">"near"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(msk, <span class="at">mask =</span> T)</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(DISCOV_rast)[<span class="fu">values</span>(DISCOV_rast) <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(DISCOV_rast) <span class="ot">&lt;-</span> <span class="st">"layer"</span></span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>DISCOV_rast<span class="ot">&lt;-</span>DISCOV_rast <span class="sc">%&gt;%</span> </span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.factor</span>()</span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a>DISCOV_Belon <span class="ot">&lt;-</span> <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a>    tidyterra<span class="sc">::</span><span class="fu">geom_spatraster</span>(<span class="at">data =</span> DISCOV_rast, </span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">fill =</span> layer),</span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a>                               <span class="at">maxcell =</span> <span class="dv">500832</span><span class="sc">*</span><span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">8</span>),</span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">values=</span><span class="fu">c</span>(<span class="st">"#DAA520"</span>,<span class="st">"#b3ff1a"</span>,<span class="st">"#389318"</span>,<span class="st">"#873e23"</span>,<span class="st">"#b3002d"</span>,<span class="st">"#a2a3a2"</span>,<span class="st">"#42c9bc"</span>,<span class="cn">NA</span>,<span class="cn">NA</span>),</span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">na.value =</span> <span class="cn">NA</span>,</span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">"DISCOV v2"</span>,</span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Bacillariophyceae"</span>,</span>
<span id="cb46-27"><a href="#cb46-27" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"Chlorophyceae"</span>,</span>
<span id="cb46-28"><a href="#cb46-28" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"Magnoliopsida"</span>,</span>
<span id="cb46-29"><a href="#cb46-29" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"Phaeophyceae"</span>,</span>
<span id="cb46-30"><a href="#cb46-30" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"Rhodophyceae"</span>,</span>
<span id="cb46-31"><a href="#cb46-31" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"Sediment"</span>,</span>
<span id="cb46-32"><a href="#cb46-32" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"Water"</span>))<span class="sc">+</span></span>
<span id="cb46-33"><a href="#cb46-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span>  <span class="fu">ext</span>(DISCOV_rast)[<span class="dv">1</span>]<span class="sc">+</span><span class="fl">0.04</span><span class="sc">*</span>(<span class="fu">ext</span>(DISCOV_rast)[<span class="dv">2</span>]<span class="sc">-</span><span class="fu">ext</span>(DISCOV_rast)[<span class="dv">1</span>]),</span>
<span id="cb46-34"><a href="#cb46-34" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span>  <span class="fu">ext</span>(DISCOV_rast)[<span class="dv">4</span>]<span class="sc">-</span><span class="fl">0.08</span><span class="sc">*</span>(<span class="fu">ext</span>(DISCOV_rast)[<span class="dv">4</span>]<span class="sc">-</span><span class="fu">ext</span>(DISCOV_rast)[<span class="dv">3</span>]),</span>
<span id="cb46-35"><a href="#cb46-35" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">"A"</span>),</span>
<span id="cb46-36"><a href="#cb46-36" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="dv">10</span>)<span class="sc">+</span></span>
<span id="cb46-37"><a href="#cb46-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(<span class="at">expand =</span> F,</span>
<span id="cb46-38"><a href="#cb46-38" aria-hidden="true" tabindex="-1"></a>           <span class="at">xlim =</span> <span class="fu">c</span>(<span class="fl">450180.4</span>, <span class="fl">451247.2</span><span class="sc">+</span><span class="dv">18</span>),</span>
<span id="cb46-39"><a href="#cb46-39" aria-hidden="true" tabindex="-1"></a>           <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">5296561</span>, <span class="dv">5297042</span>))<span class="sc">+</span></span>
<span id="cb46-40"><a href="#cb46-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede_Map</span>()<span class="sc">+</span></span>
<span id="cb46-41"><a href="#cb46-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># coord_equal()+</span></span>
<span id="cb46-42"><a href="#cb46-42" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme</span>(</span>
<span id="cb46-43"><a href="#cb46-43" aria-hidden="true" tabindex="-1"></a>     <span class="co"># axis.text.x = element_text(size = 10, angle = -20, color = NA,hjust = 0),</span></span>
<span id="cb46-44"><a href="#cb46-44" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb46-45"><a href="#cb46-45" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-46"><a href="#cb46-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-47"><a href="#cb46-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-48"><a href="#cb46-48" aria-hidden="true" tabindex="-1"></a>         <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>),</span>
<span id="cb46-49"><a href="#cb46-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-50"><a href="#cb46-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.98</span>,<span class="fl">0.98</span>),</span>
<span id="cb46-51"><a href="#cb46-51" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb46-52"><a href="#cb46-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>,<span class="st">"cm"</span>))</span>
<span id="cb46-53"><a href="#cb46-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-54"><a href="#cb46-54" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Prediction_Map/Belon_DISCOV.png"</span>,DISCOV_Belon,<span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="fl">4.225</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [33]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Utilities.Package)</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>msk <span class="ot">&lt;-</span> <span class="st">"Data/shp/mask/Belon_32630.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>Belon_RGB <span class="ot">&lt;-</span> <span class="st">"Data/RGB/Belon_RGB_MULTI.tif"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rast</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(msk, <span class="at">mask =</span> T)</span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a><span class="co"># values(Belon_RGB[[1]])[values(Belon_RGB[[1]]) == 0] = NA</span></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a><span class="co"># RGB(Belon_RGB) &lt;- 1:3</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>lim_x<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">as.numeric</span>(<span class="fu">ext</span>(Belon_RGB)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]))</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>lim_y<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">as.numeric</span>(<span class="fu">ext</span>(Belon_RGB)[<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>]))</span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a> RGB_Belon<span class="ot">&lt;-</span> <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>   tidyterra<span class="sc">::</span><span class="fu">geom_spatraster_rgb</span>(<span class="at">data =</span> Belon_RGB,</span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">r =</span> <span class="dv">1</span>,</span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">g =</span> <span class="dv">2</span>,</span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">b =</span> <span class="dv">3</span>,</span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">maxcell =</span> <span class="dv">4010533</span></span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a>         )<span class="sc">+</span></span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span>  <span class="fu">ext</span>(Belon_RGB)[<span class="dv">1</span>]<span class="sc">+</span><span class="fl">0.04</span><span class="sc">*</span>(<span class="fu">ext</span>(Belon_RGB)[<span class="dv">2</span>]<span class="sc">-</span><span class="fu">ext</span>(Belon_RGB)[<span class="dv">1</span>]),</span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span>  <span class="fu">ext</span>(Belon_RGB)[<span class="dv">4</span>]<span class="sc">-</span><span class="fl">0.08</span><span class="sc">*</span>(<span class="fu">ext</span>(Belon_RGB)[<span class="dv">4</span>]<span class="sc">-</span><span class="fu">ext</span>(Belon_RGB)[<span class="dv">3</span>]),</span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">"B"</span>),</span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="dv">10</span>)<span class="sc">+</span></span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a>    ggspatial<span class="sc">::</span><span class="fu">annotation_scale</span>(<span class="at">location =</span> <span class="st">"br"</span>,</span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a>                                <span class="at">pad_y =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>),</span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a>                                <span class="at">text_cex=</span><span class="dv">2</span>)<span class="sc">+</span></span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme_Bede_Map</span>()<span class="sc">+</span></span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(<span class="at">expand =</span> F,</span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a>           <span class="at">xlim =</span> <span class="fu">c</span>(<span class="fl">450180.4</span>, <span class="fl">451247.2</span><span class="sc">+</span><span class="dv">18</span>),</span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a>           <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">5296561</span>, <span class="dv">5297042</span>))<span class="sc">+</span></span>
<span id="cb47-40"><a href="#cb47-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># coord_equal()+</span></span>
<span id="cb47-41"><a href="#cb47-41" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme</span>(</span>
<span id="cb47-42"><a href="#cb47-42" aria-hidden="true" tabindex="-1"></a>     <span class="co"># axis.text.x = element_text(size = 10, angle = -20,hjust = 0, color = NA),</span></span>
<span id="cb47-43"><a href="#cb47-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb47-44"><a href="#cb47-44" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-45"><a href="#cb47-45" aria-hidden="true" tabindex="-1"></a>         <span class="co"># axis.ticks.y = element_blank(),</span></span>
<span id="cb47-46"><a href="#cb47-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.ticks.x=</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-47"><a href="#cb47-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>), </span>
<span id="cb47-48"><a href="#cb47-48" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb47-49"><a href="#cb47-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-50"><a href="#cb47-50" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb47-51"><a href="#cb47-51" aria-hidden="true" tabindex="-1"></a> <span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Prediction_Map/Belon_RGB.png"</span>,RGB_Belon,<span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="fl">4.225</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [34]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Utilities.Package)</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scico)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>msk <span class="ot">&lt;-</span> <span class="st">"Data/shp/mask/Belon_DEM_32630.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>DEM_rast <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">"Data/DEM/Belon_DEM_LiDAR.tif"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(msk, <span class="at">mask =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>  terra<span class="sc">::</span><span class="fu">resample</span>(Belon_RGB, <span class="at">method =</span> <span class="st">"near"</span>)</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>p999 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(DEM_rast <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span>  <span class="fu">pull</span>(Belon_DEM_LiDAR), <span class="at">probs =</span> <span class="fl">0.999</span>)</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(DEM_rast)[<span class="fu">values</span>(DEM_rast) <span class="sc">&gt;</span> p999] <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(DEM_rast)[<span class="fu">values</span>(DEM_rast) <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a>lim_x<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">as.numeric</span>(<span class="fu">ext</span>(Belon_RGB)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]))</span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a>lim_y<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">as.numeric</span>(<span class="fu">ext</span>(Belon_RGB)[<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>]))</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a>bathy_Belon <span class="ot">&lt;-</span> <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a>   <span class="co"># tidyterra::geom_spatraster_rgb(data = Belon_RGB,</span></span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a>   <span class="co">#       r = 1,</span></span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a>   <span class="co">#       g = 2,</span></span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a>   <span class="co">#       b = 3,</span></span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a>   <span class="co">#       maxcell = 1000,</span></span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a>   <span class="co">#       alpha = 0</span></span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a>   <span class="co">#       )+</span></span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a>    tidyterra<span class="sc">::</span><span class="fu">geom_spatraster</span>(<span class="at">data =</span> DEM_rast, </span>
<span id="cb48-33"><a href="#cb48-33" aria-hidden="true" tabindex="-1"></a>                               <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">fill =</span> Belon_DEM_LiDAR ),</span>
<span id="cb48-34"><a href="#cb48-34" aria-hidden="true" tabindex="-1"></a>                               <span class="at">maxcell =</span> <span class="dv">500832</span>)<span class="sc">+</span></span>
<span id="cb48-35"><a href="#cb48-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_scico</span>(<span class="at">palette =</span> <span class="st">"batlow"</span>, </span>
<span id="cb48-36"><a href="#cb48-36" aria-hidden="true" tabindex="-1"></a>                   <span class="at">na.value =</span> <span class="cn">NA</span>,</span>
<span id="cb48-37"><a href="#cb48-37" aria-hidden="true" tabindex="-1"></a>                   <span class="at">name =</span> <span class="st">"Bathymetry"</span>,</span>
<span id="cb48-38"><a href="#cb48-38" aria-hidden="true" tabindex="-1"></a>                   <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.5</span>,<span class="dv">3</span>),</span>
<span id="cb48-39"><a href="#cb48-39" aria-hidden="true" tabindex="-1"></a>                   <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">3</span>),</span>
<span id="cb48-40"><a href="#cb48-40" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"0 m"</span>,<span class="st">"1.5 m"</span>, <span class="st">"3 m"</span>)) <span class="sc">+</span></span>
<span id="cb48-41"><a href="#cb48-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span>  <span class="fu">ext</span>(Belon_RGB)[<span class="dv">1</span>]<span class="sc">+</span><span class="fl">0.04</span><span class="sc">*</span>(<span class="fu">ext</span>(Belon_RGB)[<span class="dv">2</span>]<span class="sc">-</span><span class="fu">ext</span>(Belon_RGB)[<span class="dv">1</span>]),</span>
<span id="cb48-42"><a href="#cb48-42" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span>  <span class="fu">ext</span>(Belon_RGB)[<span class="dv">4</span>]<span class="sc">-</span><span class="fl">0.08</span><span class="sc">*</span>(<span class="fu">ext</span>(Belon_RGB)[<span class="dv">4</span>]<span class="sc">-</span><span class="fu">ext</span>(Belon_RGB)[<span class="dv">3</span>]),</span>
<span id="cb48-43"><a href="#cb48-43" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">"C"</span>),</span>
<span id="cb48-44"><a href="#cb48-44" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="dv">10</span>)<span class="sc">+</span></span>
<span id="cb48-45"><a href="#cb48-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede_Map</span>()<span class="sc">+</span></span>
<span id="cb48-46"><a href="#cb48-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(<span class="at">expand =</span> F,</span>
<span id="cb48-47"><a href="#cb48-47" aria-hidden="true" tabindex="-1"></a>           <span class="at">xlim =</span> <span class="fu">c</span>(<span class="fl">450180.4</span>, <span class="fl">451247.2</span><span class="sc">+</span><span class="dv">18</span>),</span>
<span id="cb48-48"><a href="#cb48-48" aria-hidden="true" tabindex="-1"></a>           <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">5296561</span>, <span class="dv">5297042</span>))<span class="sc">+</span></span>
<span id="cb48-49"><a href="#cb48-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># coord_equal()+</span></span>
<span id="cb48-50"><a href="#cb48-50" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme</span>(</span>
<span id="cb48-51"><a href="#cb48-51" aria-hidden="true" tabindex="-1"></a>     <span class="co"># axis.text.x = element_text(size = 10, angle = -20,hjust = 0, color = NA),</span></span>
<span id="cb48-52"><a href="#cb48-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb48-53"><a href="#cb48-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb48-54"><a href="#cb48-54" aria-hidden="true" tabindex="-1"></a>         <span class="co"># axis.ticks.y = element_blank(),</span></span>
<span id="cb48-55"><a href="#cb48-55" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.ticks.x=</span> <span class="fu">element_blank</span>(),</span>
<span id="cb48-56"><a href="#cb48-56" aria-hidden="true" tabindex="-1"></a>         <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>), </span>
<span id="cb48-57"><a href="#cb48-57" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb48-58"><a href="#cb48-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-59"><a href="#cb48-59" aria-hidden="true" tabindex="-1"></a> <span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Prediction_Map/Belon_Bathy.png"</span>,bathy_Belon,<span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="fl">4.225</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span>
<span id="cb48-60"><a href="#cb48-60" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb48-61"><a href="#cb48-61" aria-hidden="true" tabindex="-1"></a> <span class="co"># nrow(DEM_rast %&gt;%  as.data.frame() %&gt;% dplyr::filter(!is.na(Belon_DEM_LiDAR)))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [35]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>msk <span class="ot">&lt;-</span> <span class="st">"Data/shp/mask/Belon_DEM_32630.shp"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_sf</span>()</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>Belon_RGB <span class="ot">&lt;-</span> <span class="st">"Data/RGB/Belon_RGB_MULTI.tif"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rast</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(msk, <span class="at">mask =</span> T)</span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>DEM_rast <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">"Data/DEM/Belon_DEM_LiDAR.tif"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crop</span>(msk, <span class="at">mask =</span> T)</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>Angle_df<span class="ot">&lt;-</span><span class="fu">read_csv</span>(<span class="st">"Data/Model/FullDF_10timesResample_DEMSlope_PredPercent.csv"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Site  <span class="sc">==</span> <span class="st">"Belon"</span>)</span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>angle_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>    tidyterra<span class="sc">::</span><span class="fu">geom_spatraster</span>(<span class="at">data =</span> DEM_rast, </span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>                               <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">fill =</span> Belon_DEM_LiDAR ),</span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a>                               <span class="at">maxcell =</span> <span class="dv">100</span>,</span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">alpha=</span><span class="dv">0</span>, </span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>                               <span class="at">show.legend =</span> F)<span class="sc">+</span></span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a>  scico<span class="sc">::</span><span class="fu">scale_fill_scico</span>(<span class="at">palette =</span> <span class="st">"batlow"</span>, <span class="at">na.value =</span> <span class="cn">NA</span>,</span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a>                   <span class="at">name =</span> <span class="st">"Bathymetry"</span>,</span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>                   <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.5</span>,<span class="dv">3</span>),</span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a>                   <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">3</span>),</span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"0 m"</span>,<span class="st">"1.5 m"</span>, <span class="st">"3 m"</span>))<span class="sc">+</span></span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a>  ggnewscale<span class="sc">::</span><span class="fu">new_scale_fill</span>()<span class="sc">+</span></span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(Angle_df, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">fill =</span> Slope_Cat)) <span class="sc">+</span></span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a>                    <span class="at">values=</span><span class="fu">c</span>(<span class="st">"#8E7CC3"</span>,<span class="st">"#76A5AF"</span>,<span class="st">"#93C47D"</span>),</span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a>                    <span class="at">na.value =</span> <span class="cn">NA</span>,</span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">"Angle"</span>,</span>
<span id="cb49-33"><a href="#cb49-33" aria-hidden="true" tabindex="-1"></a>                    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Flat"</span>,</span>
<span id="cb49-34"><a href="#cb49-34" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"Angled"</span>,</span>
<span id="cb49-35"><a href="#cb49-35" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"Steep"</span>))<span class="sc">+</span></span>
<span id="cb49-36"><a href="#cb49-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span>  <span class="fu">ext</span>(Belon_RGB)[<span class="dv">1</span>]<span class="sc">+</span><span class="fl">0.04</span><span class="sc">*</span>(<span class="fu">ext</span>(Belon_RGB)[<span class="dv">2</span>]<span class="sc">-</span><span class="fu">ext</span>(Belon_RGB)[<span class="dv">1</span>]),</span>
<span id="cb49-37"><a href="#cb49-37" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span>  <span class="fu">ext</span>(Belon_RGB)[<span class="dv">4</span>]<span class="sc">-</span><span class="fl">0.08</span><span class="sc">*</span>(<span class="fu">ext</span>(Belon_RGB)[<span class="dv">4</span>]<span class="sc">-</span><span class="fu">ext</span>(Belon_RGB)[<span class="dv">3</span>]),</span>
<span id="cb49-38"><a href="#cb49-38" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">"D"</span>),</span>
<span id="cb49-39"><a href="#cb49-39" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="dv">10</span>)<span class="sc">+</span></span>
<span id="cb49-40"><a href="#cb49-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede_Map</span>()<span class="sc">+</span></span>
<span id="cb49-41"><a href="#cb49-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(<span class="at">expand =</span> F,</span>
<span id="cb49-42"><a href="#cb49-42" aria-hidden="true" tabindex="-1"></a>           <span class="at">xlim =</span> <span class="fu">c</span>(<span class="fl">450180.4</span>, <span class="fl">451247.2</span><span class="sc">+</span><span class="dv">18</span>),</span>
<span id="cb49-43"><a href="#cb49-43" aria-hidden="true" tabindex="-1"></a>           <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">5296561</span>, <span class="dv">5297042</span>))<span class="sc">+</span></span>
<span id="cb49-44"><a href="#cb49-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># coord_equal()+</span></span>
<span id="cb49-45"><a href="#cb49-45" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">angle =</span> <span class="sc">-</span><span class="dv">20</span>,<span class="at">hjust =</span> <span class="dv">0</span>),</span>
<span id="cb49-46"><a href="#cb49-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb49-47"><a href="#cb49-47" aria-hidden="true" tabindex="-1"></a>         <span class="co"># axis.text.y = element_blank(),</span></span>
<span id="cb49-48"><a href="#cb49-48" aria-hidden="true" tabindex="-1"></a>         <span class="co"># axis.ticks.y = element_blank(), </span></span>
<span id="cb49-49"><a href="#cb49-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb49-50"><a href="#cb49-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>),</span>
<span id="cb49-51"><a href="#cb49-51" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb49-52"><a href="#cb49-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.98</span>,<span class="fl">0.98</span>),</span>
<span id="cb49-53"><a href="#cb49-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb49-54"><a href="#cb49-54" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb49-55"><a href="#cb49-55" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>,<span class="st">"cm"</span>))</span>
<span id="cb49-56"><a href="#cb49-56" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb49-57"><a href="#cb49-57" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Prediction_Map/Belon_angle.png"</span>,angle_plot,<span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="fl">4.56</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [36]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Utilities.Package)</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>RGB <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">"Paper/Figures/High_res/Prediction_Map/Belon_RGB.png"</span>)</span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="fu">RGB</span>(RGB) <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>Pred <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">"Paper/Figures/High_res/Prediction_Map/Belon_DISCOV.png"</span>)</span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a><span class="fu">RGB</span>(Pred) <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span></span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a>Bathy <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">"Paper/Figures/High_res/Prediction_Map/Belon_Bathy.png"</span>)</span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a><span class="fu">RGB</span>(Pred) <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a>Angle <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">"Paper/Figures/High_res/Prediction_Map/Belon_angle.png"</span>)</span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a><span class="fu">RGB</span>(Angle) <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span></span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a>offset <span class="ot">=</span> <span class="dv">50</span></span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a><span class="fu">ext</span>(Pred) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">4000</span>,<span class="dv">1689</span><span class="sc">-</span>offset,<span class="dv">1689</span><span class="sc">*</span><span class="dv">2</span><span class="sc">-</span>offset)</span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a><span class="fu">ext</span>(RGB) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">4000</span>,<span class="dv">0</span>,<span class="dv">1689</span>)</span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a><span class="fu">ext</span>(Bathy) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">4000</span>,<span class="sc">-</span><span class="dv">1689</span><span class="sc">+</span>(offset),offset)</span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ext</span>(Angle) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">4000</span>,<span class="sc">-</span><span class="dv">1689</span><span class="sc">+</span>(<span class="dv">2</span><span class="sc">*</span>offset)<span class="sc">-</span><span class="dv">1823</span>,<span class="sc">-</span><span class="dv">1689</span><span class="sc">+</span><span class="dv">2</span><span class="sc">*</span>offset)</span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a> Belon_patchwork<span class="ot">&lt;-</span> <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a>   tidyterra<span class="sc">::</span><span class="fu">geom_spatraster_rgb</span>(<span class="at">data =</span> Pred,</span>
<span id="cb50-29"><a href="#cb50-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">r =</span> <span class="dv">1</span>,</span>
<span id="cb50-30"><a href="#cb50-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">g =</span> <span class="dv">2</span>,</span>
<span id="cb50-31"><a href="#cb50-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">b =</span> <span class="dv">3</span>,</span>
<span id="cb50-32"><a href="#cb50-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">maxcell =</span> <span class="dv">4000</span><span class="sc">*</span><span class="dv">1689</span></span>
<span id="cb50-33"><a href="#cb50-33" aria-hidden="true" tabindex="-1"></a>         )<span class="sc">+</span></span>
<span id="cb50-34"><a href="#cb50-34" aria-hidden="true" tabindex="-1"></a>   tidyterra<span class="sc">::</span><span class="fu">geom_spatraster_rgb</span>(<span class="at">data =</span> RGB,</span>
<span id="cb50-35"><a href="#cb50-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">r =</span> <span class="dv">1</span>,</span>
<span id="cb50-36"><a href="#cb50-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">g =</span> <span class="dv">2</span>,</span>
<span id="cb50-37"><a href="#cb50-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">b =</span> <span class="dv">3</span>,</span>
<span id="cb50-38"><a href="#cb50-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">maxcell =</span> <span class="dv">4000</span><span class="sc">*</span><span class="dv">1689</span></span>
<span id="cb50-39"><a href="#cb50-39" aria-hidden="true" tabindex="-1"></a>         )<span class="sc">+</span></span>
<span id="cb50-40"><a href="#cb50-40" aria-hidden="true" tabindex="-1"></a>   tidyterra<span class="sc">::</span><span class="fu">geom_spatraster_rgb</span>(<span class="at">data =</span> Bathy,</span>
<span id="cb50-41"><a href="#cb50-41" aria-hidden="true" tabindex="-1"></a>         <span class="at">r =</span> <span class="dv">1</span>,</span>
<span id="cb50-42"><a href="#cb50-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">g =</span> <span class="dv">2</span>,</span>
<span id="cb50-43"><a href="#cb50-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">b =</span> <span class="dv">3</span>,</span>
<span id="cb50-44"><a href="#cb50-44" aria-hidden="true" tabindex="-1"></a>         <span class="at">maxcell =</span> <span class="dv">4000</span><span class="sc">*</span><span class="dv">1689</span></span>
<span id="cb50-45"><a href="#cb50-45" aria-hidden="true" tabindex="-1"></a>         )<span class="sc">+</span></span>
<span id="cb50-46"><a href="#cb50-46" aria-hidden="true" tabindex="-1"></a>   tidyterra<span class="sc">::</span><span class="fu">geom_spatraster_rgb</span>(<span class="at">data =</span> Angle,</span>
<span id="cb50-47"><a href="#cb50-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">r =</span> <span class="dv">1</span>,</span>
<span id="cb50-48"><a href="#cb50-48" aria-hidden="true" tabindex="-1"></a>         <span class="at">g =</span> <span class="dv">2</span>,</span>
<span id="cb50-49"><a href="#cb50-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">b =</span> <span class="dv">3</span>,</span>
<span id="cb50-50"><a href="#cb50-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">maxcell =</span> <span class="dv">4000</span><span class="sc">*</span><span class="dv">1823</span></span>
<span id="cb50-51"><a href="#cb50-51" aria-hidden="true" tabindex="-1"></a>         )<span class="sc">+</span></span>
<span id="cb50-52"><a href="#cb50-52" aria-hidden="true" tabindex="-1"></a>   <span class="fu">coord_equal</span>()<span class="sc">+</span></span>
<span id="cb50-53"><a href="#cb50-53" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme_void</span>()</span>
<span id="cb50-54"><a href="#cb50-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-55"><a href="#cb50-55" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/Prediction_Map/Belon_maps.png"</span>,Belon_patchwork,<span class="at">width =</span> <span class="fl">7.89</span>, <span class="at">height =</span> <span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span>
<span id="cb50-56"><a href="#cb50-56" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Low_res/Belon_maps.png"</span>,Belon_patchwork,<span class="at">width =</span> <span class="fl">7.89</span>, <span class="at">height =</span> <span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">200</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [37]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="co"># file_list &lt;- "Data/Shiny_Validation" %&gt;% </span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   list.files(".shp",recursive = T, full.names = T) %&gt;% </span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   as_tibble() %&gt;% </span></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   rename(path = "value") %&gt;% </span></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(filename = gsub(".*/","",path) %&gt;% </span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a><span class="co">#            gsub("shp","",.),</span></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a><span class="co">#          User_date = sapply(strsplit(path, "/"), function(x) x[3]))</span></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a><span class="co"># for(i in 1:nrow(file_list)){</span></span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   shp_i &lt;- read_sf(file_list$path[i]) %&gt;% </span></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a><span class="co">#     mutate(user_date = file_list$User_date[i],</span></span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a><span class="co">#            snapshot =  file_list$filename[i])</span></span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   if(i == 1){</span></span>
<span id="cb51-19"><a href="#cb51-19" aria-hidden="true" tabindex="-1"></a><span class="co">#     shp &lt;- shp_i</span></span>
<span id="cb51-20"><a href="#cb51-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   }else{</span></span>
<span id="cb51-21"><a href="#cb51-21" aria-hidden="true" tabindex="-1"></a><span class="co">#     shp &lt;- rbind(shp,shp_i)</span></span>
<span id="cb51-22"><a href="#cb51-22" aria-hidden="true" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb51-23"><a href="#cb51-23" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb51-24"><a href="#cb51-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-25"><a href="#cb51-25" aria-hidden="true" tabindex="-1"></a><span class="co"># write_sf(shp,"Data/Shiny_Validation/shp_all_validation.shp")</span></span>
<span id="cb51-26"><a href="#cb51-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-27"><a href="#cb51-27" aria-hidden="true" tabindex="-1"></a>shp <span class="ot">&lt;-</span> <span class="fu">read_sf</span>(<span class="st">"Data/Shiny_Validation/shp_all_validation.shp"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-28"><a href="#cb51-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">site =</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(snapshot, <span class="st">"_"</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>]))</span>
<span id="cb51-29"><a href="#cb51-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-30"><a href="#cb51-30" aria-hidden="true" tabindex="-1"></a>img_list <span class="ot">&lt;-</span> <span class="st">"Data/DISCOV"</span> <span class="sc">%&gt;%</span> </span>
<span id="cb51-31"><a href="#cb51-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="st">".tif"</span>, <span class="at">full.names =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb51-32"><a href="#cb51-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb51-33"><a href="#cb51-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">path =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-34"><a href="#cb51-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">filename =</span> <span class="fu">gsub</span>(<span class="st">".*/"</span>,<span class="st">""</span>,path),</span>
<span id="cb51-35"><a href="#cb51-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">site =</span> <span class="fu">case_when</span>(<span class="fu">str_detect</span>(filename,<span class="st">"Aven"</span>) <span class="sc">~</span> <span class="st">"Aven"</span>,</span>
<span id="cb51-36"><a href="#cb51-36" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">str_detect</span>(filename,<span class="st">"Belon"</span>) <span class="sc">~</span> <span class="st">"Belon"</span>,</span>
<span id="cb51-37"><a href="#cb51-37" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">str_detect</span>(filename,<span class="st">"North"</span>) <span class="sc">~</span> <span class="st">"SajaNorth"</span>,</span>
<span id="cb51-38"><a href="#cb51-38" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">str_detect</span>(filename,<span class="st">"South"</span>) <span class="sc">~</span> <span class="st">"SajaSouth"</span>))</span>
<span id="cb51-39"><a href="#cb51-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-40"><a href="#cb51-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-41"><a href="#cb51-41" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(img_list)) {</span>
<span id="cb51-42"><a href="#cb51-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb51-43"><a href="#cb51-43" aria-hidden="true" tabindex="-1"></a>  img <span class="ot">&lt;-</span> <span class="fu">rast</span>(img_list<span class="sc">$</span>path[i])</span>
<span id="cb51-44"><a href="#cb51-44" aria-hidden="true" tabindex="-1"></a>    <span class="co"># as.factor()</span></span>
<span id="cb51-45"><a href="#cb51-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(img) <span class="ot">=</span> <span class="st">"pred"</span></span>
<span id="cb51-46"><a href="#cb51-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb51-47"><a href="#cb51-47" aria-hidden="true" tabindex="-1"></a>  shp_i <span class="ot">&lt;-</span> shp <span class="sc">%&gt;%</span> </span>
<span id="cb51-48"><a href="#cb51-48" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(site <span class="sc">==</span> <span class="fu">unique</span>(img_list<span class="sc">$</span>site)[i]) <span class="sc">%&gt;%</span> </span>
<span id="cb51-49"><a href="#cb51-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Truth =</span> <span class="fu">case_when</span>(Vegetation <span class="sc">==</span> <span class="st">"Red algae"</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb51-50"><a href="#cb51-50" aria-hidden="true" tabindex="-1"></a>                             T <span class="sc">~</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-51"><a href="#cb51-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rasterize</span>(img, <span class="at">field =</span> <span class="st">"Truth"</span>)</span>
<span id="cb51-52"><a href="#cb51-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(shp_i) <span class="ot">=</span> <span class="st">"true"</span></span>
<span id="cb51-53"><a href="#cb51-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-54"><a href="#cb51-54" aria-hidden="true" tabindex="-1"></a>  stk <span class="ot">&lt;-</span> <span class="fu">c</span>(img,shp_i) <span class="sc">%&gt;%</span> </span>
<span id="cb51-55"><a href="#cb51-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>(<span class="at">xy =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb51-56"><a href="#cb51-56" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(true)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-57"><a href="#cb51-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">pred =</span> <span class="fu">case_when</span>(pred <span class="sc">==</span> <span class="dv">5</span> <span class="sc">~</span> <span class="dv">1</span>, </span>
<span id="cb51-58"><a href="#cb51-58" aria-hidden="true" tabindex="-1"></a>                            T <span class="sc">~</span> <span class="dv">0</span>),</span>
<span id="cb51-59"><a href="#cb51-59" aria-hidden="true" tabindex="-1"></a>           <span class="at">site =</span> <span class="fu">unique</span>(img_list<span class="sc">$</span>site)[i])</span>
<span id="cb51-60"><a href="#cb51-60" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb51-61"><a href="#cb51-61" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (i<span class="sc">==</span><span class="dv">1</span>){</span>
<span id="cb51-62"><a href="#cb51-62" aria-hidden="true" tabindex="-1"></a>    validation_df <span class="ot">&lt;-</span> stk</span>
<span id="cb51-63"><a href="#cb51-63" aria-hidden="true" tabindex="-1"></a>  }<span class="cf">else</span>{</span>
<span id="cb51-64"><a href="#cb51-64" aria-hidden="true" tabindex="-1"></a>    validation_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(validation_df,stk)</span>
<span id="cb51-65"><a href="#cb51-65" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb51-66"><a href="#cb51-66" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-67"><a href="#cb51-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-68"><a href="#cb51-68" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb51-69"><a href="#cb51-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-70"><a href="#cb51-70" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">confusionMatrix</span>(<span class="fu">as.factor</span>(validation_df<span class="sc">$</span>pred),<span class="fu">as.factor</span>(validation_df<span class="sc">$</span>true))</span>
<span id="cb51-71"><a href="#cb51-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-72"><a href="#cb51-72" aria-hidden="true" tabindex="-1"></a>validation_matrix <span class="ot">&lt;-</span> a<span class="sc">$</span>overall <span class="sc">%&gt;%</span> </span>
<span id="cb51-73"><a href="#cb51-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb51-74"><a href="#cb51-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb51-75"><a href="#cb51-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>({</span>
<span id="cb51-76"><a href="#cb51-76" aria-hidden="true" tabindex="-1"></a>     a<span class="sc">$</span>byClass <span class="sc">%&gt;%</span> </span>
<span id="cb51-77"><a href="#cb51-77" aria-hidden="true" tabindex="-1"></a>      <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb51-78"><a href="#cb51-78" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames_to_column</span>()</span>
<span id="cb51-79"><a href="#cb51-79" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb51-80"><a href="#cb51-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-81"><a href="#cb51-81" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(validation_matrix) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"metric"</span>, <span class="st">"value"</span>)</span>
<span id="cb51-82"><a href="#cb51-82" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(validation_matrix, <span class="st">"Data/validation_metrics.csv"</span>, <span class="at">row.names =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [38]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>df_validation <span class="ot">&lt;-</span> </span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"../Data/validation_metrics.csv"</span> <span class="sc">%&gt;%</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># "Data/validation_metrics.csv" %&gt;%</span></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read.csv</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>The classification map illustrates the diversity of benthic communities and substrates in the study area (<a href="#fig-Belon" class="quarto-xref">Figure&nbsp;6</a> A and B). Rhodophyceae (red) emerges as the dominant algal cover, forming extensive, continuous patches aligned with the mid-intertidal zones. In contrast, Bacillariophyceae (orange) and Chlorophyceae (green) exhibit more localized distributions, typically restricted to smaller, fragmented patches where specific microtopographic or hydrodynamic conditions favor their presence. Phaeophyceae (brown) is confined to limited patches, often found near transitional zones between sediment and water or in the upper intertidal area, where it is attached to rocky substrates. The water class (blue) delineates the main tidal channel, which meanders through the center of the area and influences the distribution of adjacent habitats. Across the for study sites the presence/absence of G. vermiculophylla were classified with a global accuracy of 91.1 %, a sensitivity of 96.5 % and a specificity of 71.5 %.</p>
<p>The bathymetric map reveals a continuous gradient in elevation relative to mean sea level (<a href="#fig-Belon" class="quarto-xref">Figure&nbsp;6</a> C). A comparison of bathymetry and vegetation distribution highlights a clear elevation-driven pattern in algal presence. Higher intertidal zones, located above the deeper channel areas, are associated with more extensive algal communities. In contrast, lower intertidal zones closer to the channel consistently exhibit reduced macroalgal cover.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [39]:</pre></div><div id="cell-fig-Belon" class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"./Figures/Low_res/Belon_maps.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-Belon" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-Belon-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="./Figures/Low_res/Belon_maps.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6" title="Figure&nbsp;6: DISCOV Prediction (A), RGB composition (B), Bathymetry (C) and mudflat angle (D) of the Bélon estuary site in Brttany, France. The total extent of this flight was 21 hectars with a resolution of 8 mm per pixel. Bathymetry is represented as the height above mean sea level."><img src="./Figures/Low_res/Belon_maps.png" class="img-fluid figure-img" style="width:95.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-Belon-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: DISCOV Prediction (A), RGB composition (B), Bathymetry (C) and mudflat angle (D) of the Bélon estuary site in Brttany, France. The total extent of this flight was 21 hectars with a resolution of 8 mm per pixel. Bathymetry is represented as the height above mean sea level.
</figcaption>
</figure>
</div>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [40]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Utilities.Package)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mgcv)</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidybayes)</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>Full_df<span class="ot">&lt;-</span><span class="fu">read_csv</span>(<span class="st">"Data/Model/FullDF_10timesResample_DEMSlope_PredPercent.csv"</span>)</span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>m_Red <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(Pred <span class="sc">~</span> DEM<span class="sc">+</span>Slope_Cat<span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Site)),</span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">data =</span> Full_df, <span class="at">family =</span> <span class="fu">Beta</span>(), <span class="at">cores =</span> <span class="dv">8</span>, <span class="at">iter =</span> <span class="dv">1000</span>,</span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">warmup =</span> <span class="dv">100</span>, <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9999</span>),</span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">save_pars =</span> <span class="fu">save_pars</span>(<span class="at">latent =</span> <span class="cn">TRUE</span>),</span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">file =</span> <span class="st">"Data/Model/Pred_Beta_agg_1000_DEM_SlopeCat_reSite_resampled"</span>)</span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a>NewData_1<span class="ot">&lt;-</span><span class="fu">expand_grid</span>(<span class="at">DEM=</span><span class="fu">seq</span>(<span class="fu">min</span>(Full_df<span class="sc">$</span>DEM),<span class="fu">max</span>(Full_df<span class="sc">$</span>DEM),<span class="at">length.out=</span><span class="dv">100</span>),</span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a>                       <span class="at">Slope_Cat=</span><span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">"Flat"</span>,</span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Slope"</span>,</span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Vertical"</span>)),</span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a>                       <span class="at">Site=</span><span class="fu">unique</span>(Full_df<span class="sc">$</span>Site))</span>
<span id="cb54-23"><a href="#cb54-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-24"><a href="#cb54-24" aria-hidden="true" tabindex="-1"></a>NewData1_brms<span class="ot">&lt;-</span>NewData_1<span class="sc">%&gt;%</span> </span>
<span id="cb54-25"><a href="#cb54-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_linpred_draws</span>(m_Red, <span class="at">ndraws=</span><span class="dv">2000</span>,<span class="at">seed=</span><span class="dv">123</span>,<span class="at">transform=</span><span class="cn">TRUE</span>)</span>
<span id="cb54-26"><a href="#cb54-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-27"><a href="#cb54-27" aria-hidden="true" tabindex="-1"></a>Total<span class="ot">&lt;-</span>NewData1_brms<span class="sc">%&gt;%</span> </span>
<span id="cb54-28"><a href="#cb54-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(DEM)<span class="sc">%&gt;%</span> </span>
<span id="cb54-29"><a href="#cb54-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">median_qi</span>(.linpred,<span class="at">.width=</span><span class="fu">c</span>(<span class="fl">0.89</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb54-30"><a href="#cb54-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">Slope_Cat=</span><span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">"Flat"</span>,</span>
<span id="cb54-31"><a href="#cb54-31" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Slope"</span>,</span>
<span id="cb54-32"><a href="#cb54-32" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Vertical"</span>)))</span>
<span id="cb54-33"><a href="#cb54-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-34"><a href="#cb54-34" aria-hidden="true" tabindex="-1"></a><span class="co"># (plot &lt;- NewData1_brms%&gt;% </span></span>
<span id="cb54-35"><a href="#cb54-35" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(DEM,Slope_Cat)%&gt;% </span></span>
<span id="cb54-36"><a href="#cb54-36" aria-hidden="true" tabindex="-1"></a><span class="co">#   summarise(median_qi(.linpred,.width=c(0.89)))%&gt;%</span></span>
<span id="cb54-37"><a href="#cb54-37" aria-hidden="true" tabindex="-1"></a><span class="co">#   ggplot(aes(x = DEM,y=y)) +</span></span>
<span id="cb54-38"><a href="#cb54-38" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_ribbon(aes(ymax = ymax,ymin=ymin,color=Slope_Cat),</span></span>
<span id="cb54-39"><a href="#cb54-39" aria-hidden="true" tabindex="-1"></a><span class="co">#               fill = NA,</span></span>
<span id="cb54-40"><a href="#cb54-40" aria-hidden="true" tabindex="-1"></a><span class="co">#               linetype = "dashed") +</span></span>
<span id="cb54-41"><a href="#cb54-41" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_line(aes(colour=Slope_Cat),</span></span>
<span id="cb54-42"><a href="#cb54-42" aria-hidden="true" tabindex="-1"></a><span class="co">#             linewidth=1)+</span></span>
<span id="cb54-43"><a href="#cb54-43" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_line(data = Total,aes(x=DEM,y=y),colour="black")+</span></span>
<span id="cb54-44"><a href="#cb54-44" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_fill_manual(name="Slope",values=c("#8E7CC3","#76A5AF","#93C47D"),</span></span>
<span id="cb54-45"><a href="#cb54-45" aria-hidden="true" tabindex="-1"></a><span class="co">#                     labels=c("Flat",</span></span>
<span id="cb54-46"><a href="#cb54-46" aria-hidden="true" tabindex="-1"></a><span class="co">#                             "Angled",</span></span>
<span id="cb54-47"><a href="#cb54-47" aria-hidden="true" tabindex="-1"></a><span class="co">#                             "Vertical"))+</span></span>
<span id="cb54-48"><a href="#cb54-48" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_colour_manual(name="Slope",values=c("#8E7CC3","#76A5AF","#93C47D"),</span></span>
<span id="cb54-49"><a href="#cb54-49" aria-hidden="true" tabindex="-1"></a><span class="co">#                     labels=c("Flat",</span></span>
<span id="cb54-50"><a href="#cb54-50" aria-hidden="true" tabindex="-1"></a><span class="co">#                             "Angled",</span></span>
<span id="cb54-51"><a href="#cb54-51" aria-hidden="true" tabindex="-1"></a><span class="co">#                             "Vertical"))+</span></span>
<span id="cb54-52"><a href="#cb54-52" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_y_continuous(breaks = c(0.1,0.2,0.3,0.4,0.5),</span></span>
<span id="cb54-53"><a href="#cb54-53" aria-hidden="true" tabindex="-1"></a><span class="co">#                      labels = c("10","20","30","40","50"))+</span></span>
<span id="cb54-54"><a href="#cb54-54" aria-hidden="true" tabindex="-1"></a><span class="co">#   labs(x="Bathymetry (m)",y="Intertidal Area Covered\n by Gracilaria vermiculophylla (%)")+</span></span>
<span id="cb54-55"><a href="#cb54-55" aria-hidden="true" tabindex="-1"></a><span class="co">#   facet_wrap(~Slope_Cat,ncol =2)+</span></span>
<span id="cb54-56"><a href="#cb54-56" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme_Bede()+</span></span>
<span id="cb54-57"><a href="#cb54-57" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme(strip.text = element_blank(),</span></span>
<span id="cb54-58"><a href="#cb54-58" aria-hidden="true" tabindex="-1"></a><span class="co">#         legend.position= c(0.75,0.25), </span></span>
<span id="cb54-59"><a href="#cb54-59" aria-hidden="true" tabindex="-1"></a><span class="co">#         legend.title = element_text(size = 20, margin = margin(b = 20)),</span></span>
<span id="cb54-60"><a href="#cb54-60" aria-hidden="true" tabindex="-1"></a><span class="co">#         legend.text = element_text(size  = 15),</span></span>
<span id="cb54-61"><a href="#cb54-61" aria-hidden="true" tabindex="-1"></a><span class="co">#         legend.key.size = unit(1,"cm"),</span></span>
<span id="cb54-62"><a href="#cb54-62" aria-hidden="true" tabindex="-1"></a><span class="co">#         legend.spacing.y = unit(1, "cm"),</span></span>
<span id="cb54-63"><a href="#cb54-63" aria-hidden="true" tabindex="-1"></a><span class="co">#         axis.text.x = element_text(size = 15),</span></span>
<span id="cb54-64"><a href="#cb54-64" aria-hidden="true" tabindex="-1"></a><span class="co">#         axis.text.y = element_text(size = 15),</span></span>
<span id="cb54-65"><a href="#cb54-65" aria-hidden="true" tabindex="-1"></a><span class="co">#         axis.title.x = element_text(size = 20),</span></span>
<span id="cb54-66"><a href="#cb54-66" aria-hidden="true" tabindex="-1"></a><span class="co">#         axis.title.y = element_text(size = 20), </span></span>
<span id="cb54-67"><a href="#cb54-67" aria-hidden="true" tabindex="-1"></a><span class="co">#         plot.background = element_rect(fill = "white", color = NA))+</span></span>
<span id="cb54-68"><a href="#cb54-68" aria-hidden="true" tabindex="-1"></a><span class="co">#   guides(fill = guide_legend(nrow = 1),</span></span>
<span id="cb54-69"><a href="#cb54-69" aria-hidden="true" tabindex="-1"></a><span class="co">#          colour = guide_legend(nrow = 1))</span></span>
<span id="cb54-70"><a href="#cb54-70" aria-hidden="true" tabindex="-1"></a><span class="co">#   )</span></span>
<span id="cb54-71"><a href="#cb54-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-72"><a href="#cb54-72" aria-hidden="true" tabindex="-1"></a>(plot <span class="ot">&lt;-</span> NewData1_brms<span class="sc">%&gt;%</span> </span>
<span id="cb54-73"><a href="#cb54-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(DEM,Slope_Cat)<span class="sc">%&gt;%</span> </span>
<span id="cb54-74"><a href="#cb54-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">median_qi</span>(.linpred,<span class="at">.width=</span><span class="fu">c</span>(<span class="fl">0.89</span>)))<span class="sc">%&gt;%</span></span>
<span id="cb54-75"><a href="#cb54-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> DEM,<span class="at">y=</span>y)) <span class="sc">+</span></span>
<span id="cb54-76"><a href="#cb54-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">ymax =</span> ymax,<span class="at">ymin=</span>ymin,<span class="at">color=</span>Slope_Cat),</span>
<span id="cb54-77"><a href="#cb54-77" aria-hidden="true" tabindex="-1"></a>              <span class="at">fill =</span> <span class="cn">NA</span>,</span>
<span id="cb54-78"><a href="#cb54-78" aria-hidden="true" tabindex="-1"></a>              <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb54-79"><a href="#cb54-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">colour=</span>Slope_Cat),</span>
<span id="cb54-80"><a href="#cb54-80" aria-hidden="true" tabindex="-1"></a>            <span class="at">linewidth=</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb54-81"><a href="#cb54-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> Total,<span class="fu">aes</span>(<span class="at">x=</span>DEM,<span class="at">y=</span>y),<span class="at">colour=</span><span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb54-82"><a href="#cb54-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name=</span><span class="st">"Slope"</span>,<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#8E7CC3"</span>,<span class="st">"#76A5AF"</span>,<span class="st">"#93C47D"</span>),</span>
<span id="cb54-83"><a href="#cb54-83" aria-hidden="true" tabindex="-1"></a>                    <span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Flat"</span>,</span>
<span id="cb54-84"><a href="#cb54-84" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Angled"</span>,</span>
<span id="cb54-85"><a href="#cb54-85" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Steep"</span>))<span class="sc">+</span></span>
<span id="cb54-86"><a href="#cb54-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">name=</span><span class="st">"Slope"</span>,<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#8E7CC3"</span>,<span class="st">"#76A5AF"</span>,<span class="st">"#93C47D"</span>),</span>
<span id="cb54-87"><a href="#cb54-87" aria-hidden="true" tabindex="-1"></a>                    <span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Flat"</span>,</span>
<span id="cb54-88"><a href="#cb54-88" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Angled"</span>,</span>
<span id="cb54-89"><a href="#cb54-89" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Steep"</span>))<span class="sc">+</span></span>
<span id="cb54-90"><a href="#cb54-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.2</span>,<span class="fl">0.3</span>,<span class="fl">0.4</span>,<span class="fl">0.5</span>),</span>
<span id="cb54-91"><a href="#cb54-91" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"10"</span>,<span class="st">"20"</span>,<span class="st">"30"</span>,<span class="st">"40"</span>,<span class="st">"50"</span>))<span class="sc">+</span></span>
<span id="cb54-92"><a href="#cb54-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Bathymetry (m)"</span>,<span class="at">y=</span><span class="st">"Intertidal Area Covered</span><span class="sc">\n</span><span class="st"> by Gracilaria vermiculophylla (%)"</span>)<span class="sc">+</span></span>
<span id="cb54-93"><a href="#cb54-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Slope_Cat,<span class="at">ncol =</span><span class="dv">3</span>)<span class="sc">+</span></span>
<span id="cb54-94"><a href="#cb54-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_Bede</span>()<span class="sc">+</span></span>
<span id="cb54-95"><a href="#cb54-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.text =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb54-96"><a href="#cb54-96" aria-hidden="true" tabindex="-1"></a>        <span class="co"># legend.title = element_text(size = 17),</span></span>
<span id="cb54-97"><a href="#cb54-97" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb54-98"><a href="#cb54-98" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size  =</span> <span class="dv">15</span>),</span>
<span id="cb54-99"><a href="#cb54-99" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.spacing.x =</span> <span class="fu">unit</span>(<span class="dv">5</span>,<span class="st">"cm"</span>),</span>
<span id="cb54-100"><a href="#cb54-100" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"top"</span>,</span>
<span id="cb54-101"><a href="#cb54-101" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.7</span>,<span class="st">"cm"</span>),</span>
<span id="cb54-102"><a href="#cb54-102" aria-hidden="true" tabindex="-1"></a>        <span class="co"># legend.key.spacing.y = unit(-10, "cm"),</span></span>
<span id="cb54-103"><a href="#cb54-103" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb54-104"><a href="#cb54-104" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb54-105"><a href="#cb54-105" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb54-106"><a href="#cb54-106" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>), </span>
<span id="cb54-107"><a href="#cb54-107" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>))</span>
<span id="cb54-108"><a href="#cb54-108" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb54-109"><a href="#cb54-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-110"><a href="#cb54-110" aria-hidden="true" tabindex="-1"></a>NewData1_brms<span class="sc">%&gt;%</span> </span>
<span id="cb54-111"><a href="#cb54-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(DEM,Slope_Cat)<span class="sc">%&gt;%</span> </span>
<span id="cb54-112"><a href="#cb54-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">median_qi</span>(.linpred,<span class="at">.width=</span><span class="fu">c</span>(<span class="fl">0.89</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb54-113"><a href="#cb54-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Slope_Cat) <span class="sc">%&gt;%</span> </span>
<span id="cb54-114"><a href="#cb54-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">min =</span> <span class="fu">min</span>(y),</span>
<span id="cb54-115"><a href="#cb54-115" aria-hidden="true" tabindex="-1"></a>          <span class="at">max =</span> <span class="fu">max</span>(y))</span>
<span id="cb54-116"><a href="#cb54-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-117"><a href="#cb54-117" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/High_res/GAM_slope_cover.png"</span>, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">5</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span>
<span id="cb54-118"><a href="#cb54-118" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Paper/Figures/Low_res/GAM_slope_cover.png"</span>, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">5</span>, <span class="at">dpi =</span> <span class="dv">200</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Overall, the percent cover of <em>Gracilaria vermiculophylla</em> increases with bathymetry, as shown by the general relationship (<a href="#fig-Gam_Slope" class="quarto-xref">Figure&nbsp;7</a>, black line), which rises from approximately 16% at the lowest elevation to about 30% at the highest elevation. This indicates a consistent positive association between bathymetry and algal cover.</p>
<p>When accounting for slope, the flatter the slope, the higher the percent cover of <em>G. vermiculophylla</em>. For flat slopes, the cover ranges from approximately 20% at the lowest elevation to nearly 38% at the highest elevation. In contrast, for angled slopes, the increase is less pronounced, ranging from around 16% to 32%. On steep slopes, the cover is the lowest, starting at about 15% and rising only slightly above 30% at the highest elevation (<a href="#fig-Gam_Slope" class="quarto-xref">Figure&nbsp;7</a>). This demonstrates that slope modifies the relationship, with flatter slopes supporting a greater percent cover of the algae.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [41]:</pre></div><div id="cell-fig-Gam_Slope" class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"./Figures/Low_res/GAM_slope_cover.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-Gam_Slope" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-Gam_Slope-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="./Figures/Low_res/GAM_slope_cover.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Figure&nbsp;7: DISCOV Prediction (A), RGB composition (B) and Bathymetry (C) of the Bélon estuary site in Brttany, France. The total extent of this flight was 21 hectars with a resolution of 8 mm per pixel. Bathymetry is represented as the height above mean sea level."><img src="./Figures/Low_res/GAM_slope_cover.png" class="img-fluid figure-img" style="width:95.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-Gam_Slope-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: DISCOV Prediction (A), RGB composition (B) and Bathymetry (C) of the Bélon estuary site in Brttany, France. The total extent of this flight was 21 hectars with a resolution of 8 mm per pixel. Bathymetry is represented as the height above mean sea level.
</figcaption>
</figure>
</div>
</div>
</div></div>
</section>
</section>
<section id="discussion" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Discussion</h1>
<section id="drone-mapping-g.-vermiculophylla-with-machine-learning" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="drone-mapping-g.-vermiculophylla-with-machine-learning"><span class="header-section-number">5.1</span> Drone mapping <em>G. vermiculophylla</em> with machine learning</h2>
<p>In this study, we produced the first spatial distribution maps of the invasive red alga <em>Gracilaria vermiculophylla</em> using a multispectral drone survey conducted at low tide in Atlantic estuaries representing varied environmental conditions. In southern Brittany, the species formed monospecific mats, while in the Cantabrian region of Spain, it was intermixed with other intertidal vegetation. Distinguishing among these vegetation types was a key prerequisite for the analysis.</p>
<p>To achieve this, we adapted the deep learning-based classification model DISCOV <span class="citation" data-cites="rs16234383">(<a href="#ref-rs16234383" role="doc-biblioref">Oiry et al., 2024b</a>)</span>, initially developed to discriminate seagrass from green macroalgae. Although the original model included Rhodophyceae as a class, this group constituted less than 3% of its training dataset. In contrast, the updated model presented here was trained on a dataset in which <em>G. vermiculophylla</em> covered 26 % of approximately one million pixels. This improved dataset allowed the model to achieve an accuracy of 91.1 %.</p>
<p>Rhodophytes possess unique phycobilin pigments, enabling their spectral distinction from other macroalgal groups <span class="citation" data-cites="rs14020346 olmedo2020far rs11060704">(<a href="#ref-rs14020346" role="doc-biblioref">Douay et al., 2022</a>; <a href="#ref-rs11060704" role="doc-biblioref">Mcilwaine et al., 2019</a>; <a href="#ref-olmedo2020far" role="doc-biblioref">Olmedo-Masat et al., 2020</a>)</span>. Even with the ten-band multispectral sensor used in our study, it remained feasible to discriminate the major classes of intertidal macrophytes <span class="citation" data-cites="roman2021using rs16234383 davies2023multi">(<a href="#ref-davies2023multi" role="doc-biblioref">Davies et al., 2023</a>; <a href="#ref-rs16234383" role="doc-biblioref">Oiry et al., 2024b</a>; <a href="#ref-roman2021using" role="doc-biblioref">Román et al., 2021</a>)</span>. However, the model identifies <em>G. vermiculophylla</em> at the class level (Rhodophyceae) rather than at the species level. Although hyperspectral approaches may allow finer taxonomic resolution <span class="citation" data-cites="rs14020346 olmedo2020far">(<a href="#ref-rs14020346" role="doc-biblioref">Douay et al., 2022</a>; <a href="#ref-olmedo2020far" role="doc-biblioref">Olmedo-Masat et al., 2020</a>)</span>, it is unlikely that Gracilaria species can be precisely distinguished using standard multispectral sensors.</p>
<p>Ecological factors also aid in differentiating <em>G. vermiculophylla</em>. Unlike many other macroalgae that require hard substrates, <em>G. vermiculophylla</em> establishes itself on soft-bottom sediments. In fact, it is commonly found on mudflats, anchoring its thalli in the top 10 cm of mud <span class="citation" data-cites="surget2017processus">(<a href="#ref-surget2017processus" role="doc-biblioref">Surget, 2017</a>)</span>, and inhabits the upper intertidal zone—an unusual trait for a Rhodophyte <span class="citation" data-cites="abreu2011nitrogen davoult2017multiple">(<a href="#ref-abreu2011nitrogen" role="doc-biblioref">Abreu et al., 2011</a>; <a href="#ref-davoult2017multiple" role="doc-biblioref">Davoult et al., 2017</a>)</span>. By reliably detecting <em>G. vermiculophylla</em> in these soft-substrate, upper intertidal habitats, our method provides a framework for identifying environmental conditions that favor its spread, potentially offering managers early-warning indicators to control its expansion before it reaches nuisance levels. Thus, combining spectral data with sediment characteristics provides a strong indicator of <em>G. vermiculophylla</em> presence in European Atlantic estuaries, complementing the physical variables already used in species distribution modeling <span class="citation" data-cites="jmse11020367">(<a href="#ref-jmse11020367" role="doc-biblioref">Mendoza-Segura et al., 2023</a>)</span>.</p>
<p>In addition, the scalability of drone-based surveying facilitates repeat mapping to detect temporal shifts in the distribution and abundance of <em>G. vermiculophylla.</em> Such continuous monitoring could capture seasonal patterns of colonization, allowing researchers and environmental managers to evaluate the effectiveness of mitigation measures, track long-term ecological impacts, and anticipate future shifts in habitat suitability under changing climate conditions.</p>
</section>
<section id="g.-vermiculophylla-spatial-distribution-and-mudflat-topography" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="g.-vermiculophylla-spatial-distribution-and-mudflat-topography"><span class="header-section-number">5.2</span> <em>G. vermiculophylla</em> spatial distribution and mudflat topography</h2>
</section>
<section id="spatio-temporal-monitoring-of-invasive-macroalgae" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="spatio-temporal-monitoring-of-invasive-macroalgae"><span class="header-section-number">5.3</span> Spatio-temporal monitoring of invasive macroalgae</h2>
<p>Accurate, high-resolution maps of invasive or alien species are extremely scarce <span class="citation" data-cites="fourcade2014mapping vilizzi2021global">(<a href="#ref-fourcade2014mapping" role="doc-biblioref">Fourcade et al., 2014</a>; <a href="#ref-vilizzi2021global" role="doc-biblioref">Vilizzi et al., 2021</a>)</span>, yet they enable in-depth evaluations of these species’ ecology, temporal dynamics, and niche behavior in relation to their environment. In this study, using individual flights over monospecific algal mats, we quantified how this invasive alga associates with local mudflat topography, demonstrating that its distribution is closely tied to specific topographical features, such as elevation and slope gradients. This relationship reveals how physical features of the mudflat can influence the extent and density of <em>Gracilaria vermiculophylla</em> mats. Understanding the drivers of the spatial distribution of the algae helps to anticipate the future development of this species and identify areas vulnerable to its invasion. Repeated monitoring of this type can further reveal phenological patterns, invasion dynamics, and local conspecific biological interactions—such as co-occurrence, displacement, or avoidance <span class="citation" data-cites="godoy2009different arim2006spread">(<a href="#ref-arim2006spread" role="doc-biblioref">Arim et al., 2006</a>; <a href="#ref-godoy2009different" role="doc-biblioref">Godoy et al., 2009</a>)</span>.</p>
<p>Remote sensing using multispectral drone mapping can provide high-resolution, spatially explicit data, but it must be combined with repeated, in situ field measurements to maximize its potential <span class="citation" data-cites="chadwick2020integrating zoffoli2023remote">(<a href="#ref-chadwick2020integrating" role="doc-biblioref">Chadwick et al., 2020</a>; <a href="#ref-zoffoli2023remote" role="doc-biblioref">Zoffoli et al., 2023</a>)</span>. As noted, temporal repetition makes it possible to assess dynamic processes, and integrating these mapping approaches with in situ analyses of local infauna, carbon cycling, riverine inputs, and sedimentology would yield valuable insights for local managers. Such an integrated approach could help determine how the invasive algae affects the local ecosystem and, more broadly, forecast its potential impact on other estuarine environments facing similar invasion events. This holistic approach can guide strategic interventions aimed at mitigating the alga’s spread, maintaining ecological balance, and protecting native biodiversity, ensuring that management efforts are informed by accurate, timely, and spatially explicit data.</p>
<p>Invasive species like <em>Gracilaria vermiculophylla</em> and <em>Rugulopteryx okamurae</em> can be identified using drones equipped with multispectral sensors, taking advantage of the characteristic reflectance of rhodophytes <span class="citation" data-cites="barille2025variability rs15143674">(<a href="#ref-barille2025variability" role="doc-biblioref">Barillé et al., 2025</a>; <a href="#ref-rs15143674" role="doc-biblioref">Nurdin et al., 2023</a>)</span>. However, while RGB sensors on readily available commercial drones can be used to perform accurate scene classification, their effectiveness for mapping invasive species has not yet been demonstrated <span class="citation" data-cites="cheng2017remote kazakeviciute2020assessment">(<a href="#ref-cheng2017remote" role="doc-biblioref">Cheng et al., 2017</a>; <a href="#ref-kazakeviciute2020assessment" role="doc-biblioref">Kazakeviciute-Januskeviciene et al., 2020</a>)</span>. These drones are easy to deploy, can cover large areas when flying at speeds of 15 m s<sup>-1</sup> at an altitude of 120 m, and still maintain sufficient overlap between images to support photogrammetric reconstruction. Expanding these methodologies to RGB-based detection would significantly lower barriers to entry, allowing local stakeholders with limited resources to access valuable monitoring tools for early detection and rapid response. A promising avenue for operational applications lies in testing machine learning techniques for coastal habitat mapping using RGB imagery that do not rely on enhanced spectral resolution. Considering the low cost of RGB and multispectral commercial drones, coupled with ongoing advancements in machine learning, drone-based remote sensing has now matured into a practical tool for adoption by environmental authorities in coastal management. Integrating these technologies into routine monitoring protocols can enhance surveillance capabilities, improve understanding of invasive species dynamics, and ultimately contribute to more effective conservation and restoration strategies.</p>
</section>
</section>
<section id="conclusion" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Conclusion</h1>
</section>
<section id="references" class="level1 unnumbered" data-number="7">
<h1 class="unnumbered" data-number="7"><span class="header-section-number">7</span> References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-abreu2011nitrogen" class="csl-entry" role="listitem">
Abreu, M.H., Pereira, R., Buschmann, A., Sousa-Pinto, I., Yarish, C., 2011. Nitrogen uptake responses of gracilaria vermiculophylla (ohmi) papenfuss under combined and single addition of nitrate and ammonium. Journal of Experimental Marine Biology and Ecology 407, 190–199.
</div>
<div id="ref-agisoft" class="csl-entry" role="listitem">
Agisoft, 2019. <a href="https://www.agisoft.com/">Agisoft metashape</a>.
</div>
<div id="ref-arim2006spread" class="csl-entry" role="listitem">
Arim, M., Abades, S.R., Neill, P.E., Lima, M., Marquet, P.A., 2006. Spread dynamics of invasive species. Proceedings of the National Academy of Sciences 103, 374–378.
</div>
<div id="ref-barille2025variability" class="csl-entry" role="listitem">
Barillé, L., Paterson, I.L., Oiry, S., Aris, A., Cook-Cottier, E.J., Nurdin, N., 2025. Variability of kappaphycus alvarezii cultivation in south-sulawesi (indonesia) related to the monsoon shift: Water quality, growth and colour quantification. Aquaculture Reports 40, 102557.
</div>
<div id="ref-Blanchet2014" class="csl-entry" role="listitem">
Blanchet, H., Gouillieux, B., Alizier, S., others, 2014. Multiscale patterns in the diversity and organization of benthic intertidal fauna among french atlantic estuaries. Journal of Sea Research 90, 95–110. <a href="https://doi.org/10.1016/j.seares.2014.02.014">https://doi.org/10.1016/j.seares.2014.02.014</a>
</div>
<div id="ref-brunier2022evolution" class="csl-entry" role="listitem">
Brunier, G., Tamura, T., Anthony, E.J., Dussouillez, P., Gardel, A., 2022. Evolution of the french guiana coast from late pleistocene to holocene based on chenier and beach sand dating. Regional Environmental Change 22, 122.
</div>
<div id="ref-calleja2017long" class="csl-entry" role="listitem">
Calleja, F., Galván, C., Silió-Calzada, A., Juanes, J.A., Ondiviela, B., 2017. Long-term analysis of zostera noltei: A retrospective approach for understanding seagrasses’ dynamics. Marine environmental research 130, 93–105.
</div>
<div id="ref-Castaing1995" class="csl-entry" role="listitem">
Castaing, P., Guilcher, A., 1995. Morphosedimentary evolution of ria-type estuaries. Earth Surface Processes and Landforms 20, 361–376. <a href="https://doi.org/10.1002/esp.3290200408">https://doi.org/10.1002/esp.3290200408</a>
</div>
<div id="ref-chadwick2020integrating" class="csl-entry" role="listitem">
Chadwick, K.D., Brodrick, P.G., Grant, K., Goulden, T., Henderson, A., Falco, N., Wainwright, H., Williams, K.H., Bill, M., Breckheimer, I., others, 2020. Integrating airborne remote sensing and field campaigns for ecology and earth system science. Methods in Ecology and Evolution 11, 1492–1508.
</div>
<div id="ref-chand2021low" class="csl-entry" role="listitem">
Chand, S., Bollard, B., 2021. Low altitude spatial assessment and monitoring of intertidal seagrass meadows beyond the visible spectrum using a remotely piloted aircraft system. Estuarine, Coastal and Shelf Science 255, 107299.
</div>
<div id="ref-shinypck" class="csl-entry" role="listitem">
Chang, W., Cheng, J., Allaire, J., Sievert, C., Schloerke, B., Xie, Y., Allen, J., McPherson, J., Dipert, A., Borges, B., 2024. <a href="https://CRAN.R-project.org/package=shiny">Shiny: Web application framework for r</a>.
</div>
<div id="ref-cheng2017remote" class="csl-entry" role="listitem">
Cheng, G., Han, J., Lu, X., 2017. Remote sensing image scene classification: Benchmark and state of the art. Proceedings of the IEEE 105, 1865–1883.
</div>
<div id="ref-davies2023multi" class="csl-entry" role="listitem">
Davies, B.F.R., Gernez, P., Geraud, A., Oiry, S., Rosa, P., Zoffoli, M.L., Barillé, L., 2023. Multi-and hyperspectral classification of soft-bottom intertidal vegetation using a spectral library for coastal biodiversity remote sensing. Remote Sensing of Environment 290, 113554.
</div>
<div id="ref-davies2024sentinel" class="csl-entry" role="listitem">
Davies, B.F.R., Oiry, S., Rosa, P., Zoffoli, M.L., Sousa, A.I., Thomas, O.R., Smale, D.A., Austen, M.C., Biermann, L., Attrill, M.J., others, 2024b. A sentinel watching over inter-tidal seagrass phenology across western europe and north africa. Communications Earth &amp; Environment 5, 382.
</div>
<div id="ref-davies2024intertidal" class="csl-entry" role="listitem">
Davies, B.F.R., Oiry, S., Rosa, P., Zoffoli, M.L., Sousa, A.I., Thomas, O.R., Smale, D.A., Austen, M.C., Biermann, L., Attrill, M.J., others, 2024a. Intertidal seagrass extent from sentinel-2 time-series show distinct trajectories in western europe. Remote Sensing of Environment 312, 114340.
</div>
<div id="ref-davoult2017multiple" class="csl-entry" role="listitem">
Davoult, D., Surget, G., Stiger-Pouvreau, V., Noisette, F., Riera, P., Stagnol, D., Androuin, T., Poupart, N., 2017. Multiple effects of a gracilaria vermiculophylla invasion on estuarine mudflat functioning and diversity. Marine Environmental Research 131, 227–235.
</div>
<div id="ref-rs14133124" class="csl-entry" role="listitem">
Diruit, W., Le Bris, A., Bajjouk, T., Richier, S., Helias, M., Burel, T., Lennon, M., Guyot, A., Ar Gall, E., 2022. Seaweed habitats on the shore: Characterization through hyperspectral UAV imagery and field sampling. Remote Sensing 14. <a href="https://doi.org/10.3390/rs14133124">https://doi.org/10.3390/rs14133124</a>
</div>
<div id="ref-rs14020346" class="csl-entry" role="listitem">
Douay, F., Verpoorter, C., Duong, G., Spilmont, N., Gevaert, F., 2022. New hyperspectral procedure to discriminate intertidal macroalgae. Remote Sensing 14. <a href="https://doi.org/10.3390/rs14020346">https://doi.org/10.3390/rs14020346</a>
</div>
<div id="ref-douglas2024linking" class="csl-entry" role="listitem">
Douglas, T.J., Coops, N.C., Drever, M.C., Hunt, B.P., Martin, T.G., 2024. Linking microphytobenthos distribution and mudflat geomorphology under varying sedimentary regimes using unoccupied aerial vehicle (UAV)-acquired multispectral reflectance and photogrammetry. Science of The Total Environment 173675.
</div>
<div id="ref-duffy2018spatial" class="csl-entry" role="listitem">
Duffy, J.P., Pratt, L., Anderson, K., Land, P.E., Shutler, J.D., 2018. Spatial assessment of intertidal seagrass meadows using optical imaging systems and a lightweight drone. Estuarine, Coastal and Shelf Science 200, 169–180.
</div>
<div id="ref-firth2024invasive" class="csl-entry" role="listitem">
Firth, L.B., Foggo, A., Watts, T., Knights, A.M., DeAmicis, S., 2024. Invasive macroalgae in native seagrass beds: Vectors of spread and impacts. Annals of Botany 133, 41–50.
</div>
<div id="ref-fourcade2014mapping" class="csl-entry" role="listitem">
Fourcade, Y., Engler, J.O., Rödder, D., Secondi, J., 2014. Mapping species distributions with MAXENT using a geographically biased sample of presence data: A performance assessment of methods for correcting sampling bias. PloS one 9, e97122.
</div>
<div id="ref-van2018global" class="csl-entry" role="listitem">
Ginneken, V. van, Vries, E. de, others, 2018. The global dispersal of the non-endemic invasive red alga gracilaria vermiculophylla in the ecosystems of the euro-asia coastal waters including the wadden sea unesco world heritage coastal area: Awful or awesome? Oceanography &amp; Fisheries Open Access Journal 8, 4–26.
</div>
<div id="ref-godoy2009different" class="csl-entry" role="listitem">
Godoy, O., Castro-Dı́ez, P., Valladares, F., Costa-Tenorio, M., 2009. Different flowering phenology of alien invasive species in spain: Evidence for the use of an empty temporal niche? Plant Biology 11, 803–811.
</div>
<div id="ref-grizel1991introduction" class="csl-entry" role="listitem">
Grizel, H., Heral, M., 1991. Introduction into france of the japanese oyster (crassostrea gigas). ICES Journal of Marine Science 47, 399–403.
</div>
<div id="ref-gurgel2018systematics" class="csl-entry" role="listitem">
Gurgel, C.F.D., Norris, J.N., Schmidt, W.E., Le, H.N., Fredericq, S., 2018. Systematics of the gracilariales (rhodophyta) including new subfamilies, tribes, subgenera, and two new genera, agarophyton gen. Nov. And crassa gen. nov. Phytotaxa 374, 1–23.
</div>
<div id="ref-terrapck" class="csl-entry" role="listitem">
Hijmans, R.J., 2024. <a href="https://CRAN.R-project.org/package=terra">Terra: Spatial data analysis</a>.
</div>
<div id="ref-RemonterLeTempsIGN" class="csl-entry" role="listitem">
IGN, 2024. Remonter le temps.
</div>
<div id="ref-kazakeviciute2020assessment" class="csl-entry" role="listitem">
Kazakeviciute-Januskeviciene, G., Janusonis, E., Bausys, R., Limba, T., Kiskis, M., 2020. Assessment of the segmentation of RGB remote sensing images: A subjective approach. Remote Sensing 12, 4152.
</div>
<div id="ref-krueger2017genetic" class="csl-entry" role="listitem">
Krueger-Hadfield, S.A., Kollars, N.M., Strand, A.E., Byers, J.E., Shainker, S.J., Terada, R., Greig, T.W., Hammann, M., Murray, D.C., Weinberger, F., others, 2017. Genetic identification of source and likely vector of a widespread marine invader. Ecology and evolution 7, 4432–4447.
</div>
<div id="ref-d15020161" class="csl-entry" role="listitem">
Massé, C., Viard, F., Humbert, S., Antajan, E., Auby, I., Bachelet, G., Bernard, G., Bouchet, V.M.P., Burel, T., Dauvin, J.-C., Delegrange, A., Derrien-Courtel, S., Droual, G., Gouillieux, B., Goulletquer, P., Guérin, L., Janson, A.-L., Jourde, J., Labrune, C., Lavesque, N., Leclerc, J.-C., Le Duff, M., Le Garrec, V., Noël, P., Nowaczyk, A., Pergent-Martini, C., Pezy, J.-P., Raoux, A., Raybaud, V., Ruitton, S., Sauriau, P.-G., Spilmont, N., Thibault, D., Vincent, D., Curd, A., 2023. An overview of marine non-indigenous species found in three contrasting biogeographic metropolitan french regions: Insights on distribution, origins and pathways of introduction. Diversity 15. <a href="https://doi.org/10.3390/d15020161">https://doi.org/10.3390/d15020161</a>
</div>
<div id="ref-rs11060704" class="csl-entry" role="listitem">
Mcilwaine, B., Casado, M.R., Leinster, P., 2019. Using 1st derivative reflectance signatures within a remote sensing framework to identify macroalgae in marine environments. Remote Sensing 11. <a href="https://doi.org/10.3390/rs11060704">https://doi.org/10.3390/rs11060704</a>
</div>
<div id="ref-jmse11020367" class="csl-entry" role="listitem">
Mendoza-Segura, C., Fernández, E., Beca-Carretero, P., 2023. Predicted changes in the biogeographical range of gracilaria vermiculophylla under present and future climate scenarios. Journal of Marine Science and Engineering 11. <a href="https://doi.org/10.3390/jmse11020367">https://doi.org/10.3390/jmse11020367</a>
</div>
<div id="ref-Michel2021" class="csl-entry" role="listitem">
Michel, G., Le Bot, S., Lesourd, S., Lafite, R., 2021. Morpho-sedimentological and dynamic patterns in a ria type estuary: The belon estuary (south brittany, france). Journal of Maps 17, 389–400. <a href="https://doi.org/10.1080/17445647.2021.1925170">https://doi.org/10.1080/17445647.2021.1925170</a>
</div>
<div id="ref-nebel2020review" class="csl-entry" role="listitem">
Nebel, S., Beege, M., Schneider, S., Rey, G.D., 2020. A review of photogrammetry and photorealistic 3D models in education from a psychological perspective, in: Frontiers in Education. Frontiers Media SA, p. 144.
</div>
<div id="ref-rs15143674" class="csl-entry" role="listitem">
Nurdin, N., Alevizos, E., Syamsuddin, R., Asis, H., Zainuddin, E.N., Aris, A., Oiry, S., Brunier, G., Komatsu, T., Barillé, L., 2023. Precision aquaculture drone mapping of the spatial distribution of kappaphycus alvarezii biomass and carrageenan. Remote Sensing 15. <a href="https://doi.org/10.3390/rs15143674">https://doi.org/10.3390/rs15143674</a>
</div>
<div id="ref-nyberg2007introduced" class="csl-entry" role="listitem">
Nyberg, C.D., 2007. Introduced marine macroalgae and habitat modifiers: Their ecological role and significant attributes. Department of Marine Ecology.
</div>
<div id="ref-nyberg2009flora" class="csl-entry" role="listitem">
Nyberg, C.D., Thomsen, M.S., Wallentinus, I., 2009. Flora and fauna associated with the introduced red alga gracilaria vermiculophylla. European Journal of Phycology 44, 395–403.
</div>
<div id="ref-ohmi1956contributions" class="csl-entry" role="listitem">
OHMI, H., 1956. CONTRIBUTIONS TO THE KNOWLEDGE OF GRACILARIACEAE FROM JAPAN: Ⅱ. On a new species of the genus gracilariopsis, with some considerations on its ecology. 北海道大學水産學部研究彙報 6, 271–279.
</div>
<div id="ref-oiry_2024_14218984" class="csl-entry" role="listitem">
Oiry, S., Davies, B.F.R., Pierre, G., Laurent, B., 2024a. <span>DISCOV: Drone Intertidal Sediment Classification Of Vegetation</span>. <a href="https://doi.org/10.5281/zenodo.14218984">https://doi.org/10.5281/zenodo.14218984</a>
</div>
<div id="ref-rs16234383" class="csl-entry" role="listitem">
Oiry, S., Davies, B.F.R., Sousa, A.I., Rosa, P., Zoffoli, M.L., Brunier, G., Gernez, P., Barillé, L., 2024b. Discriminating seagrasses from green macroalgae in european intertidal areas using high-resolution multispectral drone imagery. Remote Sensing 16. <a href="https://doi.org/10.3390/rs16234383">https://doi.org/10.3390/rs16234383</a>
</div>
<div id="ref-olmedo2020far" class="csl-entry" role="listitem">
Olmedo-Masat, O.M., Raffo, M.P., Rodrı́guez-Pérez, D., Arijón, M., Sánchez-Carnero, N., 2020. How far can we classify macroalgae remotely? An example using a new spectral library of species from the south west atlantic (argentine patagonia). Remote Sensing 12, 3870.
</div>
<div id="ref-ortega2005fluxes" class="csl-entry" role="listitem">
Ortega, T., Ponce, R., Forja, J., Gómez-Parra, A., 2005. Fluxes of dissolved inorganic carbon in three estuarine systems of the cantabrian sea (north of spain). Journal of Marine Systems 53, 125–142.
</div>
<div id="ref-WoRMS303450" class="csl-entry" role="listitem">
Papenfuss, G.F., 1967. <a href="https://marinespecies.org/aphia.php?p=sourcedetails&amp;id=303450">Notes on algal nomenclature - v. Various chlorophyceae and rhodophyceae</a>. Phykos 5, 95–105.
</div>
<div id="ref-peidro2024quantifying" class="csl-entry" role="listitem">
Peidro-Devesa, M.J., Martı́nez-Movilla, A., Rodrı́guez-Somoza, J.L., Sánchez, J.M., Román, M., 2024. Quantifying intertidal macroalgae stocks in the NW iberian peninsula using unmanned aerial vehicle (UAV) multispectral imagery. Regional Studies in Marine Science 103621.
</div>
<div id="ref-ramus2017invasive" class="csl-entry" role="listitem">
Ramus, A.P., Silliman, B.R., Thomsen, M.S., Long, Z.T., 2017. An invasive foundation species enhances multifunctionality in a coastal ecosystem. Proceedings of the national academy of sciences 114, 8580–8585.
</div>
<div id="ref-roca2022monitoring" class="csl-entry" role="listitem">
Roca, M., Dunbar, M.B., Román, A., Caballero, I., Zoffoli, M.L., Gernez, P., Navarro, G., 2022. Monitoring the marine invasive alien species rugulopteryx okamurae using unmanned aerial vehicles and satellites. Frontiers in Marine Science 9, 1004012.
</div>
<div id="ref-roman2024mapping" class="csl-entry" role="listitem">
Román, A., Oiry, S., Davies, B.F., Rosa, P., Gernez, P., Tovar-Sánchez, A., Navarro, G., Méléder, V., Barillé, L., 2024. Mapping intertidal microphytobenthic biomass with very high-resolution remote sensing imagery in an estuarine system. Science of The Total Environment 177025.
</div>
<div id="ref-roman2021using" class="csl-entry" role="listitem">
Román, A., Tovar-Sánchez, A., Olivé, I., Navarro, G., 2021. Using a UAV-mounted multispectral camera for the monitoring of marine macrophytes. Frontiers in Marine Science 8, 722698.
</div>
<div id="ref-romero2008sintering" class="csl-entry" role="listitem">
Romero, M., Andrés, A., Alonso, R., Viguri, J., Rincón, J.M., 2008. Sintering behaviour of ceramic bodies from contaminated marine sediments. Ceramics International 34, 1917–1924.
</div>
<div id="ref-rueness2005life" class="csl-entry" role="listitem">
Rueness, J., 2005. Life history and molecular sequences of gracilaria vermiculophylla (gracilariales, rhodophyta), a new introduction to european waters. Phycologia 44, 120–128.
</div>
<div id="ref-savitzky1964smoothing" class="csl-entry" role="listitem">
Savitzky, A., Golay, M.J., 1964. Smoothing and differentiation of data by simplified least squares procedures. Analytical chemistry 36, 1627–1639.
</div>
<div id="ref-sfriso2012spreading" class="csl-entry" role="listitem">
Sfriso, A., Wolf, M.A., Maistro, S., Sciuto, K., Moro, I., 2012. Spreading and autoecology of the invasive species gracilaria vermiculophylla (gracilariales, rhodophyta) in the lagoons of the north-western adriatic sea (mediterranean sea, italy). Estuarine, Coastal and Shelf Science 114, 192–198.
</div>
<div id="ref-Simon2024ShinyApp" class="csl-entry" role="listitem">
Simon, O., 2024. <a href="https://oirysimon.shinyapps.io/shiny_validate/">Shiny app for validation dataset building</a>.
</div>
<div id="ref-sotka2018combining" class="csl-entry" role="listitem">
Sotka, E.E., Baumgardner, A.W., Bippus, P.M., Destombe, C., Duermit, E.A., Endo, H., Flanagan, B.A., Kamiya, M., Lees, L.E., Murren, C.J., others, 2018. Combining niche shift and population genetic analyses predicts rapid phenotypic evolution during invasion. Evolutionary Applications 11, 781–793.
</div>
<div id="ref-surget2017processus" class="csl-entry" role="listitem">
Surget, G., 2017. Processus adaptatifs des v<span>é</span>g<span>é</span>taux marins face au changement climatique <span class="nocase">à</span> diff<span>é</span>rentes <span class="nocase">é</span>chelles de temps et d’espace: Dynamique de populations, m<span>é</span>tabolomique, <span class="nocase">é</span>cophysiologie et potentiels de valorisation (PhD thesis). Universit<span>é</span> de Bretagne occidentale-Brest.
</div>
<div id="ref-Tankoua2011" class="csl-entry" role="listitem">
Tankoua, O.F., Buffet, P.-E., Amiard, J.-C., Amiard-Triquet, C., Mouneyrac, C., Berthet, B., 2011. Potential influence of confounding factors (size, salinity) on biomarkers in the sentinel species scrobicularia plana used in programmes monitoring estuarine quality. Environmental Science and Pollution Research 18, 1253–1263. <a href="https://doi.org/10.1007/s11356-011-0479-3">https://doi.org/10.1007/s11356-011-0479-3</a>
</div>
<div id="ref-terada2002review" class="csl-entry" role="listitem">
Terada, R., Yamamoto, H., 2002. Review of gracilaria vermiculophylla (ohmi) papenfuss and other species in japan and asia. Taxonomy of economic seaweeds, with special reference to Pacific species 8, 225–230.
</div>
<div id="ref-thomsen2007gracilaria" class="csl-entry" role="listitem">
Thomsen, M.S., Staehr, P.A., Nyberg, C.D., Schwærter, S., Krause-Jensen, D., Silliman, B.R., 2007. Gracilaria vermiculophylla (ohmi) papenfuss, 1967 (rhodophyta, gracilariaceae) in northern europe, with emphasis on danish conditions, and what to expect in the future. Aquatic invasions 2, 83–94.
</div>
<div id="ref-thomsen2013effects" class="csl-entry" role="listitem">
Thomsen, M.S., Stæhr, P.A., Nejrup, L., Schiel, D.R., 2013. Effects of the invasive macroalgae gracilaria vermiculophylla on two co-occurring foundation species and associated invertebrates. Aquatic Invasions 8, 133–145.
</div>
<div id="ref-valle2015mapping" class="csl-entry" role="listitem">
Valle, M., Pala, V., Lafon, V., Dehouck, A., Garmendia, J.M., Borja, A., Chust, G., 2015. Mapping estuarine habitats using airborne hyperspectral imagery, with special focus on seagrass meadows. Estuarine, Coastal and Shelf Science 164, 433–442.
</div>
<div id="ref-van2003reintroduction" class="csl-entry" role="listitem">
Van Katwijk, M., 2003. Reintroduction of eelgrass (zostera marina l.) in the dutch wadden sea: A research overview and management vision, in: Challenges to the Wadden Sea Area. In: Proceedings of the 10th International Scientific Wadden Sea Symposium, Groningen, the Netherlands. pp. 173–195.
</div>
<div id="ref-vilizzi2021global" class="csl-entry" role="listitem">
Vilizzi, L., Copp, G.H., Hill, J.E., Adamovich, B., Aislabie, L., Akin, D., Al-Faisal, A.J., Almeida, D., Azmai, M.A., Bakiu, R., others, 2021. A global-scale screening of non-native aquatic organisms to identify potentially invasive species under current and future climate conditions. Science of the Total Environment 788, 147868.
</div>
<div id="ref-weinberger2008invasive" class="csl-entry" role="listitem">
Weinberger, F., Buchholz, B., Karez, R., Wahl, M., 2008. The invasive red alga gracilaria vermiculophylla in the baltic sea: Adaptation to brackish water may compensate for light limitation. Aquatic Biology 3, 251–264.
</div>
<div id="ref-williams2007global" class="csl-entry" role="listitem">
Williams, S.L., Smith, J.E., 2007. A global review of the distribution, taxonomy, and impacts of introduced seaweeds. Annu. Rev. Ecol. Evol. Syst. 38, 327–359.
</div>
<div id="ref-d14121077" class="csl-entry" role="listitem">
Zenetos, A., Tsiamis, K., Galanidi, M., Carvalho, N., Bartilotti, C., Canning-Clode, J., Castriota, L., Chainho, P., Comas-González, R., Costa, A.C., Dragičević, B., Dulčić, J., Faasse, M., Florin, A.-B., Gittenberger, A., Jakobsen, H., Jelmert, A., Kerckhof, F., Lehtiniemi, M., Livi, S., Lundgreen, K., Macic, V., Massé, C., Mavrič, B., Naddafi, R., Orlando-Bonaca, M., Petovic, S., Png-Gonzalez, L., Carbonell Quetglas, A., Ribeiro, R.S., Cidade, T., Smolders, S., Stæhr, P.A.U., Viard, F., Outinen, O., 2022. Status and trends in the rate of introduction of marine non-indigenous species in european seas. Diversity 14. <a href="https://doi.org/10.3390/d14121077">https://doi.org/10.3390/d14121077</a>
</div>
<div id="ref-zoffoli2021decadal" class="csl-entry" role="listitem">
Zoffoli, M.L., Gernez, P., Godet, L., Peters, S., Oiry, S., Barillé, L., 2021. Decadal increase in the ecological status of a north-atlantic intertidal seagrass meadow observed with multi-mission satellite time-series. Ecological Indicators 130, 108033.
</div>
<div id="ref-zoffoli2023remote" class="csl-entry" role="listitem">
Zoffoli, M.L., Gernez, P., Oiry, S., Godet, L., Dalloyau, S., Davies, B.F.R., Barillé, L., 2023. Remote sensing in seagrass ecology: Coupled dynamics between migratory herbivorous birds and intertidal meadows observed by satellite during four decades. Remote Sensing in Ecology and Conservation 9, 420–433.
</div>
</div>
</section>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr>
<ol>
<li id="fn1"><p>Institut des Substances et Organismes de la Mer, ISOMer, Nantes Université, UR 2160, F-44000 Nantes, France<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script> 
  
</body></html>